diff -Naur openexr-2.2.0-orig/CMakeLists.txt openexr-2.2.0/CMakeLists.txt
--- openexr-2.2.0-orig/CMakeLists.txt	2013-11-25 23:49:53.000000000 +0400
+++ openexr-2.2.0/CMakeLists.txt	2014-05-28 16:55:20.527800000 +0400
@@ -40,7 +40,7 @@
 FIND_PACKAGE(ZLIB REQUIRED)
 INCLUDE_DIRECTORIES(${ZLIB_INCLUDE_DIR})
 
-IF (NOT WIN32)
+IF (NOT WIN32 OR MINGW)
   SET ( PTHREAD_LIB pthread )
 ENDIF()
 
@@ -107,7 +107,9 @@
   PROPERTIES
   VERSION 22.0.0
   SOVERSION 22
-  OUTPUT_NAME "IlmImf${OPENEXR_LIBSUFFIX}"
+  OUTPUT_NAME IlmImf
+  RUNTIME_OUTPUT_NAME "IlmImf${OPENEXR_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME IlmImf
   )
   
 ##########################
@@ -188,7 +188,9 @@
   PROPERTIES
   VERSION 22.0.0
   SOVERSION 22
-  OUTPUT_NAME "IlmImfUtil${OPENEXR_LIBSUFFIX}"
+  OUTPUT_NAME IlmImfUtil
+  RUNTIME_OUTPUT_NAME "IlmImfUtil${OPENEXR_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME IlmImfUtil
   )
   
 ##########################
@@ -177,3 +180,24 @@
   DESTINATION
   ${CMAKE_INSTALL_PREFIX}/share/doc/OpenEXR-${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}/examples
   )
+
+FILE ( WRITE ${CMAKE_BINARY_DIR}/OpenEXR.pc "prefix=${CMAKE_INSTALL_PREFIX}\n" )
+FILE ( APPEND ${CMAKE_BINARY_DIR}/OpenEXR.pc "exec_prefix=\${prefix}
+libdir=\${exec_prefix}/lib
+includedir=\${prefix}/include
+OpenEXR_includedir=\${prefix}/include/OpenEXR
+
+Name: OpenEXR
+Description: OpenEXR image library
+Version: ${OPENEXR_VERSION}
+Libs: -L\${libdir} -lIlmImf
+Cflags: -I\${OpenEXR_includedir}
+Requires: IlmBase
+Libs.private: -lz
+")
+
+INSTALL ( FILES
+  ${CMAKE_BINARY_DIR}/OpenEXR.pc
+  DESTINATION
+  ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig
+)
diff -Naur openexr-2.2.0-orig/exrenvmap/CMakeLists.txt openexr-2.2.0/exrenvmap/CMakeLists.txt
--- openexr-2.2.0-orig/exrenvmap/CMakeLists.txt	2013-11-25 23:49:55.000000000 +0400
+++ openexr-2.2.0/exrenvmap/CMakeLists.txt	2014-05-28 15:33:02.099600000 +0400
@@ -12,8 +12,8 @@
 
 TARGET_LINK_LIBRARIES ( exrenvmap
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff -Naur openexr-2.2.0-orig/exrheader/CMakeLists.txt openexr-2.2.0/exrheader/CMakeLists.txt
--- openexr-2.2.0-orig/exrheader/CMakeLists.txt	2013-11-25 23:49:55.000000000 +0400
+++ openexr-2.2.0/exrheader/CMakeLists.txt	2014-05-28 15:33:40.620800000 +0400
@@ -6,8 +6,8 @@
 
 TARGET_LINK_LIBRARIES ( exrheader
   IlmImf
-  Iex${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  IlmThread
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff -Naur openexr-2.2.0-orig/exrmakepreview/CMakeLists.txt openexr-2.2.0/exrmakepreview/CMakeLists.txt
--- openexr-2.2.0-orig/exrmakepreview/CMakeLists.txt	2013-11-25 23:49:56.000000000 +0400
+++ openexr-2.2.0/exrmakepreview/CMakeLists.txt	2014-05-28 15:34:16.131600000 +0400
@@ -7,8 +7,8 @@
 
 TARGET_LINK_LIBRARIES ( exrmakepreview
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff -Naur openexr-2.2.0-orig/exrmaketiled/CMakeLists.txt openexr-2.2.0/exrmaketiled/CMakeLists.txt
--- openexr-2.2.0-orig/exrmaketiled/CMakeLists.txt	2013-11-25 23:49:56.000000000 +0400
+++ openexr-2.2.0/exrmaketiled/CMakeLists.txt	2014-05-28 15:35:22.033400000 +0400
@@ -8,8 +8,8 @@
 
 TARGET_LINK_LIBRARIES ( exrmaketiled
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff -Naur openexr-2.2.0-orig/exrmultipart/CMakeLists.txt openexr-2.2.0/exrmultipart/CMakeLists.txt
--- openexr-2.2.0-orig/exrmultipart/CMakeLists.txt	2013-11-25 23:49:56.000000000 +0400
+++ openexr-2.2.0/exrmultipart/CMakeLists.txt	2014-05-28 15:35:34.029800000 +0400
@@ -6,8 +6,8 @@
 
 TARGET_LINK_LIBRARIES ( exrmultipart
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff -Naur openexr-2.2.0-orig/exrmultiview/CMakeLists.txt openexr-2.2.0/exrmultiview/CMakeLists.txt
--- openexr-2.2.0-orig/exrmultiview/CMakeLists.txt	2013-11-25 23:49:56.000000000 +0400
+++ openexr-2.2.0/exrmultiview/CMakeLists.txt	2014-05-28 15:35:48.928400000 +0400
@@ -9,9 +9,9 @@
 TARGET_LINK_LIBRARIES ( exrmultiview
   IlmImf
   Half
-  Imath${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Imath
+  Iex
+  IlmThread
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
 )
diff -Naur openexr-2.2.0-orig/exrstdattr/CMakeLists.txt openexr-2.2.0/exrstdattr/CMakeLists.txt
--- openexr-2.2.0-orig/exrstdattr/CMakeLists.txt	2013-11-25 23:49:56.000000000 +0400
+++ openexr-2.2.0/exrstdattr/CMakeLists.txt	2014-05-28 15:36:00.868400000 +0400
@@ -6,8 +6,8 @@
 
 TARGET_LINK_LIBRARIES ( exrstdattr
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff -Naur openexr-2.2.0-orig/IlmImf/CMakeLists.txt openexr-2.2.0/IlmImf/CMakeLists.txt
--- openexr-2.2.0-orig/IlmImf/CMakeLists.txt	2013-11-25 23:49:53.000000000 +0400
+++ openexr-2.2.0/IlmImf/CMakeLists.txt	2014-05-28 15:51:53.296000000 +0400
@@ -6,8 +6,8 @@
   )
 TARGET_LINK_LIBRARIES ( b44ExpLogTable
   Half
-  Iex${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  IlmThread
   ${PTHREAD_LIB}
 )
 
@@ -25,8 +25,8 @@
 
 TARGET_LINK_LIBRARIES ( dwaLookups
   Half
-  Iex${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  IlmThread
   ${PTHREAD_LIB}
 )
 
@@ -138,9 +138,9 @@
 
 TARGET_LINK_LIBRARIES ( IlmImf 
   Half 
-  Iex${ILMBASE_LIBSUFFIX}
-  Imath${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  Imath
+  IlmThread
   ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
 )
 
@@ -143,8 +143,9 @@
 
 INSTALL ( TARGETS
   IlmImf
-  DESTINATION
-  ${CMAKE_INSTALL_PREFIX}/lib
+  RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+  LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
 )
 
 # Headers
diff -Naur openexr-2.2.0-orig/IlmImfExamples/CMakeLists.txt openexr-2.2.0/IlmImfExamples/CMakeLists.txt
--- openexr-2.2.0-orig/IlmImfExamples/CMakeLists.txt	2013-11-25 23:49:53.000000000 +0400
+++ openexr-2.2.0/IlmImfExamples/CMakeLists.txt	2014-05-28 15:26:49.508000000 +0400
@@ -15,9 +15,9 @@
 TARGET_LINK_LIBRARIES ( IlmImfExamples 
         IlmImf
         Half 
-        Iex${ILMBASE_LIBSUFFIX}
-        Imath${ILMBASE_LIBSUFFIX}
-        IlmThread${ILMBASE_LIBSUFFIX}
+        Iex
+        Imath
+        IlmThread
         ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
         )
   
diff -Naur openexr-2.2.0-orig/IlmImfFuzzTest/CMakeLists.txt openexr-2.2.0/IlmImfFuzzTest/CMakeLists.txt
--- openexr-2.2.0-orig/IlmImfFuzzTest/CMakeLists.txt	2013-11-25 23:49:53.000000000 +0400
+++ openexr-2.2.0/IlmImfFuzzTest/CMakeLists.txt	2014-05-28 15:36:27.060200000 +0400
@@ -12,9 +12,9 @@
 TARGET_LINK_LIBRARIES ( IlmImfFuzzTest 
         IlmImf
         Half
-        Iex${ILMBASE_LIBSUFFIX}
-        Imath${ILMBASE_LIBSUFFIX}
-        IlmThread${ILMBASE_LIBSUFFIX}
+        Iex
+        Imath
+        IlmThread
         ${PTHREAD_LIB} ${ZLIB_LIBRARIES})
 
 ADD_TEST ( TestIlmImfFuzz IlmImfFuzzTest )
diff -Naur openexr-2.2.0-orig/IlmImfTest/CMakeLists.txt openexr-2.2.0/IlmImfTest/CMakeLists.txt
--- openexr-2.2.0-orig/IlmImfTest/CMakeLists.txt	2013-11-25 23:49:53.000000000 +0400
+++ openexr-2.2.0/IlmImfTest/CMakeLists.txt	2014-05-28 15:36:51.041600000 +0400
@@ -66,9 +66,9 @@
 TARGET_LINK_LIBRARIES ( IlmImfTest 
         IlmImf  
         Half 
-        Iex${ILMBASE_LIBSUFFIX}
-        Imath${ILMBASE_LIBSUFFIX}
-        IlmThread${ILMBASE_LIBSUFFIX}
+        Iex
+        Imath
+        IlmThread
         ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
         )
 
diff -Naur openexr-2.2.0-orig/IlmImfUtil/CMakeLists.txt openexr-2.2.0/IlmImfUtil/CMakeLists.txt
--- openexr-2.2.0-orig/IlmImfUtil/CMakeLists.txt	2013-11-25 23:49:53.000000000 +0400
+++ openexr-2.2.0/IlmImfUtil/CMakeLists.txt	2014-05-28 15:36:51.041600000 +0400
@@ -28,9 +28,9 @@
 
 TARGET_LINK_LIBRARIES ( IlmImfUtil 
   Half 
-  Iex${ILMBASE_LIBSUFFIX}
-  Imath${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  Imath
+  IlmThread
   IlmImf
   ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
 )
@@ -40,8 +40,9 @@
 
 INSTALL ( TARGETS
   IlmImfUtil
-  DESTINATION
-  ${CMAKE_INSTALL_PREFIX}/lib
+  RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+  LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
 )
 
 # Headers
diff -Naur openexr-2.2.0-orig/IlmImfUtilTest/CMakeLists.txt openexr-2.2.0/IlmImfUtilTest/CMakeLists.txt
--- openexr-2.2.0-orig/IlmImfUtilTest/CMakeLists.txt	2013-11-25 23:49:53.000000000 +0400
+++ openexr-2.2.0/IlmImfUtilTest/CMakeLists.txt	2014-05-28 15:36:27.060200000 +0400
@@ -17,9 +17,9 @@
         IlmImfUtil
         IlmImf  
         Half 
-        Iex${ILMBASE_LIBSUFFIX}
-        Imath${ILMBASE_LIBSUFFIX}
-        IlmThread${ILMBASE_LIBSUFFIX}
+        Iex
+        Imath
+        IlmThread
         ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
         )
 
