diff -rupN openexr-2.1.0/IlmImf/Makefile.am openexr-2.1.0-new/IlmImf/Makefile.am
--- openexr-2.1.0/IlmImf/Makefile.am	2013-11-12 00:09:51.000000000 +0100
+++ openexr-2.1.0-new/IlmImf/Makefile.am	2013-12-27 00:48:46.944680594 +0100
@@ -187,9 +187,4 @@ CLEANFILES = b44ExpLogTable b44ExpLogTab
 b44ExpLogTable_SOURCES = b44ExpLogTable.cpp
 b44ExpLogTable_LDADD = @ILMBASE_LDFLAGS@ @ILMBASE_LIBS@
 
-b44ExpLogTable.h: b44ExpLogTable
-	./b44ExpLogTable > b44ExpLogTable.h
-
-BUILT_SOURCES = b44ExpLogTable.h
-
 noinst_PROGRAMS = b44ExpLogTable
