diff -rupN openexr-2.1.0/IlmImf/ImfOptimizedPixelReading.h openexr-2.1.0-new/IlmImf/ImfOptimizedPixelReading.h
--- openexr-2.1.0/IlmImf/ImfOptimizedPixelReading.h	2013-07-29 19:58:48.000000000 +0200
+++ openexr-2.1.0-new/IlmImf/ImfOptimizedPixelReading.h	2013-12-27 02:04:23.286563455 +0100
@@ -84,7 +84,7 @@ EXR_FORCEINLINE
 bool
 isPointerSSEAligned (const void* EXR_RESTRICT pPointer)
 {
-    unsigned long trailingBits = ((unsigned long)pPointer) & 15;
+    intptr_t trailingBits = ((intptr_t)pPointer) & 15;
     return trailingBits == 0;
 }
 
