--- config/Makefile.darwin.orig	2015-08-10 21:47:22.000000000 +1000
+++ config/Makefile.darwin	2015-08-12 12:17:06.000000000 +1000
@@ -2,9 +2,9 @@
 CC = cc
 LD = cc
 CFLAGS.EXTRA = -dynamic -fno-common
-CFLAGS.EXTRA += -ansi -pedantic
+CFLAGS.EXTRA += -ansi -pedantic $(CFLAGS.ARCH)
 #CFLAGS.EXTRA += -no-cpp-precomp
-LDFLAGS.EXTRA =
+LDFLAGS.EXTRA = $(LDFLAGS.ARCH)
 ifneq (undefined, $(origin GLEW_APPLE_GLX))
 CFLAGS.EXTRA += -I/usr/X11R6/include -D'GLEW_APPLE_GLX'
 LDFLAGS.GL = -L/usr/X11R6/lib -lGL -lX11
