--- src/glewinfo.c.orig	2015-08-10 21:47:22.000000000 +1000
+++ src/glewinfo.c	2016-02-24 06:17:57.000000000 +1100
@@ -11847,7 +11847,7 @@ GLboolean glewCreateContext (struct crea
   i = 0;
   contextAttrs[i++] = kCGLPFAAccelerated; /* No software rendering */
 
-  #if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_7
+  #if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
   if (params->profile & GL_CONTEXT_CORE_PROFILE_BIT)
   {
     if (params->major==3 && params->minor>=2)
