From c2106991da30800f074cfa17ea48d0ec75579976 Mon Sep 17 00:00:00 2001
From: dirk <dirk@git.imagemagick.org>
Date: Mon, 25 Jan 2016 23:22:15 +0100
Subject: [PATCH] Read XCF layer name.

---
 ChangeLog    |  6 +++++-
 coders/xcf.c | 21 +++++++++++++--------
 2 files changed, 18 insertions(+), 9 deletions(-)

diff --git a/coders/xcf.c b/coders/xcf.c
index f635769..5a9247e 100644
--- a/coders/xcf.c
+++ b/coders/xcf.c
@@ -54,6 +54,7 @@
 #include "magick/memory_.h"
 #include "magick/pixel.h"
 #include "magick/pixel-accessor.h"
+#include "magick/property.h"
 #include "magick/quantize.h"
 #include "magick/quantum-private.h"
 #include "magick/static.h"
@@ -766,6 +767,15 @@ static MagickBooleanType load_hierarchy(Image *image,XCFDocInfo *inDocInfo,
   return(MagickTrue);
 }
 
+static void InitXCFImage(XCFLayerInfo *outLayer)
+{
+  outLayer->image->page.x=outLayer->offset_x;
+  outLayer->image->page.y=outLayer->offset_y;
+  outLayer->image->page.width=outLayer->width;
+  outLayer->image->page.height=outLayer->height;
+  (void) SetImageProperty(outLayer->image,"label",(char *)outLayer->name);
+}
+
 static MagickBooleanType ReadOneLayer(const ImageInfo *image_info,Image* image,
   XCFDocInfo* inDocInfo,XCFLayerInfo *outLayer,const ssize_t layer)
 {
@@ -891,10 +901,7 @@ static MagickBooleanType ReadOneLayer(const ImageInfo *image_info,Image* image,
           outLayer->image=CloneImage(image,0,0,MagickTrue,&image->exception);
           if (outLayer->image == (Image *) NULL)
             return(MagickFalse);
-          outLayer->image->page.x=outLayer->offset_x;
-          outLayer->image->page.y=outLayer->offset_y;
-          outLayer->image->page.width=outLayer->width;
-          outLayer->image->page.height=outLayer->height;
+          InitXCFImage(outLayer);
           return(MagickTrue);
         }
     }
@@ -908,10 +915,8 @@ static MagickBooleanType ReadOneLayer(const ImageInfo *image_info,Image* image,
     ScaleCharToQuantum((unsigned char) (255-outLayer->alpha));
   (void) SetImageBackgroundColor(outLayer->image);
 
-  outLayer->image->page.x=outLayer->offset_x;
-  outLayer->image->page.y=outLayer->offset_y;
-  outLayer->image->page.width=outLayer->width;
-  outLayer->image->page.height=outLayer->height;
+  InitXCFImage(outLayer);
+
   /* set the compositing mode */
   outLayer->image->compose = GIMPBlendModeToCompositeOperator( outLayer->mode );
   if ( outLayer->visible == MagickFalse )
