diff --git a/OpenEXR/CMakeLists.txt b/OpenEXR/CMakeLists.txt
index d3e3365..6790aa7 100644
--- a/OpenEXR/CMakeLists.txt
+++ b/OpenEXR/CMakeLists.txt
@@ -174,7 +174,9 @@ SET_TARGET_PROPERTIES ( IlmImf
   PROPERTIES
   VERSION 22.0.0
   SOVERSION 22
-  OUTPUT_NAME "IlmImf${OPENEXR_LIBSUFFIX}"
+  OUTPUT_NAME IlmImf
+  RUNTIME_OUTPUT_NAME "IlmImf${OPENEXR_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME IlmImf
   )
   
 ##########################
@@ -186,7 +188,9 @@ SET_TARGET_PROPERTIES ( IlmImfUtil
   PROPERTIES
   VERSION 22.0.0
   SOVERSION 22
-  OUTPUT_NAME "IlmImfUtil${OPENEXR_LIBSUFFIX}"
+  OUTPUT_NAME IlmImfUtil
+  RUNTIME_OUTPUT_NAME "IlmImfUtil${OPENEXR_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME IlmImfUtil
   )
   
 ##########################
@@ -258,3 +262,24 @@ INSTALL ( FILES
   DESTINATION
   ${CMAKE_INSTALL_PREFIX}/share/doc/OpenEXR-${OPENEXR_VERSION}/examples
   )
+
+FILE ( WRITE ${CMAKE_BINARY_DIR}/OpenEXR.pc "prefix=${CMAKE_INSTALL_PREFIX}\n" )
+FILE ( APPEND ${CMAKE_BINARY_DIR}/OpenEXR.pc "exec_prefix=\${prefix}
+libdir=\${exec_prefix}/lib
+includedir=\${prefix}/include
+OpenEXR_includedir=\${prefix}/include/OpenEXR
+
+Name: OpenEXR
+Description: OpenEXR image library
+Version: ${OPENEXR_VERSION}
+Libs: -L\${libdir} -lIlmImf
+Cflags: -I\${OpenEXR_includedir}
+Requires: IlmBase
+Libs.private: -lz
+")
+
+INSTALL ( FILES
+  ${CMAKE_BINARY_DIR}/OpenEXR.pc
+  DESTINATION
+  ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig
+)
diff --git a/OpenEXR/IlmImf/CMakeLists.txt b/OpenEXR/IlmImf/CMakeLists.txt
index 18d90ed..dca0961 100644
--- a/OpenEXR/IlmImf/CMakeLists.txt
+++ b/OpenEXR/IlmImf/CMakeLists.txt
@@ -8,8 +8,8 @@ ADD_EXECUTABLE ( b44ExpLogTable
 
 TARGET_LINK_LIBRARIES ( b44ExpLogTable
   Half
-  Iex${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  IlmThread
   ${PTHREAD_LIB}
 )
 
@@ -25,8 +25,8 @@ ADD_EXECUTABLE ( dwaLookups
 
 TARGET_LINK_LIBRARIES ( dwaLookups
   Half
-  Iex${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  IlmThread
   ${PTHREAD_LIB}
 )
 
@@ -138,9 +138,9 @@ ADD_LIBRARY ( IlmImf ${LIB_TYPE}
 
 TARGET_LINK_LIBRARIES ( IlmImf 
   Half 
-  Iex${ILMBASE_LIBSUFFIX}
-  Imath${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  Imath
+  IlmThread
   ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
 )
 
@@ -166,8 +166,9 @@ SET_SOURCE_FILES_PROPERTIES (
 
 INSTALL ( TARGETS
   IlmImf
-  DESTINATION
-  ${CMAKE_INSTALL_PREFIX}/lib
+  RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+  LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
 )
 
 # Headers
diff --git a/OpenEXR/IlmImfExamples/CMakeLists.txt b/OpenEXR/IlmImfExamples/CMakeLists.txt
index 03061cb..54a0a54 100644
--- a/OpenEXR/IlmImfExamples/CMakeLists.txt
+++ b/OpenEXR/IlmImfExamples/CMakeLists.txt
@@ -15,9 +15,9 @@ ADD_EXECUTABLE ( IlmImfExamples
 TARGET_LINK_LIBRARIES ( IlmImfExamples 
         IlmImf
         Half 
-        Iex${ILMBASE_LIBSUFFIX}
-        Imath${ILMBASE_LIBSUFFIX}
-        IlmThread${ILMBASE_LIBSUFFIX}
+        Iex
+        Imath
+        IlmThread
         ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
         )
   
diff --git a/OpenEXR/IlmImfFuzzTest/CMakeLists.txt b/OpenEXR/IlmImfFuzzTest/CMakeLists.txt
index 12fe072..e4bf39b 100644
--- a/OpenEXR/IlmImfFuzzTest/CMakeLists.txt
+++ b/OpenEXR/IlmImfFuzzTest/CMakeLists.txt
@@ -12,9 +12,9 @@ ADD_EXECUTABLE ( IlmImfFuzzTest
 TARGET_LINK_LIBRARIES ( IlmImfFuzzTest 
         IlmImf
         Half
-        Iex${ILMBASE_LIBSUFFIX}
-        Imath${ILMBASE_LIBSUFFIX}
-        IlmThread${ILMBASE_LIBSUFFIX}
+        Iex
+        Imath
+        IlmThread
         ${PTHREAD_LIB} ${ZLIB_LIBRARIES})
 
 ADD_TEST ( TestIlmImfFuzz IlmImfFuzzTest )
diff --git a/OpenEXR/IlmImfTest/CMakeLists.txt b/OpenEXR/IlmImfTest/CMakeLists.txt
index c07d9f2..b1ef763 100644
--- a/OpenEXR/IlmImfTest/CMakeLists.txt
+++ b/OpenEXR/IlmImfTest/CMakeLists.txt
@@ -66,9 +66,9 @@ ADD_TEST ( TestIlmImf IlmImfTest )
 TARGET_LINK_LIBRARIES ( IlmImfTest 
         IlmImf  
         Half 
-        Iex${ILMBASE_LIBSUFFIX}
-        Imath${ILMBASE_LIBSUFFIX}
-        IlmThread${ILMBASE_LIBSUFFIX}
+        Iex
+        Imath
+        IlmThread
         ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
         )
 
diff --git a/OpenEXR/IlmImfUtil/CMakeLists.txt b/OpenEXR/IlmImfUtil/CMakeLists.txt
index 3eca4bc..9e30bac 100644
--- a/OpenEXR/IlmImfUtil/CMakeLists.txt
+++ b/OpenEXR/IlmImfUtil/CMakeLists.txt
@@ -28,9 +28,9 @@ ADD_LIBRARY ( IlmImfUtil ${LIB_TYPE}
 
 TARGET_LINK_LIBRARIES ( IlmImfUtil 
   Half 
-  Iex${ILMBASE_LIBSUFFIX}
-  Imath${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  Imath
+  IlmThread
   IlmImf
   ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
 )
@@ -40,8 +40,9 @@ TARGET_LINK_LIBRARIES ( IlmImfUtil
 
 INSTALL ( TARGETS
   IlmImfUtil
-  DESTINATION
-  ${CMAKE_INSTALL_PREFIX}/lib
+  RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+  LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
 )
 
 # Headers
diff --git a/OpenEXR/IlmImfUtilTest/CMakeLists.txt b/OpenEXR/IlmImfUtilTest/CMakeLists.txt
index 3663dc0..1aef059 100644
--- a/OpenEXR/IlmImfUtilTest/CMakeLists.txt
+++ b/OpenEXR/IlmImfUtilTest/CMakeLists.txt
@@ -17,9 +17,9 @@ TARGET_LINK_LIBRARIES ( IlmImfUtilTest
         IlmImfUtil
         IlmImf  
         Half 
-        Iex${ILMBASE_LIBSUFFIX}
-        Imath${ILMBASE_LIBSUFFIX}
-        IlmThread${ILMBASE_LIBSUFFIX}
+        Iex
+        Imath
+        IlmThread
         ${PTHREAD_LIB} ${ZLIB_LIBRARIES}
         )
 
diff --git a/OpenEXR/exrenvmap/CMakeLists.txt b/OpenEXR/exrenvmap/CMakeLists.txt
index 70ebd83..c604a61 100644
--- a/OpenEXR/exrenvmap/CMakeLists.txt
+++ b/OpenEXR/exrenvmap/CMakeLists.txt
@@ -12,8 +12,8 @@ ADD_EXECUTABLE ( exrenvmap
 
 TARGET_LINK_LIBRARIES ( exrenvmap
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff --git a/OpenEXR/exrheader/CMakeLists.txt b/OpenEXR/exrheader/CMakeLists.txt
index 1953836..fb9fb97 100644
--- a/OpenEXR/exrheader/CMakeLists.txt
+++ b/OpenEXR/exrheader/CMakeLists.txt
@@ -6,8 +6,8 @@ ADD_EXECUTABLE ( exrheader
 
 TARGET_LINK_LIBRARIES ( exrheader
   IlmImf
-  Iex${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Iex
+  IlmThread
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff --git a/OpenEXR/exrmakepreview/CMakeLists.txt b/OpenEXR/exrmakepreview/CMakeLists.txt
index 78b9278..ae33b6f 100644
--- a/OpenEXR/exrmakepreview/CMakeLists.txt
+++ b/OpenEXR/exrmakepreview/CMakeLists.txt
@@ -7,8 +7,8 @@ ADD_EXECUTABLE ( exrmakepreview
 
 TARGET_LINK_LIBRARIES ( exrmakepreview
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff --git a/OpenEXR/exrmaketiled/CMakeLists.txt b/OpenEXR/exrmaketiled/CMakeLists.txt
index 1e8f00d..4e1acf8 100644
--- a/OpenEXR/exrmaketiled/CMakeLists.txt
+++ b/OpenEXR/exrmaketiled/CMakeLists.txt
@@ -8,8 +8,8 @@ ADD_EXECUTABLE ( exrmaketiled
 
 TARGET_LINK_LIBRARIES ( exrmaketiled
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff --git a/OpenEXR/exrmultipart/CMakeLists.txt b/OpenEXR/exrmultipart/CMakeLists.txt
index 4f013cd..7d74c13 100644
--- a/OpenEXR/exrmultipart/CMakeLists.txt
+++ b/OpenEXR/exrmultipart/CMakeLists.txt
@@ -6,8 +6,8 @@ ADD_EXECUTABLE ( exrmultipart
 
 TARGET_LINK_LIBRARIES ( exrmultipart
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
diff --git a/OpenEXR/exrmultiview/CMakeLists.txt b/OpenEXR/exrmultiview/CMakeLists.txt
index 4c71437..cef0293 100644
--- a/OpenEXR/exrmultiview/CMakeLists.txt
+++ b/OpenEXR/exrmultiview/CMakeLists.txt
@@ -9,9 +9,9 @@ ADD_EXECUTABLE ( exrmultiview
 TARGET_LINK_LIBRARIES ( exrmultiview
   IlmImf
   Half
-  Imath${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
-  IlmThread${ILMBASE_LIBSUFFIX}
+  Imath
+  Iex
+  IlmThread
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
 )
diff --git a/OpenEXR/exrstdattr/CMakeLists.txt b/OpenEXR/exrstdattr/CMakeLists.txt
index f8904dd..b34c0c1 100644
--- a/OpenEXR/exrstdattr/CMakeLists.txt
+++ b/OpenEXR/exrstdattr/CMakeLists.txt
@@ -6,8 +6,8 @@ ADD_EXECUTABLE ( exrstdattr
 
 TARGET_LINK_LIBRARIES ( exrstdattr
   IlmImf
-  IlmThread${ILMBASE_LIBSUFFIX}
-  Iex${ILMBASE_LIBSUFFIX}
+  IlmThread
+  Iex
   Half
   ${PTHREAD_LIB}
   ${ZLIB_LIBRARIES}
