--- PKGBUILD.orig	2019-03-10 17:00:44.000000000 -0700
+++ PKGBUILD	2019-03-10 17:03:03.000000000 -0700
@@ -3,8 +3,8 @@
 # Contributer: wirx6 <wirx654@gmail.com>
 
 _realname=ffmpeg
-pkgbase="mingw-w64-${_realname}"
-pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgbase="mingw-w64-${_realname}-lgpl"
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}-lgpl"
 pkgver=4.1.1
 pkgrel=1
 pkgdesc="Complete and free Internet live audio and video broadcasting solution (mingw-w64)"
@@ -23,22 +23,21 @@
          "${MINGW_PACKAGE_PREFIX}-libcaca"
          "${MINGW_PACKAGE_PREFIX}-libmfx"
          "${MINGW_PACKAGE_PREFIX}-libmodplug"
+	 "${MINGW_PACKAGE_PREFIX}-librsvg"
          "${MINGW_PACKAGE_PREFIX}-libtheora"
          "${MINGW_PACKAGE_PREFIX}-libvorbis"
          "${MINGW_PACKAGE_PREFIX}-libvpx"
          "${MINGW_PACKAGE_PREFIX}-libwebp"
          "${MINGW_PACKAGE_PREFIX}-openal"
-         "${MINGW_PACKAGE_PREFIX}-opencore-amr"
+	 "${MINGW_PACKAGE_PREFIX}-openh264"
          "${MINGW_PACKAGE_PREFIX}-openjpeg2"
          "${MINGW_PACKAGE_PREFIX}-opus"
          "${MINGW_PACKAGE_PREFIX}-rtmpdump"
          "${MINGW_PACKAGE_PREFIX}-SDL2"
+	 "${MINGW_PACKAGE_PREFIX}-snappy"
          "${MINGW_PACKAGE_PREFIX}-speex"
          "${MINGW_PACKAGE_PREFIX}-srt"
          "${MINGW_PACKAGE_PREFIX}-wavpack"
-         "${MINGW_PACKAGE_PREFIX}-x264"
-         "${MINGW_PACKAGE_PREFIX}-x265"
-         "${MINGW_PACKAGE_PREFIX}-xvidcore"
          "${MINGW_PACKAGE_PREFIX}-zlib")
 makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
              "${MINGW_PACKAGE_PREFIX}-pkg-config"
@@ -65,7 +64,7 @@
   CC=${MINGW_PREFIX}/bin/gcc \
   CXX=${MINGW_PREFIX}/bin/g++ \
   ../${_realname}-${pkgver}/configure \
-    --prefix=${MINGW_PREFIX} \
+    --prefix=${MINGW_PREFIX}/ffmpeg-lgpl \
     --target-os=mingw32 \
     --arch=${CARCH%%-*} \
     --disable-debug \
@@ -74,7 +73,6 @@
     --enable-d3d11va \
     --enable-fontconfig \
     --enable-gnutls \
-    --enable-gpl \
     --enable-libaom \
     --enable-libass \
     --enable-libbluray \
@@ -85,29 +83,25 @@
     --enable-libmfx \
     --enable-libmodplug \
     --enable-libmp3lame \
-    --enable-libopencore_amrnb \
-    --enable-libopencore_amrwb \
+    --enable-libopenh264 \
     --enable-libopenjpeg \
     --enable-libopus \
+    --enable-librsvg \
     --enable-librtmp \
+    --enable-libsnappy \
     --enable-libspeex \
     --enable-libsrt \
     --enable-libtheora \
     --enable-libvorbis \
-    --enable-libx264 \
-    --enable-libx265 \
-    --enable-libxvid \
     --enable-libvpx \
     --enable-libwebp \
     --enable-openal \
     --enable-libwavpack \
     --enable-pic \
-    --enable-postproc \
     --enable-runtime-cpudetect \
     --enable-shared \
     --enable-static \
     --enable-swresample \
-    --enable-version3 \
     --enable-zlib \
     --disable-doc
     #--enable-pthreads
@@ -119,18 +113,18 @@
   cd "${srcdir}/build-${MINGW_CHOST}"
   # workaround for conflict with SDL main(), use it if you have SDL installed
   #make check CC_C="-c -Umain"
-  make check
+  #make check
 }
 
 package() {
   cd "${srcdir}/build-${MINGW_CHOST}"
   make DESTDIR="${pkgdir}" install
-  mv "${pkgdir}${MINGW_PREFIX}"/bin/*.lib "${pkgdir}${MINGW_PREFIX}"/lib/
+  mv "${pkgdir}${MINGW_PREFIX}/ffmpeg-lgpl"/bin/*.lib "${pkgdir}${MINGW_PREFIX}/ffmpeg-lgpl"/lib/
 
-  #find ${pkgdir}${MINGW_PREFIX}/bin -type f -name "*.exe" -exec objcopy --subsystem console {} \;
-  rm -f ${pkgdir}${MINGW_PREFIX}/lib/*.def
-  rm -f ${pkgdir}${MINGW_PREFIX}/lib/*.lib
+  #find ${pkgdir}${MINGW_PREFIX}/ffmpeg-lgpl/bin -type f -name "*.exe" -exec objcopy --subsystem console {} \;
+  rm -f ${pkgdir}${MINGW_PREFIX}/ffmpeg-lgpl/lib/*.def
+  rm -f ${pkgdir}${MINGW_PREFIX}/ffmpeg-lgpl/lib/*.lib
   
   local PREFIX_DEPS=$(cygpath -am ${MINGW_PREFIX})
-  find ${pkgdir}${MINGW_PREFIX}/lib/pkgconfig -name *.pc -exec sed -i -e"s|${PREFIX_DEPS}|${MINGW_PREFIX}|g" {} \;
+  find ${pkgdir}${MINGW_PREFIX}/ffmpeg-lgpl/lib/pkgconfig -name *.pc -exec sed -i -e"s|${PREFIX_DEPS}|${MINGW_PREFIX}/ffmpeg-lgpl|g" {} \;
 }
