From bae8f4ee7a280ee059868f86af135f6308899e7c Mon Sep 17 00:00:00 2001
From: xantares <xantares09@hotmail.com>
Date: Sun, 7 Jun 2015 18:58:55 +0200
Subject: [PATCH 4/5] fix header generation

---
 IlmImf/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/OpenEXR/IlmImf/CMakeLists.txt b/OpenEXR/IlmImf/CMakeLists.txt
index dba6037..31296b7 100644
--- a/OpenEXR/IlmImf/CMakeLists.txt
+++ b/OpenEXR/IlmImf/CMakeLists.txt
@@ -37,7 +37,7 @@ TARGET_LINK_LIBRARIES ( b44ExpLogTable
 
   ADD_CUSTOM_COMMAND (
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/b44ExpLogTable.h
-    COMMAND $<TARGET_FILE:b44ExpLogTable> > ${CMAKE_CURRENT_BINARY_DIR}/b44ExpLogTable.h
+    COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:b44ExpLogTable> > ${CMAKE_CURRENT_BINARY_DIR}/b44ExpLogTable.h
     DEPENDS b44ExpLogTable
   )
 
@@ -71,7 +71,7 @@ TARGET_LINK_LIBRARIES ( dwaLookups
 
   ADD_CUSTOM_COMMAND (
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/dwaLookups.h
-    COMMAND $<TARGET_FILE:dwaLookups> > ${CMAKE_CURRENT_BINARY_DIR}/dwaLookups.h
+    COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:dwaLookups> > ${CMAKE_CURRENT_BINARY_DIR}/dwaLookups.h
     DEPENDS dwaLookups
   )
 ELSE ()
From 90de52ef5856c276848dc2c85dd4985f9e2639f4 Mon Sep 17 00:00:00 2001
From: Stephen <stephen.kiazyk@thinkboxsoftware.com>
Date: Wed, 24 Feb 2016 14:32:49 -0600
Subject: [PATCH] Add missing OpenEXR headers to CMake install rule

---
 IlmImf/CMakeLists.txt | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/OpenEXR/IlmImf/CMakeLists.txt b/OpenEXR/IlmImf/CMakeLists.txt
index 18d90ed..11f7a1a 100644
--- a/OpenEXR/IlmImf/CMakeLists.txt
+++ b/OpenEXR/IlmImf/CMakeLists.txt
@@ -350,6 +350,29 @@ INSTALL ( FILES
     ImfDeepImageState.h
     ImfDeepImageStateAttribute.h
     ImfFloatVectorAttribute.h
+    ImfAutoArray.h
+    ImfCheckedArithmetic.h
+    ImfCompressor.h
+    ImfDwaCompressor.h
+    ImfDwaCompressorSimd.h
+    ImfFastHuf.h
+    ImfInputPartData.h
+    ImfInputStreamMutex.h
+    ImfOptimizedPixelReading.h
+    ImfOutputPartData.h
+    ImfOutputStreamMutex.h
+    ImfPizCompressor.h
+    ImfPxr24Compressor.h
+    ImfRle.h
+    ImfRleCompressor.h
+    ImfScanLineInputFile.h
+    ImfSimd.h
+    ImfStdIO.h
+    ImfSystemSpecific.h
+    ImfTiledMisc.h
+    ImfTileOffsets.h
+    ImfZip.h
+    ImfZipCompressor.h
 
   DESTINATION
     include/OpenEXR
