--- gettext-tools/libgettextpo/Makefile.in.orig	2015-07-10 01:40:00.000000000 -0500
+++ gettext-tools/libgettextpo/Makefile.in	2015-07-12 18:40:45.000000000 -0500
@@ -415,7 +412,7 @@
 @WOE32_TRUE@am__DEPENDENCIES_1 = libgettextpo.res.lo
 am__DEPENDENCIES_2 =
 libgettextpo_la_DEPENDENCIES = libgnu.la $(am__DEPENDENCIES_1) \
-	$(am__DEPENDENCIES_2)
+	$(am__DEPENDENCIES_2) ../gnulib-lib/stpncpy.lo
 am__libgettextpo_la_SOURCES_DIST = gettext-po.c ../src/str-list.c \
 	../src/dir-list.c ../src/message.c ../src/msgl-ascii.c \
 	../src/po-error.c ../src/po-xerror.c ../src/write-catalog.c \
@@ -1859,7 +1856,7 @@
 # How to build libgettextpo.la.
 # Don't make it depend on libgettextsrc.la or libgettextlib.la, since they
 # define an uncontrolled amount of symbols.
-libgettextpo_la_LIBADD = libgnu.la $(WOE32_LIBADD) $(LTLIBUNISTRING)
+libgettextpo_la_LIBADD = libgnu.la $(WOE32_LIBADD) $(LTLIBUNISTRING) ../gnulib-lib/stpncpy.lo
 libgettextpo_la_LDFLAGS = -version-info \
 	$(LTV_CURRENT):$(LTV_REVISION):$(LTV_AGE) -rpath $(libdir) \
 	@LTLIBINTL@ @LTLIBICONV@ @LTLIBEXPAT@ -lc -no-undefined \
