--- Portfile.orig	2016-05-09 10:14:04.000000000 +0200
+++ Portfile	2016-05-09 10:14:26.000000000 +0200
@@ -22,7 +22,7 @@
 checksums               rmd160  c978990189bf8fd925b4dbe76c7a0eac0b7e7434 \
                         sha256  cf112a1fb02f5b1c0fce5cab11ea8243852c139e669c44014125874b14b7dfaa
 
-depends_lib             port:jpeg \
+depends_lib             path:lib/libjpeg.dylib:jpeg \
                         port:lcms2 \
                         port:zlib
 
