diff -Naur jasper-version-2.0.22-orig/src/libjasper/jp2/jp2_cod.h jasper-version-2.0.22/src/libjasper/jp2/jp2_cod.h
--- jasper-version-2.0.22-orig/src/libjasper/jp2/jp2_cod.h	2017-09-11 09:36:23.281414800 +0300
+++ jasper-version-2.0.22/src/libjasper/jp2/jp2_cod.h	2017-09-11 13:16:28.075347800 +0300
@@ -288,10 +288,10 @@
 * Box class.
 \******************************************************************************/
 
-jp2_box_t *jp2_box_create(int type);
-void jp2_box_destroy(jp2_box_t *box);
-jp2_box_t *jp2_box_get(jas_stream_t *in);
-int jp2_box_put(jp2_box_t *box, jas_stream_t *out);
+JAS_DLLEXPORT jp2_box_t *jp2_box_create(int type);
+JAS_DLLEXPORT void jp2_box_destroy(jp2_box_t *box);
+JAS_DLLEXPORT jp2_box_t *jp2_box_get(jas_stream_t *in);
+JAS_DLLEXPORT int jp2_box_put(jp2_box_t *box, jas_stream_t *out);
 
 JAS_ATTRIBUTE_CONST
 static inline uint_least8_t JP2_DTYPETOBPC(uint_least8_t dtype)
@@ -309,7 +309,7 @@
 #define ICC_CS_YCBCR	0x59436272
 #define ICC_CS_GRAY	0x47524159
 
-const jp2_cdefchan_t *jp2_cdef_lookup(jp2_cdef_t *cdef, int channo);
+JAS_DLLEXPORT const jp2_cdefchan_t *jp2_cdef_lookup(jp2_cdef_t *cdef, int channo);
 
 
 #endif
