--- a/source/common/winxp.h.orig	2016-01-25 16:16:50.000000000 +1100
+++ b/source/common/winxp.h	2016-06-05 15:55:35.994089600 +1000
@@ -24,7 +24,7 @@
 #ifndef X265_WINXP_H
 #define X265_WINXP_H
 
-#if defined(_WIN32) && (_WIN32_WINNT < 0x0600) // _WIN32_WINNT_VISTA
+#if defined(_WIN32) && !defined(__x86_64__) // 32-bit Windows
 
 #ifdef _MSC_VER
 #include <intrin.h> // _InterlockedCompareExchange64
@@ -58,10 +58,10 @@
 
 } // namespace X265_NS
 
-#else // if defined(_WIN32) && (_WIN32_WINNT < 0x0600)
+#else // if defined(_WIN32) && !defined(__x86_64__)
 
 #define XP_CONDITION_VAR_FREE(x)
 
-#endif // _WIN32_WINNT <= _WIN32_WINNT_WINXP
+#endif // 32-bit Windows
 
 #endif // ifndef X265_WINXP_H
--- a/source/common/winxp.cpp.orig	2016-01-25 16:16:50.000000000 +1100
+++ b/source/common/winxp.cpp	2016-06-05 15:54:16.730113600 +1000
@@ -23,7 +23,7 @@
 
 #include "threading.h"
 
-#if defined(_WIN32) && (_WIN32_WINNT < 0x0600) // _WIN32_WINNT_VISTA
+#if defined(_WIN32) && !defined(__x86_64__) // 32-bit Windows
 
 namespace X265_NS {
 /* Mimic CONDITION_VARIABLE functions only supported on Vista+ */
@@ -127,4 +127,4 @@
 
 namespace { int _avoid_linker_warnings = 0; }
 
-#endif // _WIN32_WINNT <= _WIN32_WINNT_WINXP
+#endif // 32-bit Windows
