--- sources/shiboken/CMakeLists.txt.orig	2022-07-20 02:44:12.831611200 +0200
+++ sources/shiboken/CMakeLists.txt	2022-07-20 02:45:58.234299400 +0200
@@ -19,15 +19,15 @@
 option(USE_PYTHON3 "Use python3 libraries to build shiboken." FALSE)
 
 if (USE_PYTHON3)
-    find_package(Python3Libs)
-    find_package(Python3InterpWithDebug)
+    find_package(PythonLibs 3.6 REQUIRED)
+    find_package(PythonInterpWithDebug 3.6 REQUIRED)
     #use commom vars
-    set(PYTHONLIBS_FOUND ${PYTHON3LIBS_FOUND})
-    set(PYTHON_LIBRARIES ${PYTHON3_LIBRARIES})
-    set(PYTHON_INCLUDE_DIRS ${PYTHON3_INCLUDE_DIRS})
-    set(PYTHON_DEBUG_LIBRARIES ${PYTHON3_DEBUG_LIBRARIES})
-    set(PYTHONINTERP_FOUND ${PYTHON3INTERP_FOUND})
-    set(PYTHON_EXECUTABLE ${PYTHON3_EXECUTABLE})
+    #set(PYTHONLIBS_FOUND ${PYTHON3LIBS_FOUND})
+    #set(PYTHON_LIBRARIES ${PYTHON3_LIBRARIES})
+    #set(PYTHON_INCLUDE_DIRS ${PYTHON3_INCLUDE_DIRS})
+    #set(PYTHON_DEBUG_LIBRARIES ${PYTHON3_DEBUG_LIBRARIES})
+    #set(PYTHONINTERP_FOUND ${PYTHON3INTERP_FOUND})
+    #set(PYTHON_EXECUTABLE ${PYTHON3_EXECUTABLE})
 else()
     find_package(PythonLibs 2.6)
     find_package(PythonInterpWithDebug)
