--- Portfile.orig	2017-01-26 09:43:32.000000000 +0100
+++ Portfile	2017-01-26 09:46:27.000000000 +0100
@@ -33,7 +33,7 @@
 
 depends_lib             port:freeglut \
                         port:giflib \
-                        port:jpeg \
+                        path:lib/libjpeg.dylib:jpeg \
                         port:libpng \
                         port:mesa \
                         port:tiff \
