--- Portfile.orig	2017-02-10 15:17:05.000000000 +0100
+++ Portfile	2017-02-10 15:18:05.000000000 +0100
@@ -31,7 +31,7 @@
                         sha256  c928119229d4f8f35e20113ffb61f281eda267634a8dc2285af4b0ee27cf2b40
 
 depends_lib             port:giflib \
-                        port:jpeg \
+                        path:lib/libjpeg.dylib:jpeg \
                         port:libpng \
                         port:tiff
 
