--- lib/Target/AMDGPU/SIISelLowering.cpp.orig	2017-03-09 12:24:31.000000000 +0100
+++ lib/Target/AMDGPU/SIISelLowering.cpp	2017-03-09 12:25:25.000000000 +0100
@@ -18,6 +18,18 @@
#include <cmath>
#endif

+#ifndef M_PI
+#define M_PI   3.14159265358979323846
+#endif
+
+#ifndef M_PI_2
+#define M_PI_2 1.57079632679489661923
+#endif
+
+#ifndef M_PI_4
+#define M_PI_4   0.78539816339744830962
+#endif
+
#include "AMDGPU.h"
#include "AMDGPUIntrinsicInfo.h"
#include "AMDGPUSubtarget.h"
