--- Portfile.orig	2017-09-04 16:36:39.000000000 +0200
+++ Portfile	2017-09-04 16:37:13.000000000 +0200
@@ -74,6 +74,7 @@
                     --disable-multilib \
                     --enable-lto \
                     --enable-libstdcxx-time \
+                    --enable-install-libiberty \
                     --with-build-config=bootstrap-debug \
                     --with-as=${prefix}/bin/as \
                     --with-ld=${prefix}/bin/ld \
