--- Portfile.orig	2017-09-07 17:38:48.000000000 +0200
+++ Portfile	2017-09-07 17:38:59.000000000 +0200
@@ -45,7 +45,7 @@
 
 depends_build                   port:pkgconfig
 
-depends_lib                     port:jpeg \
+depends_lib                     path:lib/libjpeg.dylib:jpeg \
                                 port:libpng \
                                 port:webp \
                                 port:fontconfig \
