--- jpegicc/jpegicc.c.orig	2013-02-09 09:26:16.000000000 -0800
+++ jpegicc/jpegicc.c	2013-02-09 09:27:43.000000000 -0800
@@ -707,7 +707,7 @@ int TransformImage(char *cDefInpProf, ch
        DWORD wInput, wOutput;
        int OutputColorSpace;
        DWORD dwFlags = 0; 
-       DWORD EmbedLen;
+       unsigned int EmbedLen;
        LPBYTE EmbedBuffer;
 
 	   // Observer adaptation state (only meaningful on absolute colorimetric intent)
