--- gcc/tree-ssa-uninit.c.orig	2015-09-10 02:14:29.000000000 +0900
+++ gcc/tree-ssa-uninit.c	2016-04-17 05:52:25.000000000 +0900
@@ -1118,7 +1118,7 @@
               edge opnd_edge;
               unsigned uninit_opnds2
                   = compute_uninit_opnds_pos (opnd_def_phi);
-              gcc_assert (!MASK_EMPTY (uninit_opnds2));
+              if (!MASK_EMPTY (uninit_opnds2)) {
               opnd_edge = gimple_phi_arg_edge (phi, i);
               if (!is_use_properly_guarded (phi,
                                             opnd_edge->src,
@@ -1126,6 +1126,7 @@
                                             uninit_opnds2,
                                             visited_phis))
                   return false;
+              }
             }
           else
             return false;
