diff -rupN LibRaw-0.15.4/Makefile.am LibRaw-0.15.4-new/Makefile.am
--- LibRaw-0.15.4/Makefile.am	2013-08-23 04:55:48.000000000 +0200
+++ LibRaw-0.15.4-new/Makefile.am	2013-12-28 00:56:56.418284134 +0100
@@ -41,8 +41,8 @@ lib_libraw_r_a_CFLAGS = -pthread -w
 lib_libraw_la_SOURCES = $(lib_libraw_a_SOURCES)
 lib_libraw_r_la_SOURCES = $(lib_libraw_a_SOURCES)
 
-lib_libraw_la_LDFLAGS = -no-undefined -version-info $(LIBRAW_SHLIB_VER)
-lib_libraw_r_la_LDFLAGS = -no-undefined -version-info $(LIBRAW_SHLIB_VER)
+lib_libraw_la_LDFLAGS = -no-undefined -version-info $(LIBRAW_SHLIB_VER) -lwsock32
+lib_libraw_r_la_LDFLAGS = -no-undefined -version-info $(LIBRAW_SHLIB_VER) -lwsock32
 
 
 # Sample binaries
