--- Portfile.orig	2016-08-31 17:32:02.000000000 +0200
+++ Portfile	2016-08-31 17:32:12.000000000 +0200
@@ -29,7 +29,7 @@
 checksums               rmd160  6d09bf8974cc02c6c75b3807a656ae0574ab1a07 \
                         sha256  66d02b229d2ea9474e62c2b6cd6720fde946155cd1d0d2bffdab829790a0fb22
 
-depends_lib             port:jpeg \
+depends_lib             path:lib/libjpeg.dylib:jpeg \
                         port:tiff \
                         port:zlib
 
