--- oiio-Release-1.5.16/src/libOpenImageIO/CMakeLists.txt.orig	2015-07-10 13:11:48.682388900 +0100
+++ oiio-Release-1.5.16/src/libOpenImageIO/CMakeLists.txt	2015-07-10 13:28:23.918980400 +0100
@@ -231,7 +231,7 @@
 # Include OpenCV if using it
 if (OpenCV_FOUND)
     include_directories (${OpenCV_INCLUDE_DIR})
-    target_link_libraries (OpenImageIO opencv_core opencv_highgui)
+    target_link_libraries (OpenImageIO opencv_core opencv_highgui opencv_videoio)
 endif ()
 
 # Include OpenSSL if using it
