--- src/include/OpenImageIO/thread.h.orig	2016-01-07 13:29:19.426417934 +0100
+++ src/include/OpenImageIO/thread.h	2016-01-07 13:30:00.769308537 +0100
@@ -100,6 +100,7 @@
 #endif
 
 #if defined(__GNUC__) && (defined(_GLIBCXX_ATOMIC_BUILTINS) || (__GNUC__ * 100 + __GNUC_MINOR__ >= 401))
+#include <sched.h>
 #  define USE_GCC_ATOMICS
 #  if !defined(__clang__) && (__GNUC__ * 100 + __GNUC_MINOR__ >= 408)
 #    define OIIO_USE_GCC_NEW_ATOMICS
--- src/libtexture/texoptions.cpp.orig	2016-01-07 13:30:50.420378101 +0100
+++ src/libtexture/texoptions.cpp	2016-01-07 13:31:15.579920073 +0100
@@ -29,6 +29,7 @@
 */
 
 #include <string>
+#include <malloc.h>
 
 #include "OpenImageIO/dassert.h"
 #include "OpenImageIO/typedesc.h"
--- src/libOpenImageIO/exif.cpp.orig	2016-01-07 13:40:39.138028810 +0100
+++ src/libOpenImageIO/exif.cpp	2016-01-07 13:40:44.959153906 +0100
@@ -36,6 +36,7 @@
 #include <map>
 #include <set>
 #include <algorithm>
+#include <malloc.h>
 
 #include <boost/foreach.hpp>
 
 
