--- src/libutil/filesystem.cpp.orig	2016-02-08 17:46:57.119704500 +0100
+++ src/libutil/filesystem.cpp	2016-02-08 17:47:49.122634200 +0100
@@ -472,7 +472,7 @@
 FILE*
 Filesystem::fopen (string_view path, string_view mode)
 {
-#if defined(_WIN32) && !defined(__GNUC__)
+#if defined(_WIN32) && !defined(__GNUC__) || defined(__MINGW32__) || defined(__MINGW64__)
     // on Windows fopen does not accept UTF-8 paths, so we convert to wide char
     std::wstring wpath = Strutil::utf8_to_utf16 (path);
     std::wstring wmode = Strutil::utf8_to_utf16 (mode);
