--- Portfile.orig	2017-03-08 11:26:47.000000000 +0100
+++ Portfile	2017-03-08 11:31:35.000000000 +0100
@@ -74,6 +74,7 @@
                     --disable-multilib \
                     --enable-lto \
                     --enable-libstdcxx-time \
+                    --enable-install-libiberty \
                     --with-build-config=bootstrap-debug \
                     --with-as=${prefix}/bin/as \
                     --with-ld=${prefix}/bin/ld \
