--- Portfile.orig	2016-05-09 09:36:14.000000000 +0200
+++ Portfile	2016-05-09 09:36:24.000000000 +0200
@@ -84,6 +84,10 @@
                     --with-ar=${prefix}/bin/ar \
                     --with-bugurl=https://trac.macports.org/newticket
 
+if {${configure.sdkroot} ne ""} { 
+    configure.args-append     --with-sysroot=${configure.sdkroot} 
+} 
+
 configure.env-append \
                     AR_FOR_TARGET=${prefix}/bin/ar \
                     AS_FOR_TARGET=${prefix}/bin/as \
