--- a/qmake/generators/makefile.cpp.orig	2012-10-09 21:39:01 +0400
+++ b/qmake/generators/makefile.cpp	2012-10-09 21:47:20 +0400
@@ -94,20 +94,30 @@
 QString MakefileGenerator::mkdir_p_asstring(const QString &dir, bool escape) const
 {
     QString ret =  "@$(CHK_DIR_EXISTS) ";
+	
+	QString dirLocalOS = dir;
+
+    bool isWinCmdExe = false;
+    if(isWindowsShell() ||
+       (project->values("QMAKE_SH").isEmpty() &&
+        project->values("QMAKE_HOST.os").indexOf("Windows") != -1)) {
+        isWinCmdExe = true;
+        dirLocalOS = Option::fixPathToLocalOS(dir,false,false);
+    }
     if(escape)
-        ret += escapeFilePath(dir);
+        ret += escapeFilePath(dirLocalOS);
     else
-        ret += dir;
+        ret += dirLocalOS;
     ret += " ";
-    if(isWindowsShell())
+    if(isWinCmdExe)
         ret += "$(MKDIR)";
     else
         ret += "|| $(MKDIR)";
     ret += " ";
     if(escape)
-        ret += escapeFilePath(dir);
+        ret += escapeFilePath(dirLocalOS);
     else
-        ret += dir;
+        ret += dirLocalOS;
     ret += " ";
     return ret;
 }
