--- qt-everywhere-opensource-src-4.8.6/configure.orig	2015-03-21 18:07:24.056403965 +0100
+++ qt-everywhere-opensource-src-4.8.6/configure	2015-03-21 18:09:40.642912613 +0100
@@ -5164,6 +5164,10 @@
 if [ -z "$PKG_CONFIG" ]; then
     PKG_CONFIG=`"$WHICH" pkg-config 2>/dev/null`
 fi
+if [ -n "$PKG_CONFIG" && "$CFG_SHARED" = "no" ] ; then
+    PKG_CONFIG="$PKG_CONFIG --static"
+    [ "$OPT_VERBOSE" = "yes" ] && echo "Using '$PKG_CONFIG' because we're doing a static build"
+fi
 
 # Work out if we can use pkg-config
 if [ "$QT_CROSS_COMPILE" = "yes" ]; then
