diff -up qt-everywhere-opensource-src-4.8.1/tools/linguist/lrelease/main.cpp.qmake-qt4 qt-everywhere-opensource-src-4.8.1/tools/linguist/lrelease/main.cpp
--- qt-everywhere-opensource-src-4.8.1/tools/linguist/lrelease/main.cpp.qmake-qt4	2012-03-14 09:01:12.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.1/tools/linguist/lrelease/main.cpp	2012-05-11 08:58:12.902149676 -0500
@@ -314,7 +314,7 @@
             parseHandler.verbose = evalHandler.verbose = cd.isVerbose();
             ProFileOption option;
 #ifdef QT_BOOTSTRAPPED
-            option.initProperties(binDir + QLatin1String("/qmake"));
+            option.initProperties(binDir + QLatin1String("/qmake-qt4"));
 #else
             option.initProperties(app.applicationDirPath() + QLatin1String("/qmake"));
 #endif
diff -up qt-everywhere-opensource-src-4.8.1/tools/linguist/lupdate/main.cpp.qmake-qt4 qt-everywhere-opensource-src-4.8.1/tools/linguist/lupdate/main.cpp
--- qt-everywhere-opensource-src-4.8.1/tools/linguist/lupdate/main.cpp.qmake-qt4	2012-03-14 09:01:12.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.1/tools/linguist/lupdate/main.cpp	2012-05-11 09:02:47.393718054 -0500
@@ -765,7 +765,7 @@ int main(int argc, char **argv)
 
         parseHandler.verbose = evalHandler.verbose = !!(options & Verbose);
         ProFileOption option;
-        option.initProperties(app.applicationDirPath() + QLatin1String("/qmake"));
+        option.initProperties(app.applicationDirPath() + QLatin1String("/qmake-qt4"));
         option.setCommandLineArguments(QStringList() << QLatin1String("CONFIG+=lupdate_run"));
         ProFileParser parser(0, &parseHandler);
 
