--- a/config.tests/unix/ibase/ibase.pro	2015-05-27 14:49:31.512616600 +0300
+++ b/config.tests/unix/ibase/ibase.pro	2015-05-27 14:49:38.480009800 +0300
@@ -1,4 +1,4 @@
 SOURCES = ibase.cpp
 CONFIG -= qt dylib
 mac:CONFIG -= app_bundle
-LIBS += -lgds
+LIBS += -lfbclient
--- a/config.tests/unix/endian.test	2015-05-27 15:39:33.702970300 +0300
+++ b/config.tests/unix/endian.test	2015-05-27 15:44:28.057531900 +0300
@@ -11,7 +11,7 @@
 
 # build and run a test program
 test -d "$OUTDIR/config.tests/unix/endian" || mkdir -p "$OUTDIR/config.tests/unix/endian"
-"$OUTDIR/bin/qmake" -nocache -spec "$QMKSPEC" "QMAKE_CXXFLAGS*=$SYSROOT" "LIBS*=$SYSROOT" "QT_BUILD_TREE=$OUTDIR" "$SRCDIR/config.tests/unix/endian/endiantest.pro" -o "$OUTDIR/config.tests/unix/endian/Makefile" >/dev/null 2>&1
+"$OUTDIR/bin/qmake" -nocache -spec "$QMKSPEC" "CONFIG+=" "CONFIG-=debug_and_release" "QMAKE_CXXFLAGS*=$SYSROOT" "LIBS*=$SYSROOT" "QT_BUILD_TREE=$OUTDIR" "$SRCDIR/config.tests/unix/endian/endiantest.pro" -o "$OUTDIR/config.tests/unix/endian/Makefile" >/dev/null 2>&1
 cd "$OUTDIR/config.tests/unix/endian"
 
 
--- a/config.tests/unix/makeabs	2015-05-28 10:41:35.515153200 +0300
+++ b/config.tests/unix/makeabs	2015-05-28 10:55:00.222674400 +0300
@@ -4,6 +4,7 @@
 RES="$FILE"
 
 CUT_ARG="-b1"
+CUT_ARG2="-b2"
 if [ `uname -s` = "QNX" ]; then
     # QNX does not understand "-b1"
     CUT_ARG="-c1"
@@ -11,6 +12,8 @@
 
 if [ `echo $FILE | cut $CUT_ARG` = "/" ]; then
     true
+elif [ `echo $FILE | cut $CUT_ARG2` = ":" ]; then
+  true
 else
     RES="$PWD/$FILE"
     test -d "$RES" && RES="$RES/"
@@ -21,5 +24,10 @@
 
     RES=`echo "$RES" | sed "s,//,/,g" | sed "s,/$,,"`
 fi
+
+if [[ -n "$MSYSTEM" ]]; then
+  RES=$(cygpath -wm $RES)
+fi
+
 echo $RES #return
 
