diff --git a/examples/tools/plugandpaint/plugandpaint.pro b/examples/tools/plugandpaint/plugandpaint.pro
index 9616eb8..e188bc1 100644
--- a/examples/tools/plugandpaint/plugandpaint.pro
+++ b/examples/tools/plugandpaint/plugandpaint.pro
@@ -10,7 +10,7 @@ SOURCES        = main.cpp \
 symbian {
     LIBS           = -lpnp_basictools.lib
 } else {
-    LIBS           = -L$${QT_BUILD_TREE}/examples/tools/plugandpaint/plugins -lpnp_basictools
+    LIBS           += -L$${QT_BUILD_TREE}/examples/tools/plugandpaint/plugins -lpnp_basictools
 }
 
 if(!debug_and_release|build_pass):CONFIG(debug, debug|release) {
