diff -up qt-everywhere-opensource-src-4.8.7/src/script/script.pro.gcc8 qt-everywhere-opensource-src-4.8.7/src/script/script.pro
--- qt-everywhere-opensource-src-4.8.7/src/script/script.pro.gcc8	2015-05-07 09:14:43.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.7/src/script/script.pro	2018-05-19 16:01:24.699926959 -0500
@@ -91,6 +91,9 @@ symbian {
     TARGET.UID3=0x2001B2E1
 }
 
+# hack around gcc8 optimization bug with -O2
+QMAKE_CXXFLAGS_RELEASE += -O1
+
 symbian {
     symbian-abld|symbian-sbsv2 {
         MMP_RULES += ALWAYS_BUILD_AS_ARM
