--- a/qmake/Makefile.win32-g++-sh	2015-03-21 18:07:24.056403965 +0100
+++ b/qmake/Makefile.win32-g++-sh	2015-03-21 18:09:40.642912613 +0100
@@ -12,7 +12,7 @@
 # specific stuff for mingw g++ make
 #
 CXX	    =	g++
-CFLAGS	    =	-c -o$@ -O \
+CFLAGS	    =	-c -o$@ -O0 -ggdb \
 		-I. -Igenerators -Igenerators/unix \
 		-Igenerators/win32 -Igenerators/mac \
 		-Igenerators/symbian -Igenerators/integrity \
