--- Portfile.orig	2016-10-19 14:46:44.000000000 +0200
+++ Portfile	2016-10-19 15:00:45.000000000 +0200
@@ -12,7 +12,7 @@
 
 name                qt4-mac
 version             4.8.7
-revision            4
+revision            106
 set branch          [join [lrange [split ${version} .] 0 1] .]
 
 categories          aqua
@@ -41,7 +41,7 @@
                     port:tiff \
                     port:libpng \
                     port:libmng \
-                    port:jpeg
+                    path:lib/libjpeg.dylib:jpeg 
 
 # find a way to specify the OS MINOR version.  For OSX 10.X, this
 # value will be X.  The type is this variable is integer, so we can
@@ -276,6 +276,25 @@
     }
 }
 
+# fix default thread stack size
+patchfiles-append patch-qthread-stacksize.diff
+
+# Add possibility to set a custom global thread pool
+patchfiles-append patch-qt-custom-threadpool.diff
+
+variant gcc49 description {Compile with GCC/G++ 4.9} {
+    configure.compiler macports-gcc-4.9
+    if {[variant_isset universal]} {
+        # use the driverdriver (built using https://github.com/devernay/macportsGCCfixup) to be able to build universal binaries
+        if {[file tail ${configure.cmd}] != "cmake"} {
+            configure.args-append   CC=${configure.cc}-dd CXX=${configure.cxx}-dd
+        } else {
+            configure.args-append   -DCMAKE_C_COMPILER=${configure.cc}-dd -DCMAKE_CXX_COMPILER=${configure.cxx}-dd
+        }
+    }
+}
+
+
 # error out if trying to build on a new OSX version (> 10.12).
 
 platform darwin {
