--- Portfile.orig	2020-12-05 17:58:11.000000000 -0800
+++ Portfile	2020-12-05 18:05:15.000000000 -0800
@@ -519,6 +519,10 @@
         ${worksrcpath}/config.tests/mac
     copy ${filespath}/library_path \
         ${worksrcpath}/config.tests/mac
+
+    # <cstddef> from libc++ includes <version>, and this directory is in the include path.
+    # Since macOS is (most of the time) case insensitive, this causes the build to fail.
+    delete ${worksrcpath}/src/3rdparty/javascriptcore/VERSION
 }
 
 # Block various compilers:
