--- Portfile.orig	2017-09-25 13:40:11.000000000 +0200
+++ Portfile	2017-09-25 13:43:13.000000000 +0200
@@ -11,7 +11,7 @@
 
 name                qt4-mac
 version             4.8.7
-revision            5
+revision            115
 set branch          [join [lrange [split ${version} .] 0 1] .]
 
 categories          aqua
@@ -40,7 +40,7 @@
                     port:tiff \
                     port:libpng \
                     port:libmng \
-                    port:jpeg
+                    path:lib/libjpeg.dylib:jpeg 
 
 # find a way to specify the OS MINOR version.  For OSX 10.X, this
 # value will be X.  The type is this variable is integer, so we can
@@ -217,7 +217,8 @@
 
 platform darwin {
     if {${MINOR} == 4} {
-        patchfiles-append patch-QtHelp_10.4_only.diff
+        # this breaks the framework build
+        #patchfiles-append patch-QtHelp_10.4_only.diff
     }
 }
 
@@ -292,6 +293,32 @@
 
 patchfiles-append patch-fix_pointer_comparison_to_0.diff
 
+# (30) Backport of Qt5 patch to fix an issue with null bytes in QSetting strings (QTBUG-56124).
+patchfiles-append patch-qsettings-null.diff
+
+# (31) Patch to fix build on macOS High Sierra
+patchfiles-append patch-qt4-versions-without-underscores.diff
+
+# (32) Mac OS X 10.4 Tiger patches
+patchfiles-append patch-src_corelib_io_qfilesystemengine_unix.cpp-tiger.diff
+patchfiles-append patch-src_network_kernel_qnetworkproxy_mac.cpp-tiger.diff
+patchfiles-append patch-src_gui_painting_qprintengine_mac.mm-tiger.diff
+patchfiles-append patch-src_gui_dialogs_qfontdialog_mac.mm-tiger.diff
+patchfiles-append patch-src_gui_text_qfontdatabase_mac.cpp-tiger.diff
+patchfiles-append patch-src_gui_painting_qpaintengine_mac.cpp-tiger.diff
+patchfiles-append patch-webkit-tiger.diff
+
+# Qt4 enhancements (do not alter existing functionality):
+
+# Fix default thread stack size on Mac OS X (QTBUG-49607)
+# https://bugreports.qt.io/secure/attachment/52520/patch-qthread-stacksize.diff
+# Note that Qt 5.10 and later implement QThreadPool:stackSize to set the
+# stack size (QTBUG-2568).
+patchfiles-append patch-qthread-stacksize.diff
+
+# Add possibility to set a custom global thread pool
+patchfiles-append patch-qt-custom-threadpool.diff
+
 # error out if trying to build on a new OSX version (> 10.12).
 
 platform darwin {
