--- Portfile.orig	2017-05-16 10:24:01.000000000 +0200
+++ Portfile	2017-05-18 14:09:26.000000000 +0200
@@ -11,7 +11,7 @@
 
 name                qt4-mac
 version             4.8.7
-revision            5
+revision            115
 set branch          [join [lrange [split ${version} .] 0 1] .]
 
 categories          aqua
@@ -40,7 +40,7 @@
                     port:tiff \
                     port:libpng \
                     port:libmng \
-                    port:jpeg
+                    path:lib/libjpeg.dylib:jpeg 
 
 # find a way to specify the OS MINOR version.  For OSX 10.X, this
 # value will be X.  The type is this variable is integer, so we can
@@ -287,6 +287,21 @@
 
 patchfiles-append patch-better-invalid-fonttable-handling.diff
 
+# fix default thread stack size
+patchfiles-append patch-qthread-stacksize.diff
+
+# Add possibility to set a custom global thread pool
+patchfiles-append patch-qt-custom-threadpool.diff
+
+# Backport of Qt5 patch to fix an issue with null bytes in QSetting strings.
+# https://raw.githubusercontent.com/cartr/homebrew-qt4/41669527a2aac6aeb8a5eeb58f440d3f3498910a/patches/qsetting-nulls.patch
+patchfiles-append qsetting-nulls.patch
+
+# fix build failure with clang-4.0
+# https://trac.macports.org/ticket/54183
+# https://svnweb.freebsd.org/ports?view=revision&revision=431846
+patchfiles-append patch-src_3rdparty_webkit_Source_WebCore_html_HTMLImageElement.cpp
+
 # error out if trying to build on a new OSX version (> 10.12).
 
 platform darwin {
