--- Portfile.orig	2016-09-27 18:54:09.000000000 +0200
+++ Portfile	2016-09-27 18:55:24.000000000 +0200
@@ -12,7 +12,7 @@
 
 name                qt4-mac
 version             4.8.7
-revision            4
+revision            106
 set branch          [join [lrange [split ${version} .] 0 1] .]
 
 categories          aqua
@@ -41,7 +41,7 @@
                     port:tiff \
                     port:libpng \
                     port:libmng \
-                    port:jpeg
+                    path:lib/libjpeg.dylib:jpeg 
 
 # find a way to specify the OS MINOR version.  For OSX 10.X, this
 # value will be X.  The type is this variable is integer, so we can
@@ -267,6 +267,13 @@
 patchfiles-append   \
     patch-src_gui_dialogs_qfiledialog_mac.mm.diff
 
+# fix default thread stack size
+patchfiles-append patch-qthread-stacksize.diff
+
+# Add possibility to set a custom global thread pool
+patchfiles-append patch-qt-custom-threadpool.diff
+
+
 # error out if trying to build on a new OSX version (> 10.12).
 
 platform darwin {
