--- IlmImf/Makefile.in.orig	2016-05-22 12:57:47.000000000 -0700
+++ IlmImf/Makefile.in	2016-05-22 13:18:35.000000000 -0700
@@ -581,10 +581,11 @@
 		 ImfOptimizedPixelReading.h
 
 EXTRA_DIST = $(noinst_HEADERS) b44ExpLogTable.cpp b44ExpLogTable.h dwaLookups.cpp dwaLookups.h CMakeLists.txt
-INCLUDES = @ILMBASE_CXXFLAGS@ \
+INCLUDES = \
 	   -I$(top_builddir) \
 	   -I$(top_srcdir)/config \
-	   $(ILMBASE_CFLAGS) $(ZLIB_CFLAGS)
+	   $(ILMBASE_CFLAGS) $(ZLIB_CFLAGS) \
+	   @ILMBASE_CXXFLAGS@
 
 CLEANFILES = b44ExpLogTable b44ExpLogTable.h dwaLookups dwaLookups.h
 b44ExpLogTable_SOURCES = b44ExpLogTable.cpp
