--- Portfile.orig	2018-05-07 08:49:26.000000000 +0200
+++ Portfile	2018-05-07 08:50:59.000000000 +0200
@@ -17,10 +17,6 @@
 master_sites     ${homepage}/data/
 distname         LibRaw-${version}
 
-distfiles-append \
-    LibRaw-demosaic-pack-GPL2-${demosaicversion}${extract.suffix} \
-    LibRaw-demosaic-pack-GPL3-${demosaicversion}${extract.suffix}
-
 checksums \
     LibRaw-${version}${extract.suffix} \
         rmd160  11b0a05a3e82fa7261143ce87944cbe140029cd8 \
@@ -35,7 +31,7 @@
 depends_build    port:pkgconfig
 
 depends_lib      port:jasper \
-                 port:jpeg \
+                 path:lib/libjpeg.dylib:jpeg \
                  port:lcms2
 
 post-destroot {
@@ -44,6 +40,20 @@
     xinstall -m 644 {*}[glob -directory ${worksrcpath}/doc/ *.html] ${htmlDir}
 }
 
+variant gpl2 description {Enable GPL demosaic pack, license will be GPL-2+} {
+    distfiles-append \
+        LibRaw-demosaic-pack-GPL2-${version}${extract.suffix} \
+
+    license                 GPL-2+
+}
+
+variant gpl3 requires gpl2 description {Enable GPL demosaic pack, license will be GPL-3+} {
+    distfiles-append \
+        LibRaw-demosaic-pack-GPL3-${version}${extract.suffix}
+
+    license                 GPL-3+
+}
+
 livecheck.type  regex
 livecheck.url   ${homepage}/download/
 livecheck.regex "LibRaw-(\\d+(?:\\.\\d+)*)${extract.suffix}"
