--- Portfile.orig	2017-09-25 13:40:39.000000000 +0200
+++ Portfile	2017-09-25 13:41:22.000000000 +0200
@@ -16,10 +16,6 @@
 master_sites     ${homepage}/data/
 distname         LibRaw-${version}
 
-distfiles-append \
-    LibRaw-demosaic-pack-GPL2-${version}${extract.suffix} \
-    LibRaw-demosaic-pack-GPL3-${version}${extract.suffix}
-
 checksums \
     LibRaw-${version}${extract.suffix} \
     rmd160  4b1b4d97f459792388b8422bc2fd0b2c5d79017b \
@@ -34,7 +30,7 @@
 depends_build    port:pkgconfig
 
 depends_lib      port:jasper \
-                 port:jpeg \
+                 path:lib/libjpeg.dylib:jpeg \
                  port:lcms2
 
 post-destroot {
@@ -43,6 +39,20 @@
     xinstall -m 644 {*}[glob -directory ${worksrcpath}/doc/ *.html] ${htmlDir}
 }
 
+variant gpl2 description {Enable GPL demosaic pack, license will be GPL-2+} {
+    distfiles-append \
+        LibRaw-demosaic-pack-GPL2-${version}${extract.suffix} \
+
+    license                 GPL-2+
+}
+
+variant gpl3 requires gpl2 description {Enable GPL demosaic pack, license will be GPL-3+} {
+    distfiles-append \
+        LibRaw-demosaic-pack-GPL3-${version}${extract.suffix}
+
+    license                 GPL-3+
+}
+
 livecheck.type  regex
 livecheck.url   ${homepage}/download/
 livecheck.regex "LibRaw-(\\d+(?:\\.\\d+)*)${extract.suffix}"
