--- Portfile.orig	2024-05-07 15:29:13
+++ Portfile	2024-05-07 15:34:19
@@ -32,6 +32,7 @@
 
 # Keep this condition the same in the opencolorio, openimageio, and osl ports.
 set port_latest [expr {${os.platform} ne "darwin" || ${os.major} >= 20}]
+set port_latest 1
 
 if {!${port_latest}} {
     github.setup        AcademySoftwareFoundation OpenColorIO 1.1.1 v
@@ -96,6 +97,9 @@
     port:tinyxml
 
 configure.args-append \
+    -DOCIO_BUILD_OPENFX=OFF \
+    -DOCIO_BUILD_TESTS=OFF \
+    -DOCIO_BUILD_GPU_TESTS=OFF \
     -DUSE_EXTERNAL_YAML=ON \
     -DUSE_EXTERNAL_TINYXML=ON \
     -DUSE_EXTERNAL_LCMS=ON
@@ -110,8 +114,9 @@
 # see https://github.com/imageworks/OpenColorIO/blob/15e96c1f579d3640947a5fcb5ec831383cc3956e/src/pyglue/CMakeLists.txt#L85
 
 variant python38 description {Build the Python 3.8 bindings} conflicts python39 python310 python311 python312 {
-    depends_lib-append port:python38
+    depends_lib-append port:python38 port:py38-pybind11
     configure.args-append \
+        -DOCIO_PYTHON_VERSION=3.8 \
         -DPYTHON=${prefix}/bin/python3.8
     post-destroot {
         xinstall -d -m 0755 \
@@ -123,8 +128,9 @@
 }
 
 variant python39 description {Build the Python 3.9 bindings} conflicts python38 python310 python311 python312 {
-    depends_lib-append port:python39
+    depends_lib-append port:python39 port:py39-pybind11
     configure.args-append \
+        -DOCIO_PYTHON_VERSION=3.9 \
         -DPYTHON=${prefix}/bin/python3.9
     post-destroot {
         xinstall -d -m 0755 \
@@ -136,34 +142,36 @@
 }
 
 variant python310 description {Build the Python 3.10 bindings} conflicts python38 python39 python311 python312 {
-    depends_lib-append port:python310
+    depends_lib-append port:python310 port:py310-pybind11
     configure.args-append \
+        -DOCIO_PYTHON_VERSION=3.10 \
         -DPYTHON=${prefix}/bin/python3.10
     post-destroot {
         xinstall -d -m 0755 \
             ${destroot}${frameworks_dir}/Python.framework/Versions/3.10/lib/python3.10/site-packages
         ln -s \
-            ${prefix}/lib/python3.10/site-packages/PyOpenColorIO.so \
+            ${prefix}/lib/python3.9/site-packages/PyOpenColorIO.so \
             ${destroot}${frameworks_dir}/Python.framework/Versions/3.10/lib/python3.10/site-packages/
     }
 }
 
 
 variant python311 description {Build the Python 3.11 bindings} conflicts python38 python39 python310 python312 {
-    depends_lib-append port:python311
+    depends_lib-append port:python311 port:py311-pybind11
     configure.args-append \
+        -DOCIO_PYTHON_VERSION=3.11 \
         -DPYTHON=${prefix}/bin/python3.11
     post-destroot {
         xinstall -d -m 0755 \
             ${destroot}${frameworks_dir}/Python.framework/Versions/3.11/lib/python3.11/site-packages
         ln -s \
-            ${prefix}/lib/python3.11/site-packages/PyOpenColorIO.so \
+            ${prefix}/lib/python3.9/site-packages/PyOpenColorIO.so \
             ${destroot}${frameworks_dir}/Python.framework/Versions/3.11/lib/python3.11/site-packages/
     }
 }
 
 variant python312 description {Build the Python 3.12 bindings} conflicts python38 python39 python310 python311 {
-    depends_lib-append port:python312
+    depends_lib-append port:python312 port:py312-pybind11
     configure.args-append \
         -DPYTHON=${prefix}/bin/python3.12
     post-destroot {
@@ -181,6 +189,7 @@
 
 if {![variant_isset python38] && ![variant_isset python39] && ![variant_isset python310] && ![variant_isset python311] && ![variant_isset python312]} {
     configure.args-append \
+        -DOCIO_BUILD_PYTHON=OFF \
         -DOCIO_BUILD_PYGLUE=OFF
 }
 
