--- Portfile.orig	2022-02-26 11:10:06.000000000 -0800
+++ Portfile	2022-06-16 14:01:46.000000000 -0700
@@ -7,8 +7,7 @@
 
 boost.depends_type  build
 
-github.setup        AcademySoftwareFoundation OpenColorIO 1.1.1 v
-revision            3
+github.setup        AcademySoftwareFoundation OpenColorIO 2.1.1 v
 name                opencolorio
 categories          graphics
 maintainers         {mcalhoun @MarcusCalhoun-Lopez} openmaintainer
@@ -20,29 +19,25 @@
                     geared towards motion picture production with an emphasis on \
                     visual effects and computer animation.
 
-# please remove "stealth update" workaround on next version update
-dist_subdir         ${name}/${version}_1
-checksums           rmd160  81534822cb8350e1b7ba171c91226de996494d02 \
-                    sha256  b7def3b5383c9b35dc9c0bdd26da74e2199c2b283d52bb84847aa307f70a0bc4 \
-                    size    13830493
+checksums           rmd160  11f40b2157668416c8b5c9ecfc0d8f96a1056119 \
+                    sha256  c3c82c9e6e854db0df9185a7446c63371678c855e54d8737d01d21b4dd1de82d \
+                    size    11018986
 
 # Exclude pre-releases from livecheck
 github.livecheck.regex  {([0-9.]+)}
 
-# see https://github.com/imageworks/OpenColorIO/commit/c43cc918c3e79e324f11ca47e95bfe36e9e0dd15
-patchfiles-append   patch-upstream.diff
+patchfiles-append   missing-CGDisplayCreateUUIDFromDisplayID.diff
+patchfiles-append   pystring-cmake.diff
 
-# see https://gitweb.gentoo.org/repo/gentoo.git/tree/media-libs/opencolorio/files/opencolorio-1.1.0-yaml-cpp-0.6.patch
-patchfiles-append   patch-yaml-cpp.diff
-
-patchfiles-append   patch-yaml-cpp-0.7.diff
-
-patchfiles-append   patch-CMakeLists.txt.diff
+# source assumes at least C++11 compiler
+# OpenImageIO requires C++14
+compiler.cxx_standard 2014
+configure.args-append \
+    -DCMAKE_CXX_STANDARD=14
 
-# source assumes C++11 compiler
-compiler.cxx_standard 2011
-configure.cxxflags-append \
-    -std=c++11
+# Add MacPorts to the set of authorized configurations (see CMakeLists.txt:88)
+configure.args-append \
+    -DCMAKE_CONFIGURATION_TYPES=\"Debug\;Release\;MinSizeRel\;RelWithDebInfo\;MacPorts\"
 
 # see https://github.com/imageworks/OpenColorIO/issues/563
 configure.cxxflags-append \
@@ -63,15 +58,28 @@
 configure.args-append \
     -DUSE_EXTERNAL_YAML=ON \
     -DUSE_EXTERNAL_TINYXML=ON \
-    -DUSE_EXTERNAL_LCMS=ON
+    -DUSE_EXTERNAL_LCMS=ON \
+    -DOCIO_BUILD_OPENFX=OFF \
+    -DOCIO_BUILD_TESTS=OFF \
+    -DOCIO_BUILD_GPU_TESTS=OFF
+
+# Use OpenEXR <= 2.5 Half.
+# When OpenEXR is upgraded to 3.0, depend on openexr instead, and remove the define.
+#depends_lib-append  \
+#    port:ilmbase
+#configure.args-append \
+#    -DOCIO_USE_OPENEXR_HALF=ON
+depends_lib-append  \
+    port:imath
 
 # OpenColorIO intentially installs Python module in lib
 # see https://github.com/imageworks/OpenColorIO/blob/15e96c1f579d3640947a5fcb5ec831383cc3956e/src/pyglue/CMakeLists.txt#L85
 
-variant python27 description {Build the Python 2.7 bindings} conflicts python36 python37 python38 python39 {
-    depends_lib-append port:python27
+variant python27 description {Build the Python 2.7 bindings} conflicts python36 python37 python38 python39 python310 {
+    depends_lib-append port:python27 port:py27-pybind11
     configure.args-append \
-        -DPYTHON=${prefix}/bin/python2.7
+        -DPYTHON=${prefix}/bin/python2.7 \
+        -DOCIO_PYTHON_VERSION=2.7
     post-destroot {
         xinstall -d -m 0755 \
             ${destroot}${frameworks_dir}/Python.framework/Versions/2.7/lib/python2.7/site-packages
@@ -81,10 +89,11 @@
     }
 }
 
-variant python36 description {Build the Python 3.6 bindings} conflicts python27 python37 python38 python39 {
-    depends_lib-append port:python36
+variant python36 description {Build the Python 3.6 bindings} conflicts python27 python37 python38 python39 python310 {
+    depends_lib-append port:python36 port:py36-pybind11
     configure.args-append \
-        -DPYTHON=${prefix}/bin/python3.6
+        -DPYTHON=${prefix}/bin/python3.6\
+        -DOCIO_PYTHON_VERSION=3.6
     post-destroot {
         xinstall -d -m 0755 \
             ${destroot}${frameworks_dir}/Python.framework/Versions/3.6/lib/python3.6/site-packages
@@ -94,10 +103,11 @@
     }
 }
 
-variant python37 description {Build the Python 3.7 bindings} conflicts python27 python36 python38 python39 {
-    depends_lib-append port:python37
+variant python37 description {Build the Python 3.7 bindings} conflicts python27 python36 python38 python39 python310 {
+    depends_lib-append port:python37 port:py37-pybind11
     configure.args-append \
-        -DPYTHON=${prefix}/bin/python3.7
+        -DPYTHON=${prefix}/bin/python3.7\
+        -DOCIO_PYTHON_VERSION=3.7
     post-destroot {
         xinstall -d -m 0755 \
             ${destroot}${frameworks_dir}/Python.framework/Versions/3.7/lib/python3.7/site-packages
@@ -107,10 +117,11 @@
     }
 }
 
-variant python38 description {Build the Python 3.8 bindings} conflicts python27 python36 python37 python39 {
-    depends_lib-append port:python38
+variant python38 description {Build the Python 3.8 bindings} conflicts python27 python36 python37 python39 python310 {
+    depends_lib-append port:python38 port:py38-pybind11
     configure.args-append \
-        -DPYTHON=${prefix}/bin/python3.8
+        -DPYTHON=${prefix}/bin/python3.8\
+        -DOCIO_PYTHON_VERSION=3.8
     post-destroot {
         xinstall -d -m 0755 \
             ${destroot}${frameworks_dir}/Python.framework/Versions/3.8/lib/python3.8/site-packages
@@ -120,10 +131,11 @@
     }
 }
 
-variant python39 description {Build the Python 3.9 bindings} conflicts python27 python36 python37 python38 {
-    depends_lib-append port:python39
+variant python39 description {Build the Python 3.9 bindings} conflicts python27 python36 python37 python38 python310 {
+    depends_lib-append port:python39 port:py39-pybind11
     configure.args-append \
-        -DPYTHON=${prefix}/bin/python3.9
+        -DPYTHON=${prefix}/bin/python3.9\
+        -DOCIO_PYTHON_VERSION=3.9
     post-destroot {
         xinstall -d -m 0755 \
             ${destroot}${frameworks_dir}/Python.framework/Versions/3.9/lib/python3.9/site-packages
@@ -133,13 +145,28 @@
     }
 }
 
-if {![variant_isset python27] && ![variant_isset python36] && ![variant_isset python37] && ![variant_isset python38] && ![variant_isset python39]} {
-    default_variants +python39
+variant python310 description {Build the Python 3.10 bindings} conflicts python27 python36 python37 python38 python39 {
+    depends_lib-append port:python310 port:py310-pybind11
+    configure.args-append \
+        -DPYTHON=${prefix}/bin/python3.10\
+        -DOCIO_PYTHON_VERSION=3.10
+    post-destroot {
+        xinstall -d -m 0755 \
+            ${destroot}${frameworks_dir}/Python.framework/Versions/3.10/lib/python3.10/site-packages
+        ln -s \
+            ${prefix}/lib/python3.10/site-packages/PyOpenColorIO.so \
+            ${destroot}${frameworks_dir}/Python.framework/Versions/3.10/lib/python3.10/site-packages/
+    }
+}
+
+if {![variant_isset python27] && ![variant_isset python36] && ![variant_isset python37] && ![variant_isset python38] && ![variant_isset python39] && ![variant_isset python310]} {
+    default_variants +python310
 }
 
-if {![variant_isset python27] && ![variant_isset python36] && ![variant_isset python37] && ![variant_isset python38] && ![variant_isset python39]} {
+if {![variant_isset python27] && ![variant_isset python36] && ![variant_isset python37] && ![variant_isset python38] && ![variant_isset python39] && ![variant_isset python310]} {
     configure.args-append \
-        -DOCIO_BUILD_PYGLUE=OFF
+        -DOCIO_BUILD_PYGLUE=OFF \
+        -DOCIO_BUILD_PYTHON=OFF
 }
 
 # make neither x11 nor quartz default since openimageio depends on opencolorio
