--- build/make/configure.sh.orig	2020-08-01 00:27:21.000000000 +0200
+++ build/make/configure.sh	2020-08-01 00:29:56.000000000 +0200
@@ -854,7 +854,7 @@
       fi
       ;;
     x86*-darwin*)
-      osx_sdk_dir="$(show_darwin_sdk_path macosx)"
+      osx_sdk_dir="$(configure.sdkroot)"
       if [ -d "${osx_sdk_dir}" ]; then
         add_cflags  "-isysroot ${osx_sdk_dir}"
         add_ldflags "-isysroot ${osx_sdk_dir}"
