--- build/make/configure.sh.orig      2019-09-25 18:44:14.000000000 -0400
+++ build/make/configure.sh   2019-10-23 15:07:31.000000000 -0500
@@ -885,7 +885,7 @@ process_common_toolchain() {
       fi
       ;;
     x86*-darwin*)
-      osx_sdk_dir="$(show_darwin_sdk_path macosx)"
+      osx_sdk_dir="$(configure.sdkroot)"
       if [ -d "${osx_sdk_dir}" ]; then
         add_cflags  "-isysroot ${osx_sdk_dir}"
         add_ldflags "-isysroot ${osx_sdk_dir}"
