--- build/make/Makefile.orig	2018-02-06 19:44:37.000000000 -0800
+++ build/make/Makefile	2018-02-08 07:00:14.000000000 -0800
@@ -200,7 +200,7 @@
 	$(qexec)$(AS) $(ASFLAGS) -o $@ $<
 
 .PRECIOUS: %.c.S
-%.c.S: CFLAGS += -DINLINE_ASM
+%.c.S: CFLAGS += -DINLINE_ASM $(CFLAGS_S)
 $(BUILD_PFX)%.c.S: %.c
 	$(if $(quiet),@echo "    [GEN] $@")
 	$(qexec)$(if $(CONFIG_DEPENDENCY_TRACKING),,mkdir -p $(dir $@))
