--- build/make/Makefile.orig	2016-07-20 18:15:41.000000000 -0700
+++ build/make/Makefile	2016-11-28 09:23:37.000000000 -0800
@@ -192,7 +192,7 @@
 	$(qexec)$(AS) $(ASFLAGS) -o $@ $<
 
 .PRECIOUS: %.c.S
-%.c.S: CFLAGS += -DINLINE_ASM
+%.c.S: CFLAGS += -DINLINE_ASM $(CFLAGS_S)
 $(BUILD_PFX)%.c.S: %.c
 	$(if $(quiet),@echo "    [GEN] $@")
 	$(qexec)$(if $(CONFIG_DEPENDENCY_TRACKING),,mkdir -p $(dir $@))
