--- Portfile.orig	2016-05-17 17:23:03.000000000 +0200
+++ Portfile	2016-06-24 11:18:04.000000000 +0200
@@ -82,12 +82,17 @@
                     --disable-multilib \
                     --enable-lto \
                     --enable-libstdcxx-time \
+                    --enable-install-libiberty \
                     --with-build-config=bootstrap-debug \
                     --with-as=${prefix}/bin/as \
                     --with-ld=${prefix}/bin/ld \
                     --with-ar=${prefix}/bin/ar \
                     --with-bugurl=https://trac.macports.org/newticket
 
+if {${configure.sdkroot} ne ""} { 
+    configure.args-append     --with-sysroot=${configure.sdkroot} 
+} 
+
 configure.env-append \
                     AR_FOR_TARGET=${prefix}/bin/ar \
                     AS_FOR_TARGET=${prefix}/bin/as \
