--- bus/dir-watch-kqueue.c.orig	2014-01-06 09:02:19.000000000 -0700
+++ bus/dir-watch-kqueue.c	2014-04-05 06:07:05.000000000 -0700
@@ -259,7 +259,7 @@
           /* FIXME - less lame error handling for failing to add a watch;
            * we may need to sleep.
            */
-          fd = open (new_dirs[i], O_RDONLY | O_CLOEXEC);
+          fd = open (new_dirs[i], O_RDONLY);
           if (fd < 0)
             {
               if (errno != ENOENT)
@@ -274,6 +274,7 @@
                   continue;
                 }
             }
+          _dbus_fd_set_close_on_exec(fd);
 
           EV_SET (&ev, fd, EVFILT_VNODE, EV_ADD | EV_ENABLE | EV_CLEAR,
                   NOTE_DELETE | NOTE_EXTEND | NOTE_WRITE | NOTE_RENAME, 0, 0);
