--- dbus/dbus-sysdeps-unix.c.orig	2014-01-17 09:59:48.000000000 -0700
+++ dbus/dbus-sysdeps-unix.c	2014-03-21 12:39:25.000000000 -0700
@@ -3385,7 +3385,7 @@
       close (result_pipe[READ_END]);
       close (errors_pipe[READ_END]);
 
-      if (dup2 (fd, 0) == -1) /* setup stdin */
+      if (fd != 0 && dup2 (fd, 0) == -1) /* setup stdin */
         _exit (1);
       if (dup2 (result_pipe[WRITE_END], 1) == -1) /* setup stdout */
         _exit (1);
