--- Portfile.orig	2016-05-09 09:59:08.000000000 +0200
+++ Portfile	2016-05-09 10:02:48.000000000 +0200
@@ -37,7 +37,7 @@
 checksums           rmd160  3d5d6951a36baf32ab0e0958d3b4a9413b7f2e07 \
                     sha256  4d57a50907b510e3049a4bba0d7888930fdfc16ce49f1bf693e5b6247370d68c
 
-depends_lib         port:jpeg \
+depends_lib         path:lib/libjpeg.dylib:jpeg \
                     port:xz \
                     port:zlib
 
