--- x264/Makefile.orig	2013-12-21 21:45:07.000000000 +0000
+++ x264/Makefile	2013-12-22 21:25:49.711777600 +0000
@@ -392,8 +392,9 @@
 
 install-lib-dev:
 	$(INSTALL) -d $(DESTDIR)$(includedir)
-	$(INSTALL) -d $(DESTDIR)$(libdir)/pkgconfig
+	$(INSTALL) -d $(DESTDIR)$(includedir) $(DESTDIR)$(includedir)/avisynth $(DESTDIR)$(libdir)/pkgconfig
 	$(INSTALL) -m 644 $(SRCPATH)/x264.h x264_config.h $(DESTDIR)$(includedir)
+	$(INSTALL) -m 644 $(SRCPATH)/extras/avisynth_c.h $(DESTDIR)$(includedir)/avisynth
 	$(INSTALL) -m 644 x264.pc $(DESTDIR)$(libdir)/pkgconfig
 
 install-lib-static: lib-static install-lib-dev
