--- runtime/src/kmp_wrapper_getpid.h.orig	2019-10-11 17:25:02.000000000 -0700
+++ runtime/src/kmp_wrapper_getpid.h	2019-10-11 17:28:49.000000000 -0700
@@ -22,7 +22,11 @@
 #include <unistd.h>
 #if KMP_OS_DARWIN
 // OS X
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
 #define __kmp_gettid() syscall(SYS_thread_selfid)
+#else
+#define __kmp_gettid() syscall(SYS_gettid)
+#endif
 #elif KMP_OS_FREEBSD
 #include <pthread_np.h>
 #define __kmp_gettid() pthread_getthreadid_np()
