--- configurel.orig
+++ configure
@@ -1087,6 +1087,7 @@
 #-------------------------------------------------------------------------------
 # parse command line arguments
 #-------------------------------------------------------------------------------
+JOBS=
 
 # parse the arguments, setting things to "yes" or "no"
 while [ "$#" -gt 0 ]; do
@@ -2538,6 +2539,9 @@
             UNKNOWN_OPT=yes
         fi
         ;;
+    jobs)
+        JOBS=-j$VAL
+        ;;
     *)
         UNKNOWN_OPT=yes
         ;;
@@ -5137,7 +5141,7 @@
     done
 
     QMAKE_BUILD_ERROR=no
-    (cd "$outpath/qmake"; "$MAKE") || QMAKE_BUILD_ERROR=yes
+    (cd "$outpath/qmake"; "$MAKE" $JOBS) || QMAKE_BUILD_ERROR=yes
     [ '!' -z "$QCONFIG_H" ] && mv -f "$QCONFIG_H" "$QMAKE_QCONFIG_H" #move qmake's qconfig.h to qconfig.h.qmake
     [ '!' -z "$OLD_QCONFIG_H" ] && mv -f "${OLD_QCONFIG_H}.old" "$OLD_QCONFIG_H" #put back qconfig.h
     [ "$QMAKE_BUILD_ERROR" = "yes" ] && exit 2
