--- qmake/generators/unix/unixmake.cpp.orig
+++ qmake/generators/unix/unixmake.cpp
@@ -951,7 +951,7 @@
             else if(project->first("TEMPLATE") == "app" && !project->isEmpty("QMAKE_STRIPFLAGS_APP"))
                 ret += " " + var("QMAKE_STRIPFLAGS_APP");
             if(bundle)
-                ret = " \"" + dst_targ + "/Contents/MacOS/$(QMAKE_TARGET)\"";
+                ret += " \"" + dst_targ + "/Contents/MacOS/$(QMAKE_TARGET)\"";
             else
                 ret += " \"" + dst_targ + "\"";
         }
