--- src/xmlpatterns/api/qcoloroutput_p.h	2015-05-07 16:14:48.000000000 +0200
+++ src/xmlpatterns/api/qcoloroutput_p.h.gcc6	2017-03-08 19:03:24.000000000 +0100
@@ -70,8 +70,8 @@
             ForegroundShift = 10,
             BackgroundShift = 20,
             SpecialShift    = 20,
-            ForegroundMask  = ((1 << ForegroundShift) - 1) << ForegroundShift,
-            BackgroundMask  = ((1 << BackgroundShift) - 1) << BackgroundShift
+            ForegroundMask  = 0x1f << ForegroundShift,
+            BackgroundMask  = 0x7 << BackgroundShift
         };
 
     public:
--- configure	2017-03-09 09:06:41.000000000 +0100
+++ configure.gcc6	2017-03-09 09:04:02.000000000 +0100
@@ -7734,7 +7734,7 @@
     *-g++*)
 	# Check gcc's version
 	case "$(${QMAKE_CONF_COMPILER} -dumpversion)" in
-	    5*|4*|3.4*)
+	    8*|7*|6*|5*|4*|3.4*)
 		;;
             3.3*)
                 canBuildWebKit="no"
@@ -8050,7 +8050,7 @@
     3.*)
         COMPILER_VERSION="3.*"
         ;;
-    5*|4.*)
+    8*|7*|6*|5*|4.*)
         COMPILER_VERSION="4"
         ;;
     *)
