diff -ur qt-everywhere-opensource-src-4.8.7/configure qt-everywhere-opensource-src-4.8.7-hardcode-buildkey/configure
--- qt-everywhere-opensource-src-4.8.7/configure	2015-05-07 16:14:56.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.7-hardcode-buildkey/configure	2020-08-19 23:43:10.792653485 +0200
@@ -7732,18 +7732,6 @@
 	canBuildQtXmlPatterns="no"
 	;;
     *-g++*)
-	# Check gcc's version
-	case "$(${QMAKE_CONF_COMPILER} -dumpversion)" in
-	    5*|4*|3.4*)
-		;;
-            3.3*)
-                canBuildWebKit="no"
-                ;;
-	    *)
-		canBuildWebKit="no"
-		canBuildQtXmlPatterns="no"
-		;;
-	esac
 	;;
     solaris-cc*)
         # Check the compiler version
@@ -8043,19 +8031,7 @@
         ;;
     esac
 
-    case "$COMPILER_VERSION" in
-    2.95.*)
-        COMPILER_VERSION="2.95.*"
-        ;;
-    3.*)
-        COMPILER_VERSION="3.*"
-        ;;
-    5*|4.*)
-        COMPILER_VERSION="4"
-        ;;
-    *)
-        ;;
-    esac
+    COMPILER_VERSION="4"
     [ '!' -z "$COMPILER_VERSION" ] && COMPILER="g++-${COMPILER_VERSION}"
     ;;
 icc*)
