diff -up qt-everywhere-opensource-src-4.8.7/qmake/Makefile.unix.qmake_LFLAGS qt-everywhere-opensource-src-4.8.7/qmake/Makefile.unix
--- qt-everywhere-opensource-src-4.8.7/qmake/Makefile.unix.qmake_LFLAGS	2015-05-07 09:14:42.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.7/qmake/Makefile.unix	2018-02-15 08:25:13.168838577 -0600
@@ -3,7 +3,7 @@ BUILD_PATH = @BUILD_PATH@
 QTOBJS = @QMAKE_QTOBJS@
 QTSRCS = @QMAKE_QTSRCS@
 QMAKESPEC = @QMAKESPEC@
-LFLAGS = @QMAKE_LFLAGS@
+LFLAGS = @QMAKE_LFLAGS@ $(QMAKE_LFLAGS_RELEASE)
 
 #qmake code
 OBJS=project.o property.o main.o makefile.o unixmake2.o unixmake.o  \
