--- src/gui/kernel/qguiplatformplugin.cpp.adwaita	2015-02-12 14:32:45.217935391 +0100
+++ src/gui/kernel/qguiplatformplugin.cpp	2015-02-12 14:46:11.471866038 +0100
@@ -165,8 +165,13 @@ QString QGuiPlatformPlugin::styleName()
     case DE_GNOME: {
         QStringList availableStyles = QStyleFactory::keys();
         // Set QGtkStyle for GNOME if available
+        QString adwaitaStyleKey = QString::fromLatin1("adwaita");
         QString gtkStyleKey = QString::fromLatin1("GTK+");
-        if (availableStyles.contains(gtkStyleKey)) {
+        if (availableStyles.contains(adwaitaStyleKey)) {
+            stylename = adwaitaStyleKey;
+            break;
+        }
+        else if (availableStyles.contains(gtkStyleKey)) {
             stylename = gtkStyleKey;
             break;
         }
