--- src/corelib/global/qglobal.h.orig	2016-04-21 13:41:22.617001856 +0200
+++ src/corelib/global/qglobal.h	2016-04-21 13:42:41.292982598 +0200
@@ -331,6 +331,9 @@
 #    define Q_AUTOTEST_EXPORT
 #endif
 
+#define Q_INIT_RESOURCE_EXTERN(name) \
+    extern int QT_MANGLE_NAMESPACE(qInitResources_ ## name) ();
+
 #define Q_INIT_RESOURCE(name) \
     do { extern int QT_MANGLE_NAMESPACE(qInitResources_ ## name) ();       \
         QT_MANGLE_NAMESPACE(qInitResources_ ## name) (); } while (0)
