diff -up webkit-qtwebkit-23/Source/WebCore/WebCore.pri.debuginfo webkit-qtwebkit-23/Source/WebCore/WebCore.pri
--- webkit-qtwebkit-23/Source/WebCore/WebCore.pri.debuginfo	2013-02-21 22:28:57.791764043 -0600
+++ webkit-qtwebkit-23/Source/WebCore/WebCore.pri	2013-02-21 22:31:02.994198797 -0600
@@ -7,6 +7,12 @@
 
 SOURCE_DIR = $${ROOT_WEBKIT_DIR}/Source/WebCore
 
+#equals(QT_ARCH, s390)|equals(QT_ARCH, arm)|equals(QT_ARCH, mips)|equals(QT_ARCH, i386)|equals(QT_ARCH, i686)|equals(QT_ARCH, x86_64)|equals(QT_ARCH, powerpc64)|equals(QT_ARCH, powerpc) {
+    message("WebCore workaround for QtWebkit: do not build with -g, but with -g1")
+    QMAKE_CXXFLAGS_RELEASE -= -g
+    QMAKE_CXXFLAGS_RELEASE += -g1
+#}
+
 QT *= network
 haveQt(5): QT *= core-private gui-private
 
