diff -up webkit-qtwebkit-23/Source/WebKit/qt/declarative/experimental/experimental.pri.no_rpath webkit-qtwebkit-23/Source/WebKit/qt/declarative/experimental/experimental.pri
--- webkit-qtwebkit-23/Source/WebKit/qt/declarative/experimental/experimental.pri.no_rpath	2014-09-24 06:42:05.000000000 -0500
+++ webkit-qtwebkit-23/Source/WebKit/qt/declarative/experimental/experimental.pri	2014-10-16 07:34:02.266399979 -0500
@@ -29,8 +29,8 @@ haveQt(5): QT += network quick quick-pri
 
 DESTDIR = $${ROOT_BUILD_DIR}/imports/$${TARGET.module_name}
 
-CONFIG += rpath
-RPATHDIR_RELATIVE_TO_DESTDIR = ../../lib
+#CONFIG += rpath
+#RPATHDIR_RELATIVE_TO_DESTDIR = ../../lib
 
 SOURCES += plugin.cpp
 
diff -up webkit-qtwebkit-23/Source/WebKit/qt/declarative/public.pri.no_rpath webkit-qtwebkit-23/Source/WebKit/qt/declarative/public.pri
--- webkit-qtwebkit-23/Source/WebKit/qt/declarative/public.pri.no_rpath	2014-09-24 06:42:05.000000000 -0500
+++ webkit-qtwebkit-23/Source/WebKit/qt/declarative/public.pri	2014-10-16 07:33:46.122390687 -0500
@@ -36,8 +36,8 @@ WEBKIT += wtf
 
 DESTDIR = $${ROOT_BUILD_DIR}/imports/$${TARGET.module_name}
 
-CONFIG += rpath
-RPATHDIR_RELATIVE_TO_DESTDIR = ../../lib
+#CONFIG += rpath
+#RPATHDIR_RELATIVE_TO_DESTDIR = ../../lib
 
 SOURCES += plugin.cpp
 
diff -up webkit-qtwebkit-23/Tools/qmake/mkspecs/features/rpath.prf.no_rpath webkit-qtwebkit-23/Tools/qmake/mkspecs/features/rpath.prf
--- webkit-qtwebkit-23/Tools/qmake/mkspecs/features/rpath.prf.no_rpath	2014-09-24 06:42:05.000000000 -0500
+++ webkit-qtwebkit-23/Tools/qmake/mkspecs/features/rpath.prf	2014-10-13 14:11:19.723829903 -0500
@@ -6,7 +6,7 @@
 
 equals(ROOT_BUILD_DIR, $$dirname(DESTDIR)): RPATHDIR_RELATIVE_TO_DESTDIR = ../lib
 
-linux-*:!isEmpty(RPATHDIR_RELATIVE_TO_DESTDIR) {
+linux-rpath_me_harder_no_no_no:!isEmpty(RPATHDIR_RELATIVE_TO_DESTDIR) {
     # Do the rpath by hand since it's not possible to use ORIGIN in QMAKE_RPATHDIR
     # this expands to $ORIGIN (after qmake and make), it does NOT read a qmake var.
     QMAKE_RPATHDIR = \$\$ORIGIN/$${RPATHDIR_RELATIVE_TO_DESTDIR} $${QMAKE_RPATHDIR}
