diff -up webkit-qtwebkit/Source/WebCore/plugins/PluginDatabase.cpp.orig webkit-qtwebkit/Source/WebCore/plugins/PluginDatabase.cpp
--- webkit-qtwebkit/Source/WebCore/plugins/PluginDatabase.cpp.orig	2011-05-12 10:06:43.000000000 +0200
+++ webkit-qtwebkit/Source/WebCore/plugins/PluginDatabase.cpp	2011-05-12 10:08:08.000000000 +0200
@@ -380,6 +380,7 @@ Vector<String> PluginDatabase::defaultPl
     paths.append("/usr/lib/firefox/plugins");
     paths.append("/usr/lib64/browser-plugins");
     paths.append("/usr/lib/browser-plugins");
+    paths.append("/usr/lib/mozilla/plugins-wrapped");
     paths.append("/usr/lib/mozilla/plugins");
     paths.append("/usr/local/netscape/plugins");
     paths.append("/opt/mozilla/plugins");
@@ -389,6 +390,7 @@ Vector<String> PluginDatabase::defaultPl
     paths.append("/usr/lib/netscape/plugins");
     paths.append("/usr/lib/netscape/plugins-libc5");
     paths.append("/usr/lib/netscape/plugins-libc6");
+    paths.append("/usr/lib64/mozilla/plugins-wrapped");
     paths.append("/usr/lib64/netscape/plugins");
     paths.append("/usr/lib64/mozilla/plugins");
     paths.append("/usr/lib/nsbrowser/plugins");
