--- src/corelib/tools/qelapsedtimer_mac.cpp.orig	2011-09-12 08:49:29.000000000 +0200
+++ src/corelib/tools/qelapsedtimer_mac.cpp	2012-02-02 21:30:46.000000000 +0100
@@ -62,7 +62,7 @@
 {
     if (info.denom == 0)
         mach_timebase_info(&info);
-    qint64 nsecs = cpuTime * info.numer / info.denom;
+    qint64 nsecs = (qint64)(((double) cpuTime) * ((double) info.numer) / ((double) info.denom));
     return nsecs;
 }
 
