--- src/corelib/io/qprocess_unix.cpp.orig	2015-10-14 13:02:18.000000000 -0400
+++ src/corelib/io/qprocess_unix.cpp	2015-10-14 13:02:43.000000000 -0400
@@ -1302,11 +1302,11 @@
 
     // read a byte from the death pipe
     char c;
-    qt_safe_read(deathPipe[0], &c, 1);
+    qint64 readcount = qt_safe_read(deathPipe[0], &c, 1);
 
     // check if our process is dead
     int exitStatus;
-    if (qt_safe_waitpid(pid_t(pid), &exitStatus, WNOHANG) > 0) {
+    if (qt_safe_waitpid(pid_t(pid), &exitStatus, readcount > 0 ? WNOHANG : 0) > 0) {
         processManager()->remove(q);
         crashed = !WIFEXITED(exitStatus);
         exitCode = WEXITSTATUS(exitStatus);
