--- src/corelib/global/qlibraryinfo.cpp.orig
+++ src/corelib/global/qlibraryinfo.cpp
@@ -262,6 +262,11 @@
             path = QT_CONFIGURE_LIBRARIES_PATH;
             break;
 #endif
+#ifdef QT_CONFIGURE_FRAMEWORKS_PATH
+        case FrameworksPath:
+            path = QT_CONFIGURE_FRAMEWORKS_PATH;
+            break;
+#endif
 #ifdef QT_CONFIGURE_BINARIES_PATH
         case BinariesPath:
             path = QT_CONFIGURE_BINARIES_PATH;
@@ -327,6 +332,10 @@
             key = QLatin1String("Libraries");
             defaultValue = QLatin1String("lib");
             break;
+        case FrameworksPath:
+            key = QLatin1String("Frameworks");
+            defaultValue = QLatin1String("Library/Frameworks");
+            break;
         case BinariesPath:
             key = QLatin1String("Binaries");
             defaultValue = QLatin1String("bin");
