--- src/3rdparty/webkit/Source/WebCore/plugins/mac/PluginViewMac.mm.orig
+++ src/3rdparty/webkit/Source/WebCore/plugins/mac/PluginViewMac.mm
@@ -233,8 +233,10 @@
         setNPWindowIfNeeded();
 
     // TODO: Implement null timer throttling depending on plugin activation
+#ifndef NP_NO_CARBON
     m_nullEventTimer = adoptPtr(new Timer<PluginView>(this, &PluginView::nullEventTimerFired));
     m_nullEventTimer->startRepeating(0.02);
+#endif
 
     m_lastMousePos.h = m_lastMousePos.v = 0;
 
