--- src/gui/painting/qprintengine_mac.mm.orig
+++ src/gui/painting/qprintengine_mac.mm
@@ -187,7 +187,11 @@
             paperMargins.top = topMargin;
             paperMargins.right = rightMargin;
             paperMargins.bottom = bottomMargin;
+#if (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_5)
             PMPaperCreateCustom(printer, paperId, QCFString("Custom size"), customSize.width(), customSize.height(), &paperMargins, &customPaper);
+#else
+            PMPaperCreate(printer, paperId, QCFString("Custom size"), customSize.width(), customSize.height(), &paperMargins, &customPaper);
+#endif
             PMPageFormat tmp;
             PMCreatePageFormatWithPMPaper(&tmp, customPaper);
             PMCopyPageFormat(tmp, format);
