--- config.tests/unix/precomp.test.orig
+++ config.tests/unix/precomp.test
@@ -4,7 +4,7 @@
 COMPILER=$1
 VERBOSE=$2
 
-case "$COMPILER" in
+case $COMPILER in
 icpc)
     cat >header.h <<EOF
 #define HEADER_H
@@ -27,8 +27,8 @@
     rm -f header.pchi header.o source.o
     ;;
 
-*g++*|c++|*qcc*)
-    case `"$COMPILER" -dumpversion 2>/dev/null` in
+*g++*|c++|*qcc*|*gcc*|cc)
+    case `$COMPILER -dumpversion 2>/dev/null` in
     3.*)
         ;;
     *)
