--- configure.orig	2019-11-03 08:14:10.000000000 -0700
+++ configure	2019-11-03 08:21:30.000000000 -0700
@@ -8030,6 +8030,13 @@
     QMAKE_CONFIG="$QMAKE_CONFIG exceptions_off"
 fi
 
+if [ "$PLATFORM_MAC" = "yes" ]; then
+    QMakeVar add QMAKE_CFLAGS "-std=c99"
+    QMakeVar add QMAKE_CXXFLAGS "-std=c++98"
+    QMakeVar add QMAKE_OBJECTIVE_CFLAGS "-std=c99"
+    QMakeVar add QMAKE_OBJECTIVE_CXXFLAGS "-std=c++98"
+fi
+
 # On Mac, set the minimum deployment target for the different architechtures 
 # using the Xarch compiler option when supported (10.5 and up).  On 10.4 the
 # deployment version is set to 10.4 globally using the QMAKE_MACOSX_DEPLOYMENT_TARGET
