--- src/network/kernel/qnetworkproxy_mac.cpp.orig
+++ src/network/kernel/qnetworkproxy_mac.cpp
@@ -148,6 +148,7 @@
 }
 
 
+#if (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_5)
 static QNetworkProxy proxyFromDictionary(CFDictionaryRef dict)
 {
     QNetworkProxy::ProxyType proxyType = QNetworkProxy::DefaultProxy;
@@ -180,6 +181,7 @@
 
     return QNetworkProxy(proxyType, hostName, port, user, password);
 }
+#endif
 
 const char * cfurlErrorDescription(SInt32 errorCode)
 {
