--- fontconfig-2.10.92/conf.d/Makefile.am.orig	2013-04-13 06:32:27 +0400
+++ fontconfig-2.10.92/conf.d/Makefile.am	2013-04-13 06:37:08 +0400
@@ -83,9 +83,9 @@
 	@(echo cd $(DESTDIR)$(configdir);			\
 	  cd $(DESTDIR)$(configdir);				\
 	  for i in $(CONF_LINKS); do				\
-	    echo $(RM) $$i";" ln -s $(templatedir)/$$i .;	\
+	    echo $(RM) $$i";" ln -s $(DESTDIR)$(templatedir)/$$i .;	\
 	    $(RM) $$i;						\
-	    ln -s $(templatedir)/$$i .;				\
+	    ln -s $(DESTDIR)$(templatedir)/$$i .;				\
 	  done)
 uninstall-local:
 	@(echo cd $(DESTDIR)$(configdir);			\
