diff --git a/src/video_output_opengl_qt.cpp b/src/video_output_opengl_qt.cpp
index 5488b37..9df62c1 100644
--- a/src/video_output_opengl_qt.cpp
+++ b/src/video_output_opengl_qt.cpp
@@ -99,10 +99,20 @@ void video_output_opengl_qt_widget::initializeGL()
             throw exc(std::string("cannot initialize GLEW: ")
                     + reinterpret_cast<const char *>(glewGetErrorString(err)));
         }
+        const char * gl_renderer = (const char*)glGetString(GL_RENDERER);
+        /* disable GLSL anyway on GPUs that are known to switch to software mode... feel free to expand this list. */
+        /* MacOS calls the card GMA 950, but everywhere else the PCI ID is named 945GM */
+        /* MacOS calls the card GMA X3100, otherwise known as GM965/GL960 */ 
+        bool disable_glsl = strstr(gl_renderer, "GMA 950") || strstr(gl_renderer, "945GM") ||
+                strstr(gl_renderer, "X3100") || strstr(gl_renderer, "965GM");
+        if (disable_glsl)
+        {
+            msg::wrn("Disabling shaders (hardware known tu run GLSL in software mode)");
+        }
         _vo->initialize(
                 glewIsSupported("GL_ARB_pixel_buffer_object"),
                 glewIsSupported("GL_ARB_texture_non_power_of_two"),
-                glewIsSupported("GL_ARB_fragment_shader"));
+                !disable_glsl && glewIsSupported("GL_ARB_fragment_shader"));
     }
     catch (std::exception &e)
     {
