diff --git a/src/video_output_qt.cpp b/src/video_output_qt.cpp
index 98c78e0..728072d 100644
--- a/src/video_output_qt.cpp
+++ b/src/video_output_qt.cpp
@@ -877,6 +877,7 @@ void video_output_qt::center()
 void video_output_qt::enter_fullscreen()
 {
     if (!_fullscreen) {
+        _widget->stop_rendering();
         // If the container is a window, we save its geometry here so that
         // we can restore it later.
         if (!_container_is_external)
@@ -940,12 +941,14 @@ void video_output_qt::enter_fullscreen()
             _screensaver_inhibited = true;
         }
         _fullscreen = true;
+        _widget->start_rendering();
     }
 }
 
 void video_output_qt::exit_fullscreen()
 {
     if (_fullscreen) {
+        _widget->stop_rendering();
         // Resume the screensaver before disabling fullscreen, so that our window ID
         // still represents the fullscreen window and was the same when suspending the screensaver.
         if (_screensaver_inhibited) {
@@ -953,6 +956,7 @@ void video_output_qt::exit_fullscreen()
             _screensaver_inhibited = false;
         }
         // Re-embed the container widget into the main window if necessary
+        // TODO: This breaks the the OpenGL context (or at least the FBOs) on macosx
         if (_container_is_external)
             _container_widget->setWindowFlags(Qt::Widget);
         _container_widget->setWindowFlags(_container_widget->windowFlags()
@@ -967,6 +971,7 @@ void video_output_qt::exit_fullscreen()
         _container_widget->raise();
         _container_widget->grab_focus();
         _fullscreen = false;
+        _widget->start_rendering();
     }
 }
 
