diff --git a/src/gui.cpp b/src/gui.cpp
index feedbb9..b33c133 100644
--- a/src/gui.cpp
+++ b/src/gui.cpp
@@ -723,15 +723,15 @@ controls_widget::controls_widget(QSettings *settings, QWidget *parent)
     QGridLayout *row1_layout = new QGridLayout;
     _play_button = new QPushButton(get_icon("media-playback-start"), "");
     _play_button->setToolTip(_("<p>Play.</p>"));
-    connect(_play_button, SIGNAL(pressed()), this, SLOT(play_pressed()));
+    connect(_play_button, SIGNAL(clicked()), this, SLOT(play_pressed()));
     row1_layout->addWidget(_play_button, 1, 0);
     _pause_button = new QPushButton(get_icon("media-playback-pause"), "");
     _pause_button->setToolTip(_("<p>Pause.</p>"));
-    connect(_pause_button, SIGNAL(pressed()), this, SLOT(pause_pressed()));
+    connect(_pause_button, SIGNAL(clicked()), this, SLOT(pause_pressed()));
     row1_layout->addWidget(_pause_button, 1, 1);
     _stop_button = new QPushButton(get_icon("media-playback-stop"), "");
     _stop_button->setToolTip(_("<p>Stop.</p>"));
-    connect(_stop_button, SIGNAL(pressed()), this, SLOT(stop_pressed()));
+    connect(_stop_button, SIGNAL(clicked()), this, SLOT(stop_pressed()));
     row1_layout->addWidget(_stop_button, 1, 2);
     row1_layout->addWidget(new QWidget, 1, 3);
     _loop_button = new QPushButton(get_icon("media-playlist-repeat"), "");
@@ -745,46 +745,46 @@ controls_widget::controls_widget(QSettings *settings, QWidget *parent)
     _fullscreen_button->setToolTip(_("<p>Switch to fullscreen mode. "
                 "You can leave fullscreen mode by pressing the f key.</p>"));
     _fullscreen_button->setCheckable(true);
-    connect(_fullscreen_button, SIGNAL(pressed()), this, SLOT(fullscreen_pressed()));
+    connect(_fullscreen_button, SIGNAL(clicked()), this, SLOT(fullscreen_pressed()));
     row1_layout->addWidget(_fullscreen_button, 1, 6);
     _center_button = new QPushButton(get_icon("view-restore"), "");
     _center_button->setToolTip(_("<p>Center the video area on your screen.</p>"));
-    connect(_center_button, SIGNAL(pressed()), this, SLOT(center_pressed()));
+    connect(_center_button, SIGNAL(clicked()), this, SLOT(center_pressed()));
     row1_layout->addWidget(_center_button, 1, 7);
     row1_layout->addWidget(new QWidget, 1, 8);
     _bbb_button = new QPushButton(get_icon("media-seek-backward"), "");
     _bbb_button->setFixedSize(_bbb_button->minimumSizeHint());
     _bbb_button->setIconSize(_bbb_button->iconSize() * 12 / 10);
     _bbb_button->setToolTip(_("<p>Seek backward 10 minutes.</p>"));
-    connect(_bbb_button, SIGNAL(pressed()), this, SLOT(bbb_pressed()));
+    connect(_bbb_button, SIGNAL(clicked()), this, SLOT(bbb_pressed()));
     row1_layout->addWidget(_bbb_button, 1, 9);
     _bb_button = new QPushButton(get_icon("media-seek-backward"), "");
     _bb_button->setFixedSize(_bb_button->minimumSizeHint());
     _bb_button->setToolTip(_("<p>Seek backward 1 minute.</p>"));
-    connect(_bb_button, SIGNAL(pressed()), this, SLOT(bb_pressed()));
+    connect(_bb_button, SIGNAL(clicked()), this, SLOT(bb_pressed()));
     row1_layout->addWidget(_bb_button, 1, 10);
     _b_button = new QPushButton(get_icon("media-seek-backward"), "");
     _b_button->setFixedSize(_b_button->minimumSizeHint());
     _b_button->setIconSize(_b_button->iconSize() * 8 / 10);
     _b_button->setToolTip(_("<p>Seek backward 10 seconds.</p>"));
-    connect(_b_button, SIGNAL(pressed()), this, SLOT(b_pressed()));
+    connect(_b_button, SIGNAL(clicked()), this, SLOT(b_pressed()));
     row1_layout->addWidget(_b_button, 1, 11);
     _f_button = new QPushButton(get_icon("media-seek-forward"), "");
     _f_button->setFixedSize(_f_button->minimumSizeHint());
     _f_button->setIconSize(_f_button->iconSize() * 8 / 10);
     _f_button->setToolTip(_("<p>Seek forward 10 seconds.</p>"));
-    connect(_f_button, SIGNAL(pressed()), this, SLOT(f_pressed()));
+    connect(_f_button, SIGNAL(clicked()), this, SLOT(f_pressed()));
     row1_layout->addWidget(_f_button, 1, 12);
     _ff_button = new QPushButton(get_icon("media-seek-forward"), "");
     _ff_button->setFixedSize(_ff_button->minimumSizeHint());
     _ff_button->setToolTip(_("<p>Seek forward 1 minute.</p>"));
-    connect(_ff_button, SIGNAL(pressed()), this, SLOT(ff_pressed()));
+    connect(_ff_button, SIGNAL(clicked()), this, SLOT(ff_pressed()));
     row1_layout->addWidget(_ff_button, 1, 13);
     _fff_button = new QPushButton(get_icon("media-seek-forward"), "");
     _fff_button->setFixedSize(_fff_button->minimumSizeHint());
     _fff_button->setIconSize(_fff_button->iconSize() * 12 / 10);
     _fff_button->setToolTip(_("<p>Seek forward 10 minutes.</p>"));
-    connect(_fff_button, SIGNAL(pressed()), this, SLOT(fff_pressed()));
+    connect(_fff_button, SIGNAL(clicked()), this, SLOT(fff_pressed()));
     row1_layout->addWidget(_fff_button, 1, 14);
     row1_layout->setRowStretch(0, 0);
     row1_layout->setColumnStretch(3, 1);
@@ -1094,7 +1094,7 @@ fullscreen_dialog::fullscreen_dialog(QWidget* parent) : QDialog(parent)
     _inhibit_screensaver_box->setToolTip(_("<p>Inhibit the screensaver during fullscreen playback.</p>"));
 
     QPushButton* ok_btn = new QPushButton(_("OK"));
-    connect(ok_btn, SIGNAL(pressed()), this, SLOT(close()));
+    connect(ok_btn, SIGNAL(clicked()), this, SLOT(close()));
 
     QGridLayout *layout0 = new QGridLayout();
     layout0->addWidget(lbl, 0, 0, 1, 3);
@@ -1256,7 +1256,7 @@ color_dialog::color_dialog(QWidget *parent) : QDialog(parent), _lock(false)
     connect(_s_spinbox, SIGNAL(valueChanged(double)), this, SLOT(s_spinbox_changed(double)));
 
     QPushButton *ok_button = new QPushButton(_("OK"));
-    connect(ok_button, SIGNAL(pressed()), this, SLOT(close()));
+    connect(ok_button, SIGNAL(clicked()), this, SLOT(close()));
 
     QGridLayout *layout = new QGridLayout;
     layout->addWidget(c_label, 0, 0);
@@ -1391,7 +1391,7 @@ crosstalk_dialog::crosstalk_dialog(QWidget *parent) : QDialog(parent), _lock(fal
     connect(_b_spinbox, SIGNAL(valueChanged(double)), this, SLOT(spinbox_changed()));
 
     QPushButton *ok_button = new QPushButton(_("OK"));
-    connect(ok_button, SIGNAL(pressed()), this, SLOT(close()));
+    connect(ok_button, SIGNAL(clicked()), this, SLOT(close()));
 
     QGridLayout *layout = new QGridLayout;
     layout->addWidget(rtfm_label, 0, 0, 1, 2);
@@ -1461,7 +1461,7 @@ zoom_dialog::zoom_dialog(QWidget *parent) : QDialog(parent), _lock(false)
     connect(_z_spinbox, SIGNAL(valueChanged(double)), this, SLOT(z_spinbox_changed(double)));
 
     QPushButton *ok_button = new QPushButton(_("OK"));
-    connect(ok_button, SIGNAL(pressed()), this, SLOT(close()));
+    connect(ok_button, SIGNAL(clicked()), this, SLOT(close()));
 
     QGridLayout *layout = new QGridLayout;
     layout->addWidget(info_label, 0, 0, 1, 3);
@@ -1539,7 +1539,7 @@ audio_dialog::audio_dialog(QWidget *parent) : QDialog(parent), _lock(false)
     connect(_delay_spinbox, SIGNAL(valueChanged(int)), this, SLOT(delay_changed()));
 
     QPushButton *ok_button = new QPushButton(_("OK"));
-    connect(ok_button, SIGNAL(pressed()), this, SLOT(close()));
+    connect(ok_button, SIGNAL(clicked()), this, SLOT(close()));
 
     QGridLayout *layout = new QGridLayout;
     layout->addWidget(device_label, 0, 0);
@@ -1647,10 +1647,10 @@ subtitle_dialog::subtitle_dialog(QWidget *parent) : QDialog(parent), _lock(false
     _color_button->setToolTip(_color_checkbox->toolTip());
     set_color_button(dispatch::parameters().subtitle_color() > std::numeric_limits<uint32_t>::max()
             ? 0xffffffu : dispatch::parameters().subtitle_color());
-    connect(_color_button, SIGNAL(pressed()), this, SLOT(color_button_pressed()));
+    connect(_color_button, SIGNAL(clicked()), this, SLOT(color_button_pressed()));
 
     QPushButton *ok_button = new QPushButton(_("OK"));
-    connect(ok_button, SIGNAL(pressed()), this, SLOT(close()));
+    connect(ok_button, SIGNAL(clicked()), this, SLOT(close()));
 
     QGridLayout *layout = new QGridLayout;
     layout->addWidget(info_label, 0, 0, 1, 2);
@@ -1926,7 +1926,7 @@ video_dialog::video_dialog(QWidget *parent) : QDialog(parent), _lock(false)
     connect(_g_spinbox, SIGNAL(valueChanged(double)), this, SLOT(g_spinbox_changed(double)));
 
     QPushButton *ok_button = new QPushButton(_("OK"));
-    connect(ok_button, SIGNAL(pressed()), this, SLOT(close()));
+    connect(ok_button, SIGNAL(clicked()), this, SLOT(close()));
 
     QGridLayout *layout = new QGridLayout;
     layout->addWidget(crop_ar_label, 0, 0);
@@ -2104,9 +2104,9 @@ open_device_dialog::open_device_dialog(
     _frame_rate_den_spinbox->setRange(_frame_rate_groupbox->isChecked() ? 1 : 0, 65535);
     _frame_rate_den_spinbox->setValue(dev_request.frame_rate_den);
     QPushButton *cancel_btn = new QPushButton(_("Cancel"));
-    connect(cancel_btn, SIGNAL(pressed()), this, SLOT(reject()));
+    connect(cancel_btn, SIGNAL(clicked()), this, SLOT(reject()));
     QPushButton *ok_btn = new QPushButton(_("OK"));
-    connect(ok_btn, SIGNAL(pressed()), this, SLOT(accept()));
+    connect(ok_btn, SIGNAL(clicked()), this, SLOT(accept()));
 
     QGridLayout *frame_size_layout = new QGridLayout;
     frame_size_layout->addWidget(_frame_width_spinbox, 0, 0);
@@ -2771,7 +2771,7 @@ void main_window::file_open()
                     "or if left and right view are stored in separate files.</p>"));
         QCheckBox *dlg_checkbox = new QCheckBox(_("Do not show this message again."));
         QPushButton *dlg_ok_button = new QPushButton(_("OK"));
-        connect(dlg_ok_button, SIGNAL(pressed()), dlg, SLOT(accept()));
+        connect(dlg_ok_button, SIGNAL(clicked()), dlg, SLOT(accept()));
         QGridLayout *dlg_layout = new QGridLayout;
         dlg_layout->addWidget(dlg_icon_label, 0, 0);
         dlg_layout->addWidget(dlg_tip_label, 0, 1);
@@ -2819,8 +2819,8 @@ void main_window::file_open_urls()
     QPushButton *cancel_btn = new QPushButton(_("Cancel"));
     QPushButton *ok_btn = new QPushButton(_("OK"));
     ok_btn->setDefault(true);
-    connect(cancel_btn, SIGNAL(pressed()), url_dialog, SLOT(reject()));
-    connect(ok_btn, SIGNAL(pressed()), url_dialog, SLOT(accept()));
+    connect(cancel_btn, SIGNAL(clicked()), url_dialog, SLOT(reject()));
+    connect(ok_btn, SIGNAL(clicked()), url_dialog, SLOT(accept()));
     QGridLayout *layout = new QGridLayout();
     layout->addWidget(url_label, 0, 0);
     layout->addWidget(url_edit, 0, 1, 1, 3);
@@ -3097,7 +3097,7 @@ void main_window::help_about()
     tab_widget->addTab(license_text, _("License"));
 
     QPushButton *ok_btn = new QPushButton(_("OK"));
-    connect(ok_btn, SIGNAL(pressed()), dialog, SLOT(accept()));
+    connect(ok_btn, SIGNAL(clicked()), dialog, SLOT(accept()));
 
     QGridLayout *layout = new QGridLayout();
     layout->addWidget(logo, 0, 0);
