diff --git a/src/decoder_ffmpeg.cpp b/src/decoder_ffmpeg.cpp
index 6ca5d12..d53af35 100644
--- a/src/decoder_ffmpeg.cpp
+++ b/src/decoder_ffmpeg.cpp
@@ -500,6 +500,10 @@ bool decoder_ffmpeg::read()
     {
         if (packet.stream_index == _stuff->video_streams[i])
         {
+            if (av_dup_packet(&packet) < 0) {
+                msg::dbg(_filename + ": cannot dup packet");
+                return false;
+            }
             _stuff->video_packet_queues[i].push_back(packet);
             msg::dbg(_filename + ": "
                     + str::from(_stuff->video_packet_queues[i].size())
@@ -511,6 +515,10 @@ bool decoder_ffmpeg::read()
     {
         if (packet.stream_index == _stuff->audio_streams[i])
         {
+            if (av_dup_packet(&packet) < 0) {
+                msg::dbg(_filename + ": cannot dup packet");
+                return false;
+            }
             _stuff->audio_packet_queues[i].push_back(packet);
             msg::dbg(_filename + ": "
                     + str::from(_stuff->audio_packet_queues[i].size())
@@ -553,6 +561,7 @@ int64_t decoder_ffmpeg::read_video_frame(int video_stream)
         avcodec_decode_video2(_stuff->video_codec_ctxs[video_stream],
                 _stuff->frames[video_stream], &frame_finished,
                 &(_stuff->packets[video_stream]));
+        av_free_packet(&(_stuff->packets[video_stream])); // moved from release_video_frame
     }
     while (!frame_finished);
 
@@ -573,7 +582,7 @@ int64_t decoder_ffmpeg::read_video_frame(int video_stream)
 
 void decoder_ffmpeg::release_video_frame(int video_stream)
 {
-    av_free_packet(&(_stuff->packets[video_stream]));
+    //av_free_packet(&(_stuff->packets[video_stream])); // moved above
 }
 
 void decoder_ffmpeg::get_video_frame(int video_stream, video_frame_format fmt,
