diff --git a/src/gui.cpp b/src/gui.cpp
index 3be7b63..f8a4f67 100644
--- a/src/gui.cpp
+++ b/src/gui.cpp
@@ -1175,7 +1175,7 @@ fullscreen_dialog::fullscreen_dialog(QWidget* parent) : QDialog(parent)
     _flip_right_box->setChecked(dispatch::parameters().fullscreen_flip_right());
     _flop_right_box->setChecked(dispatch::parameters().fullscreen_flop_right());
     _3d_ready_sync_box->setChecked(dispatch::parameters().fullscreen_3d_ready_sync());
-#ifdef Q_WS_X11
+#if defined(Q_WS_X11) || defined(Q_WS_MAC)
     _inhibit_screensaver_box->setChecked(dispatch::parameters().fullscreen_inhibit_screensaver());
 #else
     _inhibit_screensaver_box->setChecked(false);
diff --git a/src/video_output_qt.cpp b/src/video_output_qt.cpp
index 2516607..51cda90 100644
--- a/src/video_output_qt.cpp
+++ b/src/video_output_qt.cpp
@@ -47,6 +47,10 @@
 # include <X11/Xlib.h>
 # include <GL/glxew.h>
 #endif
+#ifdef Q_WS_MAC
+#include <IOKit/pwr_mgt/IOPMLib.h>
+#include <AvailabilityMacros.h>
+#endif
 
 #include "gettext.h"
 #define _(string) gettext(string)
@@ -819,7 +823,13 @@ void video_output_qt::suspend_screensaver()
 #elif defined(Q_WS_WIN)
     /* TODO */
 #elif defined(Q_WS_MAC)
-    /* TODO */
+# if defined (MAC_OS_X_VERSION_10_6) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_6
+    // API introduced in 10.6
+    IOPMAssertionCreateWithName(kIOPMAssertionTypeNoDisplaySleep, kIOPMAssertionLevelOn, CFSTR ("Bino"), &_disableDisplaySleepAssertion);
+# else
+    // API introduced in OSX 10.5, deprecated in 10.6
+    IOPMAssertionCreate(kIOPMAssertionTypeNoDisplaySleep, kIOPMAssertionLevelOn, &_disableDisplaySleepAssertion);
+# endif
 #endif
 }
 
@@ -834,7 +844,7 @@ void video_output_qt::resume_screensaver()
 #elif defined(Q_WS_WIN)
     /* TODO */
 #elif defined(Q_WS_MAC)
-    /* TODO */
+    IOPMAssertionRelease(_disableDisplaySleepAssertion);
 #endif
 }
 
diff --git a/src/video_output_qt.h b/src/video_output_qt.h
index d20cbe1..3d2f47f 100644
--- a/src/video_output_qt.h
+++ b/src/video_output_qt.h
@@ -179,6 +179,9 @@ private:
     bool _screensaver_inhibited;
     bool _recreate_context;
     bool _recreate_context_stereo;
+#ifdef Q_WS_MAC
+    unsigned int _disableDisplaySleepAssertion;
+#endif
 
     void create_widget();
     void mouse_set_pos(float dest);
