diff --git a/Makefile.am b/Makefile.am
index 1fef733..8e3def7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -50,7 +50,7 @@ package-macosx:
 	$(INSTALL) -m 755 src/bino "$$BINO_APP/Contents/MacOS/Bino" && \
 	$(INSTALL) -m 644 $(top_srcdir)/pkg/macosx/Bino.icns "$$BINO_APP/Contents/Resources" && \
 	$(INSTALL) -m 644 $(top_srcdir)/pkg/macosx/Info.plist "$$BINO_APP/Contents" && \
-	$(MAKE) -C doc install-dist_docDATA docdir="$$BINO_APP/Contents/Resources/Bino Help" && \
+	$(MAKE) -C doc install-dist_htmlDATA htmldir="$$BINO_APP/Contents/Resources/Bino Help" && \
 	$(MAKE) -C po install-strip localedir="$$BINO_APP/Contents/Resources/locale" && \
 	macdeployqt "$$BINO_APP" && \
 	rm -rf Bino.app && \
