--- a/src/main.cpp	2013-01-28 16:11:32.000000000 +0100
+++ b/src/main.cpp	2013-01-28 16:13:01.000000000 +0100
@@ -205,7 +205,10 @@
 #endif
     qInstallMsgHandler(qt_msg_handler);
     QApplication *qt_app = new QApplication(argc, argv, have_display);
-    QTextCodec::setCodecForCStrings(QTextCodec::codecForLocale()); // necessary for i18n via gettext
+    // QApplication constructor resets the locale settings to the system default.
+    // We must now set LC_CTYPE to use UTF-8, so that gettext() returns properly-formatted strings
+    bool setlocale_success = setlocale(LC_CTYPE, "C.UTF-8") || setlocale(LC_CTYPE, "en_US.UTF-8") || setlocale(LC_CTYPE, "en_GB.UTF-8") || setlocale(LC_CTYPE, "UTF-8"); 
+    QTextCodec::setCodecForCStrings(QTextCodec::codecForName("utf8")); // necessary for i18n via gettext
     QCoreApplication::setOrganizationName("Bino");
     QCoreApplication::setOrganizationDomain("bino3d.org");
     QCoreApplication::setApplicationName(PACKAGE_NAME);
