diff --git a/src/player_qt.cpp b/src/player_qt.cpp
index 8111685..379c374 100644
--- a/src/player_qt.cpp
+++ b/src/player_qt.cpp
@@ -1761,9 +1761,12 @@ main_window::main_window(QSettings *settings, const player_init_data &init_data)
     file_menu->addSeparator();
     QAction *file_quit_act = new QAction(_("&Quit..."), this);
     file_quit_act->setShortcut(QKeySequence::Quit);
+    file_quit_act->setMenuRole(QAction::QuitRole);
     connect(file_quit_act, SIGNAL(triggered()), this, SLOT(close()));
     file_menu->addAction(file_quit_act);
     QMenu *preferences_menu = menuBar()->addMenu(_("&Preferences"));
+    // note: whenever the preferences menu becomes a preferences panel, don't forget
+    // to preferences_act->setMenuRole(QAction::PreferencesRole) on the menu item
     QAction *preferences_colors_act = new QAction(_("Display &Color Adjustments..."), this);
     connect(preferences_colors_act, SIGNAL(triggered()), this, SLOT(preferences_colors()));
     preferences_menu->addAction(preferences_colors_act);
@@ -1790,6 +1793,7 @@ main_window::main_window(QSettings *settings, const player_init_data &init_data)
     connect(help_keyboard_act, SIGNAL(triggered()), this, SLOT(help_keyboard()));
     help_menu->addAction(help_keyboard_act);
     QAction *help_about_act = new QAction(_("&About"), this);
+    help_about_act->setMenuRole(QAction::AboutRole);
     connect(help_about_act, SIGNAL(triggered()), this, SLOT(help_about()));
     help_menu->addAction(help_about_act);
 
