diff --git a/src/player_qt.cpp b/src/player_qt.cpp
index bd2a399..f5ed836 100644
--- a/src/player_qt.cpp
+++ b/src/player_qt.cpp
@@ -21,6 +21,7 @@
 #include "config.h"
 
 #include <QCoreApplication>
+#include <QApplication>
 #include <QMainWindow>
 #include <QGridLayout>
 #include <QCloseEvent>
@@ -613,6 +614,9 @@ main_window::main_window(QSettings *settings, const player_init_data &init_data)
     connect(help_about_act, SIGNAL(triggered()), this, SLOT(help_about()));
     help_menu->addAction(help_about_act);
 
+    // Handle FileOpen events
+    QApplication::instance()->installEventFilter(this);
+
     show();     // Must happen before opening initial files!
     raise();
 
@@ -909,6 +913,19 @@ void main_window::help_about()
     QMessageBox::about(this, tr("About " PACKAGE_NAME), blurb);
 }
 
+bool main_window::eventFilter(QObject *obj, QEvent *event)
+{
+    if(event->type() == QEvent::FileOpen)
+    {
+        open(QStringList(static_cast<QFileOpenEvent *>(event)->file()), true);
+        return true;
+    }
+    else 
+    {
+        // pass the event on to the parent class
+        return QMainWindow::eventFilter(obj, event);
+    }
+}
 
 player_qt::player_qt() : player(player::slave)
 {
diff --git a/src/player_qt.h b/src/player_qt.h
index 64ef65b..4396761 100644
--- a/src/player_qt.h
+++ b/src/player_qt.h
@@ -152,6 +152,7 @@ private slots:
 protected:
     void moveEvent(QMoveEvent *event);
     void closeEvent(QCloseEvent *event);
+    bool eventFilter(QObject*, QEvent*);
 
 public:
     main_window(QSettings *settings, const player_init_data &init_data);
