diff --git a/src/video_output_opengl.cpp b/src/video_output_opengl.cpp
index 55d9e46..f36d0f4 100644
--- a/src/video_output_opengl.cpp
+++ b/src/video_output_opengl.cpp
@@ -351,7 +351,17 @@ static void draw_quad(float x, float y, float w, float h)
 
 void video_output_opengl::display(enum video_output::mode mode, float x, float y, float w, float h)
 {
-    glClear(GL_COLOR_BUFFER_BIT);
+    if (mode != stereo)
+    {
+        glClear(GL_COLOR_BUFFER_BIT);
+    }
+    else
+    {
+        glDrawBuffer(GL_BACK_LEFT);
+        glClear(GL_COLOR_BUFFER_BIT);
+        glDrawBuffer(GL_BACK_RIGHT);
+        glClear(GL_COLOR_BUFFER_BIT);
+    }
     if (!_have_valid_data[_active_tex_set])
     {
         return;
