diff --git a/src/lib_versions.cpp b/src/lib_versions.cpp
index 42f01a2..f4078fe 100644
--- a/src/lib_versions.cpp
+++ b/src/lib_versions.cpp
@@ -31,12 +31,12 @@ extern "C"
 
 #include <ass/ass.h>
 
-#ifdef __APPLE__
-#  include <OpenAL/al.h>
-#  include <OpenAL/alc.h>
-#else
+#if !defined(__APPLE__) || defined(HAVE_AL_AL_H)
 #  include <AL/al.h>
 #  include <AL/alc.h>
+#else
+#  include <OpenAL/al.h>
+#  include <OpenAL/alc.h>
 #endif
 
 #include <GL/glew.h>
diff --git a/src/video_output_qt.cpp b/src/video_output_qt.cpp
index 807df2b..b838294 100644
--- a/src/video_output_qt.cpp
+++ b/src/video_output_qt.cpp
@@ -34,6 +34,7 @@ static GLEWContext* glewGetContext() { return &_glewContext; }
 #include <cstring>
 #include <unistd.h>
 
+#include <QCoreApplication>
 #include <QApplication>
 #include <QDesktopWidget>
 #include <QGridLayout>
