diff --git a/src/video_output_qt.cpp b/src/video_output_qt.cpp
index 6f1e5be..cdffe8f 100644
--- a/src/video_output_qt.cpp
+++ b/src/video_output_qt.cpp
@@ -543,6 +543,7 @@ void video_output_qt::init()
         create_widget();
         _widget->makeCurrent();
         set_opengl_versions();
+        glewExperimental = GL_TRUE;
         GLenum err = glewInit();
 #ifdef Q_WS_X11
         if (err == GLEW_OK)
@@ -555,7 +556,19 @@ void video_output_qt::init()
             throw exc(str::asprintf(_("Cannot initialize GLEW: %s"),
                         reinterpret_cast<const char *>(glewGetErrorString(err))));
         }
-        if (!glewIsSupported("GL_VERSION_2_1 GL_EXT_framebuffer_object"))
+        // if (!glewIsSupported("GL_VERSION_2_1 GL_EXT_framebuffer_object"))
+        if (!(GLEW_VERSION_1_3 && // at least OpenGL 1.3 is reasonable
+              //GL_VERSION_1_5: (mostly VBOs)
+              GLEW_ARB_vertex_buffer_object && glBindBuffer && glBufferData && glDeleteBuffers && glGenBuffers && glMapBuffer &&  glUnmapBuffer &&
+              //GL_VERSION_2_0: (mostly GLSL stuff)
+              GLEW_ARB_shader_objects && glAttachShader && glCompileShader && glCreateProgram && glCreateShader && glDeleteProgram && glDeleteShader && glGetAttachedShaders && glGetShaderInfoLog && glGetShaderiv && glGetProgramiv && glGetUniformLocation && glIsProgram && glLinkProgram && glShaderSource && glUseProgram && glUniform1f && glUniform1i &&  glUniform3f &&
+              GLEW_ARB_texture_non_power_of_two && // most textures in Bino are NPOT (part of OpenGL 2.0)
+              //GL_VERSION_2_1:
+              (GLEW_VERSION_2_1 || GLEW_EXT_texture_sRGB) && // sRGB textures is part of OpenGL 2.1
+              // Some drivers (e.g. GeForce 6600 on Mac) only provide the EXT version of the following extension, though an ARB version exists
+              GLEW_EXT_framebuffer_object && glGenFramebuffersEXT
+              //GL_ARB_framebuffer_sRGB is not needed (see video_output.cpp)
+              ))
         {
             throw exc(std::string(_("This OpenGL implementation does not support "
                             "OpenGL 2.1 and framebuffer objects.")));
