diff --git a/configure.ac b/configure.ac
index 9e357c6..e2f59ab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -132,7 +132,12 @@ AC_DEFINE_UNQUOTED([LIBASS_PKGCONFIG_VERSION], [$LIBASS_PKGCONFIG_VERSION], [lib
 
 dnl OpenAL
 PKG_CHECK_MODULES([libopenal], [openal >= 0.0], [HAVE_LIBOPENAL=1], [HAVE_LIBOPENAL=0])
-if test "$HAVE_LIBOPENAL" != "1"; then
+if test "$HAVE_LIBOPENAL" = "1"; then
+    CPPFLAGS_BAK="$CPPFLAGS"
+    CPPFLAGS="$CPPFLAGS $libopenal_CFLAGS"
+    AC_CHECK_HEADERS([AL/al.h])
+    CPPFLAGS="$CPPFLAGS_BAK"
+else
     AC_MSG_WARN([required library libopenal not found:])
     AC_MSG_WARN([$libopenal_PKG_ERRORS])
     AC_MSG_WARN([libopenal is provided by OpenAL; Debian package: libopenal-dev])
diff --git a/src/audio_output.h b/src/audio_output.h
index a3969fb..121f3b8 100644
--- a/src/audio_output.h
+++ b/src/audio_output.h
@@ -25,12 +25,12 @@
 #include <vector>
 #include <string>
 
-#ifdef __APPLE__
-#  include <OpenAL/al.h>
-#  include <OpenAL/alc.h>
-#else
+#if !defined(__APPLE__) || defined(HAVE_AL_AL_H)
 #  include <AL/al.h>
 #  include <AL/alc.h>
+#else
+#  include <OpenAL/al.h>
+#  include <OpenAL/alc.h>
 #endif
 
 #include "media_data.h"
