diff --git a/src/subtitle_renderer.cpp b/src/subtitle_renderer.cpp
index d41c7ac..38bf35e 100644
--- a/src/subtitle_renderer.cpp
+++ b/src/subtitle_renderer.cpp
@@ -138,30 +138,59 @@ void subtitle_renderer::render(uint32_t *bgra32_buffer)
 
 const char *subtitle_renderer::get_fontconfig_conffile()
 {
-#if (defined _WIN32 || defined __WIN32__) && !defined __CYGWIN__
+#if ((defined _WIN32 || defined __WIN32__) && !defined __CYGWIN__) || defined __APPLE__
     /* Fontconfig annoyingly requires a configuration file, but there is no
      * default location for it on Windows. So we just create a temporary one.
      * Note that due to lack of mkstemp() on Windows, this code has race
      * conditions. Note also that if something goes wrong and we return NULL
      * here, the application will likely crash when we try to render a subtitle. */
+    FILE *f;
+#if ((defined _WIN32 || defined __WIN32__) && !defined __CYGWIN__)
     char tmppathname[MAX_PATH];
     static char tmpfilename[MAX_PATH];
-    FILE *f;
     DWORD ret;
     ret = GetTempPath(MAX_PATH, tmppathname);
     if (ret > MAX_PATH || ret == 0)
         return NULL;
     if (GetTempFileName(tmppathname, "fonts_conf", 0, tmpfilename) == 0)
         return NULL;
+#else
+    static char tmpfilename[256];
+    strcpy(tmpfilename, "/tmp/fontsXXXXXX.conf");
+    mkstemp(tmpfilename);
+#endif
     if (!(f = fopen(tmpfilename, "w")))
         return NULL;
     fprintf(f,
             "<?xml version=\"1.0\"?>\n"
             "<!DOCTYPE fontconfig SYSTEM \"fonts.dtd\">\n"
             "<fontconfig>\n"
+#if ((defined _WIN32 || defined __WIN32__) && !defined __CYGWIN__)
             "<dir>WINDOWSFONTDIR</dir>\n"
             "<dir>~/.fonts</dir>\n"
+#else
+            "<dir>/usr/share/fonts</dir>\n"
+            "<dir>/usr/X11/lib/X11/fonts</dir>\n"
+            "<dir>/usr/X11/share/fonts</dir>\n"
+            "<dir>/opt/X11/share/fonts</dir>\n"
+#  ifdef __APPLE__
+            "<dir>/Library/Fonts</dir>\n"
+            "<dir>/Network/Library/Fonts</dir>\n"
+            "<dir>/System/Library/Fonts</dir>\n"
+            "<dir>~/Library/Application Support/Bino/fonts</dir>\n"
+#  endif
+#endif
+
+#if ((defined _WIN32 || defined __WIN32__) && !defined __CYGWIN__)
             "<cachedir>WINDOWSTEMPDIR_FONTCONFIG_CACHE</cachedir>\n"
+#else
+            "<cachedir>/var/cache/fontconfig</cachedir>\n"
+            "<cachedir>/usr/X11/var/cache/fontconfig</cachedir>\n"
+            "<cachedir>/opt/X11/var/cache/fontconfig</cachedir>\n"
+#  ifdef __APPLE__
+            "<cachedir>~/Library/Application Support/Bino/cache/fonts</cachedir>\n"
+#  endif
+#endif
             "<cachedir>~/.fontconfig</cachedir>\n"
             "<config>\n"
             "<blank>\n"
