diff --git a/src/main.cpp b/src/main.cpp
index 9e76033..83148f8 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -241,9 +241,80 @@ int main(int argc, char *argv[])
         init_data.log_level = msg::REQ;
     }
     init_data.filenames = arguments;
+    init_data.video_state.swap_eyes = swap_eyes.value();
     if (input_mode.value() == "")
     {
         init_data.input_mode = input::automatic;
+        if (init_data.filenames.size() == 2)
+        {
+            init_data.input_mode = input::separate;
+            if (init_data.filenames[0].find("-r.") != std::string::npos)
+            {
+                init_data.video_state.swap_eyes = !init_data.video_state.swap_eyes;
+            }
+        }
+        else if (init_data.filenames.size() == 1) {
+            if (init_data.filenames[0].find("-lr.") != std::string::npos)
+            {
+                init_data.input_mode = input::left_right;
+            }
+            else if (init_data.filenames[0].find("-rl.") != std::string::npos)
+            {
+                init_data.input_mode = input::left_right;
+                init_data.video_state.swap_eyes = !init_data.video_state.swap_eyes;
+            }
+            else if (init_data.filenames[0].find("-lrq.") != std::string::npos)
+            {
+                init_data.input_mode = input::left_right_half;
+            }
+            else if (init_data.filenames[0].find("-rtq.") != std::string::npos)
+            {
+                init_data.input_mode = input::left_right_half;
+                init_data.video_state.swap_eyes = !init_data.video_state.swap_eyes;
+            }
+            else if (init_data.filenames[0].find("-ab.") != std::string::npos)
+            {
+                init_data.input_mode = input::top_bottom;
+            }
+            else if (init_data.filenames[0].find("-ba.") != std::string::npos)
+            {
+                init_data.input_mode = input::top_bottom;
+                init_data.video_state.swap_eyes = !init_data.video_state.swap_eyes;
+            }
+            else if (init_data.filenames[0].find("-abq.") != std::string::npos)
+            {
+                init_data.input_mode = input::top_bottom_half;
+            }
+            else if (init_data.filenames[0].find("-baq.") != std::string::npos)
+            {
+                init_data.input_mode = input::top_bottom_half;
+                init_data.video_state.swap_eyes = !init_data.video_state.swap_eyes;
+            }
+            else if (init_data.filenames[0].find("-eo.") != std::string::npos)
+            {
+                init_data.input_mode = input::even_odd_rows;
+                // all image lines are given in this case, and there should be no interpolation [TODO]
+            }
+            else if (init_data.filenames[0].find("-oe.") != std::string::npos)
+            {
+                init_data.input_mode = input::even_odd_rows;
+                init_data.video_state.swap_eyes = !init_data.video_state.swap_eyes;
+                // all image lines are given in this case, and there should be no interpolation [TODO]
+            }
+            else if (init_data.filenames[0].find("-eoq.") != std::string::npos || init_data.filenames[0].find("-3dir.") != std::string::npos)
+            {
+                init_data.input_mode = input::even_odd_rows;
+            }
+            else if (init_data.filenames[0].find("-oeq.") != std::string::npos || init_data.filenames[0].find("-3di.") != std::string::npos)
+            {
+                init_data.input_mode = input::even_odd_rows;
+                init_data.video_state.swap_eyes = !init_data.video_state.swap_eyes;
+            }
+            else if (init_data.filenames[0].find("-2d.") != std::string::npos)
+            {
+                init_data.input_mode = input::mono;
+            }
+        }
     }
     else
     {
@@ -281,7 +352,6 @@ int main(int argc, char *argv[])
         }
     }
     init_data.video_state.fullscreen = fullscreen.value();
-    init_data.video_state.swap_eyes = swap_eyes.value();
     if (center.value())
     {
         init_data.video_flags |= video_output::center;
