diff --git a/src/player_qt.cpp b/src/player_qt.cpp
index 55e5bce..b9fca5f 100644
--- a/src/player_qt.cpp
+++ b/src/player_qt.cpp
@@ -613,6 +613,7 @@ main_window::main_window(QSettings *settings, const player_init_data &init_data)
     help_menu->addAction(help_about_act);
 
     show();     // Must happen before opening initial files!
+    raise();
 
     // Player and timer
     _player = new player_qt_internal(_video_container_widget);
diff --git a/src/video_output_opengl_qt.cpp b/src/video_output_opengl_qt.cpp
index 7591bfe..640cf04 100644
--- a/src/video_output_opengl_qt.cpp
+++ b/src/video_output_opengl_qt.cpp
@@ -371,6 +371,7 @@ void video_output_opengl_qt::open(
 
     _widget->show();
     _container_widget->show();
+    _container_widget->raise();
 }
 
 void video_output_opengl_qt::activate()
