diff --git a/src/decoder_ffmpeg.cpp b/src/decoder_ffmpeg.cpp
index b3d5b35..16d8ee6 100644
--- a/src/decoder_ffmpeg.cpp
+++ b/src/decoder_ffmpeg.cpp
@@ -27,6 +27,14 @@ extern "C"
 #include <libswscale/swscale.h>
 }
 
+// forward compatibility with ffmpeg 0.10
+#if LIBAVCODEC_VERSION_MAJOR > 52
+#define CODEC_TYPE_VIDEO AVMEDIA_TYPE_VIDEO
+#define CODEC_TYPE_AUDIO AVMEDIA_TYPE_AUDIO
+#endif
+#if LIBAVFORMAT_VERSION_MAJOR > 52
+#define PKT_FLAG_KEY AV_PKT_FLAG_KEY
+#endif
 #include <vector>
 #include <deque>
 #include <limits>
