--- bino/configure.ac.orig	2011-05-07 16:18:59.000000000 +0200
+++ bino/configure.ac	2011-05-07 16:23:35.000000000 +0200
@@ -46,7 +46,7 @@
 AC_CHECK_FUNCS([backtrace clock_gettime sigaction vasprintf])
 
 dnl ffmpeg
-PKG_CHECK_MODULES([libavformat], [libavformat >= 52.64.2], [HAVE_LIBAVFORMAT=1], [HAVE_LIBAVFORMAT=0])
+PKG_CHECK_MODULES([libavformat], [libavformat >= 52.64.2 libavcodec libavutil], [HAVE_LIBAVFORMAT=1], [HAVE_LIBAVFORMAT=0])
 if test "$HAVE_LIBAVFORMAT" != "1"; then
     AC_MSG_WARN([library libavformat >= 52.64.2 not found:])
     AC_MSG_WARN([$libavformat_PKG_ERRORS])
