diff --git a/src/main.cpp b/src/main.cpp
index 4b3fb5c..72cd36d 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -112,6 +112,12 @@ int main(int argc, char *argv[])
     options.push_back(&eq_render_client);
 
     std::vector<std::string> arguments;
+    // strip -psn* option when launching from the Finder on mac
+    if (argc > 1 && !strncmp("-psn", argv[1], 4))
+    {
+        --argc;
+        ++argv;
+    }
     if (!opt::parse(argc, argv, options, 0, 3, arguments))
     {
         return 1;
