diff --git a/src/player_qt.cpp b/src/player_qt.cpp
index 719b8ee..cf0c3a6 100644
--- a/src/player_qt.cpp
+++ b/src/player_qt.cpp
@@ -592,9 +592,9 @@ main_window::main_window(QSettings *settings, const player_init_data &init_data)
     file_open_act->setShortcut(QKeySequence::Open);
     connect(file_open_act, SIGNAL(triggered()), this, SLOT(file_open()));
     file_menu->addAction(file_open_act);
-    QAction *file_open_url_act = new QAction(tr("Open &URL..."), this);
-    connect(file_open_url_act, SIGNAL(triggered()), this, SLOT(file_open_url()));
-    file_menu->addAction(file_open_url_act);
+    QAction *file_open_urls_act = new QAction(tr("Open &URL(s)..."), this);
+    connect(file_open_urls_act, SIGNAL(triggered()), this, SLOT(file_open_urls()));
+    file_menu->addAction(file_open_urls_act);
     file_menu->addSeparator();
     QAction *file_quit_act = new QAction(tr("&Quit..."), this);
 #if QT_VERSION >= 0x040600
@@ -838,30 +838,32 @@ void main_window::file_open()
     open(file_names, true);
 }
 
-void main_window::file_open_url()
+void main_window::file_open_urls()
 {
     QDialog *url_dialog = new QDialog(this);
-    url_dialog->setWindowTitle("Open URL");
-    QLabel *url_label = new QLabel("URL:");
+    url_dialog->setWindowTitle("Open URL(s)");
+    QLabel *url_label = new QLabel("URL(s):");
+    url_label->setToolTip("<p>Enter one or more space separated URLs.</p>");
     QLineEdit *url_edit = new QLineEdit("");
+    url_edit->setToolTip(url_label->toolTip());
     url_edit->setMinimumWidth(256);
-    QPushButton *ok_btn = new QPushButton("OK");
     QPushButton *cancel_btn = new QPushButton("Cancel");
-    connect(ok_btn, SIGNAL(pressed()), url_dialog, SLOT(accept()));
-    connect(cancel_btn, SIGNAL(pressed()), url_dialog, SLOT(reject()));
+    QPushButton *ok_btn = new QPushButton("OK");
+    ok_btn->setDefault(true);
+    connect(cancel_btn, SIGNAL(clicked()), url_dialog, SLOT(reject()));
+    connect(ok_btn, SIGNAL(clicked()), url_dialog, SLOT(accept()));
     QGridLayout *layout = new QGridLayout();
     layout->addWidget(url_label, 0, 0);
     layout->addWidget(url_edit, 0, 1, 1, 3);
-    layout->addWidget(ok_btn, 2, 2);
-    layout->addWidget(cancel_btn, 2, 3);
+    layout->addWidget(cancel_btn, 2, 2);
+    layout->addWidget(ok_btn, 2, 3);
     layout->setColumnStretch(1, 1);
     url_dialog->setLayout(layout);
     url_dialog->exec();
     if (url_dialog->result() == QDialog::Accepted
             && !url_edit->text().isEmpty())
     {
-        QString url = url_edit->text();
-        open(QStringList(url), true);
+        open(url_edit->text().split(" ", QString::SkipEmptyParts), true);
     }
 }
 
diff --git a/src/player_qt.h b/src/player_qt.h
index f387452..370ca7c 100644
--- a/src/player_qt.h
+++ b/src/player_qt.h
@@ -144,7 +144,7 @@ private slots:
     void move_event();
     void playloop_step();
     void file_open();
-    void file_open_url();
+    void file_open_urls();
     void help_keyboard();
     void help_about();
 
