diff --git a/src/video_output_qt.cpp b/src/video_output_qt.cpp
index 7a80130..f007947 100644
--- a/src/video_output_qt.cpp
+++ b/src/video_output_qt.cpp
@@ -295,13 +295,18 @@ void video_output_qt::init()
         create_widget();
         _widget->makeCurrent();
         set_opengl_versions();
+        glewExperimental = GL_TRUE;
         GLenum err = glewInit();
         if (err != GLEW_OK)
         {
             throw exc(str::asprintf(_("Cannot initialize GLEW: %s"),
                         reinterpret_cast<const char *>(glewGetErrorString(err))));
         }
-        if (!glewIsSupported("GL_VERSION_2_1 GL_EXT_framebuffer_object"))
+        if (!(GLEW_VERSION_1_3 && GLEW_ARB_texture_non_power_of_two && GLEW_EXT_framebuffer_object &&
+              glGenBuffers && glGenFramebuffers && glDeleteBuffers && glMapBuffer && glBindBuffer && glBufferData && glUnmapBuffer &&
+              glCreateShader && glShaderSource && glCompileShader && glGetShaderiv && glGetShaderInfoLog && glAttachShader && glGetAttachedShaders && glDeleteShader &&
+              glCreateProgram && glGetProgramiv && glLinkProgram && glUseProgram && glDeleteProgram && glIsProgram &&
+              glUniform1i && glUniform1f && glUniform3f && glGetUniformLocation))
         {
             throw exc(std::string(_("This OpenGL implementation does not support "
                             "OpenGL 2.1 and framebuffer objects.")));
