--- rtmpdump/librtmp/rtmp.c.orig	2014-03-12 09:27:27.855517800 +0000
+++ rtmpdump/librtmp/rtmp.c	2014-03-12 09:32:34.378941400 +0000
@@ -2485,7 +2485,7 @@
 #elif defined(USE_GNUTLS)
 typedef struct md5_ctx	MD5_CTX;
 #define MD5_Init(ctx)	md5_init(ctx)
-#define MD5_Update(ctx,data,len)	md5_update(ctx,len,data)
+#define MD5_Update(ctx,data,len)	md5_update(ctx,(unsigned) len,(const uint8_t *) data)
 #define MD5_Final(dig,ctx)	md5_digest(ctx,MD5_DIGEST_LENGTH,dig)
 #else
 #endif
