--- rtmpdump/librtmp/hashswf.c.orig	2014-03-02 18:20:23.000000000 +0000
+++ rtmpdump/librtmp/hashswf.c	2014-03-12 09:37:14.957070300 +0000
@@ -48,7 +48,7 @@
 #endif
 #undef HMAC_CTX
 #define HMAC_CTX	struct hmac_sha256_ctx
-#define HMAC_setup(ctx, key, len)	hmac_sha256_set_key(&ctx, len, key)
+#define HMAC_setup(ctx, key, len)	hmac_sha256_set_key(&ctx, len, (const uint8_t *) key)
 #define HMAC_crunch(ctx, buf, len)	hmac_sha256_update(&ctx, len, buf)
 #define HMAC_finish(ctx, dig, dlen)	dlen = SHA256_DIGEST_LENGTH; hmac_sha256_digest(&ctx, SHA256_DIGEST_LENGTH, dig)
 #define HMAC_close(ctx)
