--- Portfile.orig	2016-05-09 11:42:47.000000000 +0200
+++ Portfile	2016-05-09 11:43:34.000000000 +0200
@@ -32,7 +32,7 @@
                     port:shared-mime-info \
                     port:libpng \
                     port:jasper \
-                    port:jpeg \
+                    path:lib/libjpeg.dylib:jpeg \
                     port:tiff
 
 patchfiles          patch-tests-pixbuf-readonly-to-mutable.c.diff
