--- m4/acinclude.m4.orig	2014-05-25 05:30:58.000000000 -0700
+++ m4/acinclude.m4	2010-07-08 09:10:23.000000000 -0700
@@ -266,7 +266,7 @@
 CPPFLAGS="${GL_CFLAGS} ${CPPFLAGS}"
 ax_save_LIBS="${LIBS}"
 LIBS=""
-ax_check_libs="-lopengl32 -lGL"
+ax_check_libs="-lopengl32"
 for ax_lib in ${ax_check_libs}; do
   AS_IF([test X$ax_compiler_ms = Xyes],
         [ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`],
@@ -402,7 +402,7 @@
 [AX_CHECK_GLU_PROGRAM],
 [ax_cv_check_glu_libglu=yes],
 [LIBS=""
-ax_check_libs="-lglu32 -lGLU"
+ax_check_libs="-lglu32"
 for ax_lib in ${ax_check_libs}; do
   AS_IF([test X$ax_compiler_ms = Xyes],
         [ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`],
@@ -540,7 +540,7 @@
 CPPFLAGS="${GLUT_CFLAGS} ${CPPFLAGS}"
 ax_save_LIBS="${LIBS}"
 LIBS=""
-ax_check_libs="-lglut32 -lglut"
+ax_check_libs="-lglut32"
 for ax_lib in ${ax_check_libs}; do
   AS_IF([test X$ax_compiler_ms = Xyes],
         [ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`],
