--- Portfile.orig	2015-09-20 16:57:44.000000000 +0200
+++ Portfile	2015-09-20 16:57:44.000000000 +0200
@@ -64,6 +64,7 @@
 configure.args      --enable-languages=c,c++,objc,obj-c++,lto,fortran,java \
                     --libdir=${prefix}/lib/${name} \
                     --includedir=${prefix}/include/${name} \
+                    --with-sysroot=${configure.sdkroot} \
                     --infodir=${prefix}/share/info \
                     --mandir=${prefix}/share/man \
                     --datarootdir=${prefix}/share/gcc-${major} \
@@ -81,6 +82,7 @@
                     --disable-multilib \
                     --enable-lto \
                     --enable-libstdcxx-time \
+                    --enable-install-libiberty \
                     --with-as=${prefix}/bin/as \
                     --with-ld=${prefix}/bin/ld \
                     --with-ar=${prefix}/bin/ar \
