--- Portfile.orig	2015-11-24 12:21:32.000000000 +0100
+++ Portfile	2015-11-24 12:22:39.000000000 +0100
@@ -16,7 +16,7 @@
 #conflicts           qt3 qt3-mac qt4-mac-devel
 conflicts           qt3 qt3-mac qt5-mac
 version             4.8.7
-revision            2
+revision            102
 set branch          [join [lrange [split ${version} .] 0 1] .]
 
 categories          aqua
@@ -271,6 +271,9 @@
 patchfiles-append   \
     patch-src_gui_dialogs_qfiledialog_mac.mm.diff
 
+# fix default thread stack size
+patchfiles-append patch-qthread-stacksize.diff
+
 # error out if trying to build on a new OSX version (> 10.11).
 
 platform darwin {
