--- a/src/include/OpenImageIO/typedesc.h.orig	2014-01-12 23:34:55.402417300 +0000
+++ b/src/include/OpenImageIO/typedesc.h	2014-01-12 23:35:17.974708400 +0000
@@ -47,6 +47,13 @@
 struct OIIO_API TypeDesc {
     /// BASETYPE is a simple enum describing the base data types that
     /// correspond (mostly) to the C/C++ built-in types.
+// ansidecl.h contains:
+// #if defined (__STDC__) || defined(__cplusplus) .. || defined(_WIN32)
+// "#define PTR		void *"
+#ifdef PTR
+#define PTR_OLD_DEFINE PTR
+#undef PTR
+#endif
     enum BASETYPE {
         UNKNOWN,            ///< unknown type
         NONE,               ///< void/no type
@@ -79,6 +79,10 @@
         PTR,                ///< A pointer value.
         LASTBASE
     };
+// Set PTR back what it previously contained.
+#ifdef PTR_OLD_DEFINE
+#define PTR PTR_OLD_DEFINE
+#endif
 
     /// AGGREGATE describes whether our TypeDesc is a simple scalar of one
     /// of the BASETYPE's, or one of several simple aggregates.
