--- a/src/include/OpenImageIO/simd.h
+++ b/src/include/OpenImageIO/simd.h
@@ -77,7 +77,7 @@
 
 #if defined(__CUDA_ARCH__)
     // Cuda -- don't include any of these headers
-#elif defined(_WIN32)
+#elif defined(_MSC_VER) || (defined(__MINGW32__) && !defined(__aarch64__))
 #  include <intrin.h>
 #elif defined(__GNUC__) && (defined(__x86_64__) || defined(__i386__)) || defined(__e2k__)
 #  include <x86intrin.h>
