--- oiio-Release-1.8.16/src/libutil/thread.cpp.orig	2018-11-04 12:10:21.836261200 +0300
+++ oiio-Release-1.8.16/src/libutil/thread.cpp	2018-11-04 12:12:41.306238400 +0300
@@ -215,7 +215,7 @@
         bool hasTerminatedThread = std::any_of (this->threads.begin(), this->threads.end(),
                 [](std::unique_ptr<std::thread>& t){
                     DWORD rcode;
-                    GetExitCodeThread(t->native_handle(), &rcode);
+                    GetExitCodeThread((HANDLE)t->native_handle(), &rcode);
                     return rcode != STILL_ACTIVE;
             });
 
