diff --git a/src/cmake/modules/FindLibRaw.cmake b/src/cmake/modules/FindLibRaw.cmake
index c0768ac9..b92ad44a 100644
--- a/src/cmake/modules/FindLibRaw.cmake
+++ b/src/cmake/modules/FindLibRaw.cmake
@@ -26,7 +26,7 @@ ENDIF()
 
 FIND_PATH(LibRaw_INCLUDE_DIR libraw/libraw.h
           HINTS
-          ${LIBRAW_PATH}
+          ${LIBRAW_PATH}/include
           ${PC_LIBRAW_INCLUDEDIR}
           ${PC_LibRaw_INCLUDE_DIRS}
           PATH_SUFFIXES libraw
@@ -34,14 +34,14 @@ FIND_PATH(LibRaw_INCLUDE_DIR libraw/libraw.h
 
 FIND_LIBRARY(LibRaw_LIBRARIES NAMES raw libraw
              HINTS
-             ${LIBRAW_PATH}
+             ${LIBRAW_PATH}/lib
              ${PC_LIBRAW_LIBDIR}
              ${PC_LIBRAW_LIBRARY_DIRS}
             )
 
 FIND_LIBRARY(LibRaw_r_LIBRARIES NAMES raw_r
              HINTS
-             ${LIBRAW_PATH}
+             ${LIBRAW_PATH}/lib
              ${PC_LIBRAW_R_LIBDIR}
              ${PC_LIBRAW_R_LIBRARY_DIRS}
             )
