diff --git a/src/include/OpenImageIO/hash.h b/src/include/OpenImageIO/hash.h
index 5b1c884a..bfd7eeae 100644
--- a/src/include/OpenImageIO/hash.h
+++ b/src/include/OpenImageIO/hash.h
@@ -331,7 +331,7 @@ OIIO_HOSTDEVICE inline constexpr uint64_t Uint128High64(const uint128_t x) {
     return x.second;
 }
 
-OIIO_HOSTDEVICE inline constexpr uint128_t Uint128(uint64_t lo, uint64_t hi) {
+OIIO_HOSTDEVICE inline OIIO_CONSTEXPR14 uint128_t Uint128(uint64_t lo, uint64_t hi) {
     return uint128_t(lo, hi);
 }
 
