--- configure.old	2015-09-27 18:43:30.000000000 +0200
+++ configure	2015-09-27 18:44:18.000000000 +0200
@@ -6480,7 +6480,6 @@
     check_optflags -fno-tree-vectorize
     check_cflags -Werror=format-security
     check_cflags -Werror=implicit-function-declaration
-    check_cflags -Werror=missing-prototypes
     check_cflags -Werror=return-type
     check_cflags -Werror=vla
     check_cflags -Wformat
