--- libswscale/ppc/yuv2rgb_altivec.c.orig	2020-08-22 16:39:38.000000000 -0700
+++ libswscale/ppc/yuv2rgb_altivec.c	2020-08-22 16:46:11.000000000 -0700
@@ -283,6 +283,17 @@
  * ------------------------------------------------------------------------------
  */
 
+
+#if !HAVE_VSX
+static inline vector unsigned char vec_xl(signed long long offset, const ubyte *addr)
+{
+	    const vector unsigned char *v_addr = (const vector unsigned char *) (addr + offset);
+	    vector unsigned char align_perm = vec_lvsl(offset, addr);
+	
+	    return (vector unsigned char) vec_perm(v_addr[0], v_addr[1], align_perm);
+}
+#endif /* !HAVE_VSX */
+
 #define DEFCSP420_CVT(name, out_pixels)                                       \
 static int altivec_ ## name(SwsContext *c, const unsigned char **in,          \
                             int *instrides, int srcSliceY, int srcSliceH,     \
