--- libavcodec/videotoolboxenc.c.old	2018-01-18 17:00:25.000000000 +0100
+++ libavcodec/videotoolboxenc.c	2018-01-24 18:51:02.000000000 +0100
@@ -41,7 +41,7 @@ enum { kCMVideoCodecType_HEVC = 'hvc1' }
 
 typedef OSStatus (*getParameterSetAtIndex)(CMFormatDescriptionRef videoDesc,
                                            size_t parameterSetIndex,
-                                           const uint8_t * _Nullable *parameterSetPointerOut,
+                                           const uint8_t **parameterSetPointerOut,
                                            size_t *parameterSetSizeOut,
                                            size_t *parameterSetCountOut,
                                            int *NALUnitHeaderLengthOut);
