--- Portfile.orig	2020-10-27 16:13:30.000000000 -0700
+++ Portfile	2020-10-28 17:24:50.000000000 -0700
@@ -48,7 +48,7 @@
                             port:djvulibre \
                             port:xz \
                             port:jbigkit \
-                            port:jpeg \
+                            path:lib/libjpeg.dylib:jpeg \
                             port:lcms2 \
                             port:libpng \
                             port:libraw \
@@ -67,6 +67,9 @@
                             port:libxml2 \
                             port:libheif
 
+patchfiles \
+    ImageMagick-6.9.10-gimp210.diff
+
 # Magick-config etc. use pkg-config
 depends_lib-append          port:pkgconfig
 
@@ -83,11 +86,13 @@
                             --with-dps \
                             --with-bzlib \
                             --with-djvu \
+                            --with-fftw \
                             --with-fontconfig \
                             --with-gslib \
                             --with-jbig \
                             --with-jpeg \
                             --with-lcms \
+                            --without-lzma \
                             --with-openjp2 \
                             --with-png \
                             --with-tiff \
@@ -120,6 +125,77 @@
     }
 }
 
+variant natron conflicts x11 requires pango rsvg q32 hdri description {Natron version} {
+    if {${configure.cxx_stdlib} eq "libstdc++"} {
+        # configure with GCC 4.9 for proper OpenMP support
+        configure.compiler macports-gcc-4.9
+    } else {
+        # libc++-based build
+        # configure with clang-4.0 for proper OpenMP support
+        if {${os.major} < 11} {
+           configure.compiler macports-clang-4.0
+        } else {
+           configure.compiler macports-clang-5.0
+        }
+    }
+    configure.args-append --enable-openmp
+    depends_lib-delete      port:ghostscript
+    configure.args-replace  --with-dps --without-dps
+    configure.args-replace  --with-gslib --without-gslib
+    depends_lib-delete       port:djvulibre
+    configure.args-replace  --with-djvu --without-djvu
+    depends_lib-delete       port:fftw-3
+    configure.args-replace  --with-fftw --without-fftw
+    depends_lib-delete       port:jbigkit
+    configure.args-replace  --with-jbig --without-jbig
+    depends_lib-delete       port:jbigkit
+    configure.args-replace  --with-jbig --without-jbig
+    depends_lib-delete       port:jpeg
+    configure.args-replace  --with-jpeg --without-jpeg
+    # PNG is required for povray,openraster,krita plugin, see https://github.com/olear/openfx-arena/commit/48064135604d1c2d07362bc16cecc7a16a640d3e#commitcomment-14356716
+    # depends_lib-delete       port:png
+    # configure.args-replace  --with-png --without-png
+    depends_lib-delete       port:openjpeg
+    configure.args-replace  --with-openjp2 --without-openjp2
+    depends_lib-delete       port:openexr
+    configure.args-replace  --with-openexr --without-openexr
+    depends_lib-delete       port:tiff
+    configure.args-replace  --with-tiff --without-tiff
+    depends_lib-delete       port:webp
+    configure.args-replace  --with-webp --without-webp
+    # zlib is required by PNG
+    #depends_lib-delete       port:zlib
+    #configure.args-replace  --with-zlib --without-zlib
+    depends_lib-delete       port:bzip2
+    configure.args-replace  --with-bzlib --without-bzlib
+    configure.args-replace  --enable-shared --disable-shared
+    configure.args-replace  --with-modules --without-modules
+    #pre-extract {
+    #    if {${version} != "6.9.7-5"} {
+    #        ui_msg "Natron variant requires ImageMagick version 6.9.7-5 (this is ${version}), please modify the Portfile."
+    #        return -code error "incompatible ImageMagick version"
+    #    }
+    #}
+}
+
+# OpenMP support seems to be broken in gcc 4.2.1 and llvm-gcc-4.2 (OpenMP can only be used in the main thread):
+# http://sourceforge.net/tracker/index.php?func=detail&aid=1964341&group_id=170962&atid=856209
+# http://gcc.gnu.org/bugzilla/show_bug.cgi?id=36242
+# clang has proper OpenMP support
+#   -- Frederic.Devernay@inria.fr
+if {${configure.compiler} == "llvm-gcc-4.2" || ${configure.compiler} == "gcc-4.2" || ${configure.compiler} == "gcc-4.0"} {
+    configure.args-append --disable-openmp
+}
+
+post-configure {
+    reinplace -E {s|-arch [a-z0-9_]+||g} \
+        ${worksrcpath}/magick/Magick-config \
+        ${worksrcpath}/magick/MagickCore-config \
+        ${worksrcpath}/Magick++/bin/Magick++-config \
+        ${worksrcpath}/wand/MagickWand-config \
+        ${worksrcpath}/wand/Wand-config
+}
+
 test.run                    yes
 test.target                 check
 test.env                    DYLD_LIBRARY_PATH=${worksrcpath}/magick/.libs
@@ -127,6 +203,23 @@
 # ImageMagick uses .la files at runtime to find its coder modules.
 destroot.delete_la_files    no
 
+post-destroot {
+    if {${configure.cxx_stdlib} eq "libc++"} {
+        # The pkg-config files contain -lgomp instead of -L/opt/local/lib/libomp -omp
+        foreach f [glob -directory ${destroot}${prefix}/lib/pkgconfig Magick*pc Wand*pc ImageMagick*pc] {
+            reinplace -locale C {/Libs.private: /s:-lgomp:-L${prefix}/lib/libomp -lomp:g} ${f}
+        }
+    }
+}
+
+variant q32 description {Use Q32 quantum depth} {
+    configure.args-append   --with-quantum-depth=32
+}
+
+variant hdri description {High Dynamic Range Imagery} {
+    configure.args-replace  --disable-hdri --enable-hdri
+}
+
 variant graphviz description {Support Graphviz} {
     depends_lib-append      path:bin/dot:graphviz
     configure.args-replace  --without-gvc --with-gvc
@@ -161,6 +254,11 @@
 
 default_variants            +x11
 
+variant xquartz {
+    configure.args-append   --with-x --x-includes=/opt/X11/include --x-librari=/opt/X11/lib
+    configure.args-delete   --without-x
+}
+
 livecheck.type              regex
 livecheck.url               [lindex ${master_sites} 0]
 livecheck.regex             ${name}-(6(?:\\.\\d+)+(?:-\\d+)?)\.tar
