--- oiio-Release-1.8.10/src/cmake/modules/FindPugiXML.cmake.orig	2018-04-28 08:04:18.150113300 +0300
+++ oiio-Release-1.8.10/src/cmake/modules/FindPugiXML.cmake	2018-04-28 08:12:29.664477900 +0300
@@ -10,12 +10,12 @@
            NAMES pugixml.hpp
            PATHS ${PUGIXML_HOME}/include
            /usr/local/include
-           /usr/local/include/pugixml-1.8)
+           PATH_SUFFIXES pugixml-1.8 pugixml-1.9)
 find_library (PUGIXML_LIBRARY
               NAMES pugixml
               PATHS ${PUGIXML_HOME}/lib
               /usr/local/lib
-              /usr/local/lib/pugixml-1.8)
+              PATH_SUFFIXES pugixml-1.8 pugixml-1.9)
 
 # Support the REQUIRED and QUIET arguments, and set PUGIXML_FOUND if found.
 include (FindPackageHandleStandardArgs)
