--- PKGBUILD.orig	2018-05-02 11:36:43.000000000 +0200
+++ PKGBUILD	2018-05-02 11:36:43.000000000 +0200
@@ -43,9 +43,9 @@
   mkdir -p "${srcdir}/build-${MINGW_CHOST}"
   cd "${srcdir}/build-${MINGW_CHOST}"
 
-  CPPFLAGS+=" -DLIBRAW_NODLL"
+  CPPFLAGS+=" -DLIBRAW_BUILDLIB -D__STDC_LIMIT_MACROS"
   ../${_realname}-${pkgver}/configure \
-      --prefix=${MINGW_PREFIX} \
+      --prefix=${MINGW_PREFIX}/libraw-gpl3 \
       --build=${MINGW_CHOST} \
       --host=${MINGW_CHOST} \
       --enable-jasper \
@@ -53,7 +53,7 @@
       --enable-demosaic-pack-gpl2 \
       --enable-demosaic-pack-gpl3
 
-  make
+  make -j${MKJOBS}
 }
 
 package() {
