--- PKGBUILD.orig	2018-05-02 11:36:43.000000000 +0200
+++ PKGBUILD	2018-05-02 11:51:52.000000000 +0200
@@ -1,11 +1,11 @@
 # Maintainer: Alexey Pavlov <alexpux@gmail.com>
 
 _realname=LibRaw
-pkgbase=mingw-w64-libraw
-pkgname="${MINGW_PACKAGE_PREFIX}-libraw"
-provides=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-replaces=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgbase=mingw-w64-libraw-gpl3
+pkgname="${MINGW_PACKAGE_PREFIX}-libraw-gpl3"
+provides=("${MINGW_PACKAGE_PREFIX}-${_realname}-gpl3")
+replaces=("${MINGW_PACKAGE_PREFIX}-${_realname}-gpl3")
+conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}-gpl3")
 pkgver=0.18.9
 pkgrel=1
 pkgdesc="Library for reading RAW files obtained from digital photo cameras (mingw-w64)"
@@ -43,9 +43,9 @@
   mkdir -p "${srcdir}/build-${MINGW_CHOST}"
   cd "${srcdir}/build-${MINGW_CHOST}"
 
-  CPPFLAGS+=" -DLIBRAW_NODLL"
+  CPPFLAGS+=" -DLIBRAW_BUILDLIB -D__STDC_LIMIT_MACROS"
   ../${_realname}-${pkgver}/configure \
-      --prefix=${MINGW_PREFIX} \
+      --prefix=${MINGW_PREFIX}/libraw-gpl3 \
       --build=${MINGW_CHOST} \
       --host=${MINGW_CHOST} \
       --enable-jasper \
@@ -53,7 +53,7 @@
       --enable-demosaic-pack-gpl2 \
       --enable-demosaic-pack-gpl3
 
-  make
+  make -j${MKJOBS}
 }
 
 package() {
