--- tests/pixbuf-readonly-to-mutable.c.orig	2015-02-17 16:14:21.000000000 -0800
+++ tests/pixbuf-readonly-to-mutable.c	2015-03-07 22:36:27.000000000 -0800
@@ -77,7 +77,7 @@
 
     buf = g_new0 (MappedBuf, 1);
     buf->len = pages * pagesize;
-    buf->buf = mmap (NULL, buf->len, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+    buf->buf = mmap (NULL, buf->len, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
     g_assert (buf->buf != NULL);
 
     memcpy (buf->buf, gdk_pixbuf_get_pixels (reference), pixlen);
