<<<<<<< HEAD
--- Portfile.orig	2017-09-07 17:45:54.000000000 +0200
+++ Portfile	2017-09-07 17:46:10.000000000 +0200
=======
--- Portfile.orig	2017-11-22 09:21:41.000000000 +0100
+++ Portfile	2017-11-22 09:21:41.000000000 +0100
>>>>>>> 9542ac6a3... MacPorts: update (first batch)
@@ -37,7 +37,7 @@
 depends_lib                 port:expat \
                             port:freetype \
                             port:fontconfig \
-                            port:jpeg \
+                            path:lib/libjpeg.dylib:jpeg \
                             port:libiconv \
                             port:libpng \
                             port:tiff \
