--- Portfile.orig	2016-10-19 14:46:44.000000000 +0200
+++ Portfile	2016-10-19 16:25:09.000000000 +0200
@@ -32,7 +32,7 @@
                     port:expat \
                     port:fontconfig \
                     port:freetype \
-                    port:jpeg \
+                    path:lib/libjpeg.dylib:jpeg \
                     path:lib/pkgconfig/glib-2.0.pc:glib2 \
                     path:lib/pkgconfig/cairo.pc:cairo \
                     port:lcms2 \
@@ -84,6 +84,18 @@
     configure.args-delete   --disable-poppler-qt5
 }
 
+variant gcc49 description {Compile with GCC/G++ 4.9} {
+    configure.compiler macports-gcc-4.9
+    if {[variant_isset universal]} {
+        # use the driverdriver (built using https://github.com/devernay/macportsGCCfixup) to be able to build universal binaries
+        if {[file tail ${configure.cmd}] != "cmake"} {
+            configure.args-append   CC=${configure.cc}-dd CXX=${configure.cxx}-dd
+        } else {
+            configure.args-append   -DCMAKE_C_COMPILER=${configure.cc}-dd -DCMAKE_CXX_COMPILER=${configure.cxx}-dd
+        }
+    }
+}
+
 if {${subport} ne ${name}} {
     depends_extract-append  port:git
 
