--- Portfile.orig	2018-06-18 15:18:36.000000000 +0200
+++ Portfile	2018-06-18 15:18:36.000000000 +0200
@@ -34,7 +34,7 @@
                     port:expat \
                     port:fontconfig \
                     port:freetype \
-                    port:jpeg \
+                    path:lib/libjpeg.dylib:jpeg \
                     path:lib/pkgconfig/glib-2.0.pc:glib2 \
                     path:lib/pkgconfig/cairo.pc:cairo \
                     port:lcms2 \
