--- Portfile.orig	2016-10-19 14:46:44.000000000 +0200
+++ Portfile	2016-10-19 16:29:53.000000000 +0200
@@ -32,7 +32,7 @@
                     port:expat \
                     port:fontconfig \
                     port:freetype \
-                    port:jpeg \
+                    path:lib/libjpeg.dylib:jpeg \
                     path:lib/pkgconfig/glib-2.0.pc:glib2 \
                     path:lib/pkgconfig/cairo.pc:cairo \
                     port:lcms2 \
