diff -Naur libs/mpi/src/python/datatypes.cpp libs/mpi/src/python/datatypes.cpp
--- libs/mpi/src/python/datatypes.cpp	2007-11-25 13:38:02.000000000 -0500
+++ libs/mpi/src/python/datatypes.cpp	2013-03-11 20:59:57.171732691 -0400
@@ -13,6 +13,10 @@
 #include <boost/mpi/python/serialize.hpp>
 #include <boost/mpi.hpp>
 
+#if PY_MAJOR_VERSION >= 3
+#define PyInt_Type PyLong_Type
+#endif
+
 namespace boost { namespace mpi { namespace python {
 
 void export_datatypes()
diff -Naur libs/mpi/build/__init__.py llibs/mpi/build/__init__.py
--- libs/mpi/build/__init__.py	2013-11-27 12:52:37.000000000 -0600
+++ libs/mpi/build/__init__.py	2013-11-27 12:52:55.000000000 -0600
@@ -6,5 +6,5 @@
     import mpi
     sys.setdlopenflags(flags)
 else:
-    import mpi
+    from . import mpi
