--- tools/build/src/tools/python.jam.orig	2009-10-06 15:56:18.000000000 +0200
+++ tools/build/src/tools/python.jam	2009-10-06 15:53:48.000000000 +0200
@@ -546,6 +546,13 @@
         libraries ?= $(default-library-path) ;
         includes ?= $(default-include-path) ;
     }
+    else if $(target-os) = darwin
+    {
+        includes ?= $(prefix)/Headers ;
+
+        local lib = $(exec-prefix)/lib ;
+        libraries ?= $(lib)/python$(version)/config $(lib) ;
+    }
     else
     {
         includes ?= $(prefix)/include/python$(version) ;
