--- internal/libraw_x3f.cpp.orig	2019-07-03 04:32:22.000000000 +0200
+++ internal/libraw_x3f.cpp	2019-07-03 04:36:47.000000000 +0200
@@ -1391,7 +1391,7 @@
                                x3f_directory_entry_t *DE,
                                int bits,
                                int row,
-                               int offset,
+                               int16_t offset,
                                int *minimum)
 {
   x3f_directory_entry_header_t *DEH = &DE->header;
