diff --git Makefile.dist Makefile.dist
index cb2f159..3605b94 100644
--- Makefile.dist
+++ Makefile.dist
@@ -81,10 +81,10 @@ bin/mem_image: lib/libraw.a samples/mem_image.cpp
 	${CXX} -DLIBRAW_NOTHREADS  ${CFLAGS} -o bin/mem_image samples/mem_image.cpp -L./lib -lraw  -lm  ${LDADD}
 
 bin/dcraw_half: lib/libraw.a object/dcraw_half.o
-	${CC} -DLIBRAW_NOTHREADS  ${CFLAGS} -o bin/dcraw_half object/dcraw_half.o -L./lib -lraw  -lm -lstdc++  ${LDADD}
+	${CC} -DLIBRAW_NOTHREADS  ${CFLAGS} -o bin/dcraw_half object/dcraw_half.o -L./lib -lraw  -lm ${LDADD}
 
 bin/half_mt: lib/libraw_r.a object/half_mt.o
-	${CC}   -pthread ${CFLAGS} -o bin/half_mt object/half_mt.o -L./lib -lraw_r  -lm -lstdc++  ${LDADD}
+	${CC}   -pthread ${CFLAGS} -o bin/half_mt object/half_mt.o -L./lib -lraw_r  -lm  ${LDADD}
 
 bin/dcraw_emu: lib/libraw.a samples/dcraw_emu.cpp
 	${CXX} -DLIBRAW_NOTHREADS  ${CFLAGS} -o bin/dcraw_emu samples/dcraw_emu.cpp -L./lib -lraw  -lm  ${LDADD}
diff --git libraw.pc.in libraw.pc.in
index 0e530b2..9844828 100644
--- libraw.pc.in
+++ libraw.pc.in
@@ -7,5 +7,5 @@ Name: libraw
 Description: Raw image decoder library (non-thread-safe)
 Requires: @PACKAGE_REQUIRES@
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lraw -lstdc++@PC_OPENMP@
+Libs: -L${libdir} -lraw @PC_OPENMP@
 Cflags: -I${includedir}/libraw
diff --git libraw_r.pc.in libraw_r.pc.in
index a7f4535..cf48381 100644
--- libraw_r.pc.in
+++ libraw_r.pc.in
@@ -7,5 +7,5 @@ Name: libraw
 Description: Raw image decoder library (thread-safe)
 Requires: @PACKAGE_REQUIRES@
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lraw_r -lstdc++@PC_OPENMP@
+Libs: -L${libdir} -lraw_r @PC_OPENMP@
 Cflags: -I${includedir}/libraw
