--- Makefile.dist.orig	2023-01-10 11:32:32.000000000 +0000
+++ Makefile.dist	2023-01-10 11:35:00.000000000 +0000
@@ -167,10 +167,10 @@
 	${CXX} -DLIBRAW_NOTHREADS  ${CFLAGS} -o bin/mem_image samples/mem_image_sample.cpp -L./lib -lraw  -lm  ${LDADD}
 
 bin/dcraw_half: lib/libraw.a object/dcraw_half.o
-	${CC} -DLIBRAW_NOTHREADS  ${CFLAGS} -o bin/dcraw_half object/dcraw_half.o -L./lib -lraw  -lm -lstdc++  ${LDADD}
+	${CC} -DLIBRAW_NOTHREADS  ${CFLAGS} -o bin/dcraw_half object/dcraw_half.o -L./lib -lraw  -lm  ${LDADD}
 
 bin/half_mt: lib/libraw_r.a object/half_mt.o
-	${CC}   -pthread ${CFLAGS} -o bin/half_mt object/half_mt.o -L./lib -lraw_r  -lm -lstdc++  ${LDADD}
+	${CC}   -pthread ${CFLAGS} -o bin/half_mt object/half_mt.o -L./lib -lraw_r  -lm  ${LDADD}
 
 bin/dcraw_emu: lib/libraw.a samples/dcraw_emu.cpp
 	${CXX} -DLIBRAW_NOTHREADS  ${CFLAGS} -o bin/dcraw_emu samples/dcraw_emu.cpp -L./lib -lraw  -lm  ${LDADD}
--- libraw.pc.in.orig	2023-01-10 11:35:20.000000000 +0000
+++ libraw.pc.in	2023-01-10 11:35:43.000000000 +0000
@@ -7,6 +7,6 @@
 Description: Raw image decoder library (non-thread-safe)
 Requires: @PACKAGE_REQUIRES@
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lraw -lstdc++@PC_OPENMP@
+Libs: -L${libdir} -lraw @PC_OPENMP@
 Libs.private: @PACKAGE_LIBS_PRIVATE@
 Cflags: -I${includedir}/libraw -I${includedir}
--- libraw_r.pc.in.orig	2023-01-10 11:35:53.000000000 +0000
+++ libraw_r.pc.in	2023-01-10 11:36:10.000000000 +0000
@@ -7,6 +7,6 @@
 Description: Raw image decoder library (thread-safe)
 Requires: @PACKAGE_REQUIRES@
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lraw_r -lstdc++@PC_OPENMP@
+Libs: -L${libdir} -lraw_r @PC_OPENMP@
 Libs.private: @PACKAGE_LIBS_PRIVATE@
 Cflags: -I${includedir}/libraw -I${includedir}
