--- Portfile.orig	2016-01-11 17:06:04.000000000 +0100
+++ Portfile	2016-01-11 17:06:11.000000000 +0100
@@ -35,7 +35,7 @@
 configure.args      --mandir=${prefix}/share/man \
                     --disable-win32 --disable-conio --disable-slang \
                     --disable-imlib2 --disable-doc \
-                    --disable-x11 --enable-ncurses --enable-imlib2 \
+                    --disable-x11 --enable-ncurses \
                     --disable-ruby --disable-csharp
 
 # Disable silent rules
@@ -53,6 +53,12 @@
 	                        --x-libraries=${prefix}/lib 
 }
 
+variant imlib2 {
+	depends_lib-append	port:imlib2
+	configure.args-delete   --disable-imlib2
+	configure.args-append	--enable-imlib2 
+}
+
 compiler.blacklist  llvm-gcc-4.2
 
 test.run            yes
