--- Portfile.orig	2016-10-27 16:57:35.000000000 +0200
+++ Portfile	2016-10-27 16:59:07.000000000 +0200
@@ -78,6 +78,24 @@
     }
 }
 
+variant nochdir description {disable chdir} {
+    configure.args-delete -DGLFW_USE_CHDIR=on
+    configure.args-append -DGLFW_USE_CHDIR=off
+}
+
+variant nomenubar description {disable menubar} {
+    configure.args-delete -DGLFW_USE_MENUBAR=on
+    configure.args-append -DGLFW_USE_MENUBAR=off
+}
+
+variant noretina description {disable retina} {
+    configure.args-delete -DGLFW_USE_RETINA=on
+    configure.args-append -DGLFW_USE_RETINA=off
+}
+
+variant natron conflicts noretina requires nochdir nomenubar description {Natron version} {}
+
+
 # use the real home page, not github's
 
 homepage            http://www.glfw.org/
@@ -92,7 +110,11 @@
     -DBUILD_SHARED_LIBS=on \
     -DGLFW_BUILD_EXAMPLES=off \
     -DGLFW_BUILD_TESTS=off \
-    -DGLFW_CMAKE_CONFIG_PATH=share
+    -DGLFW_CMAKE_CONFIG_PATH=share \
+    -DGLFW_USE_CHDIR=on \
+    -DGLFW_USE_MENUBAR=on \
+    -DGLFW_USE_RETINA=on
+
 
 # remove top-level library path, such that internal libraries are used
 # instead of any already-installed ones.
