--- Portfile.orig	2016-06-07 09:14:29.000000000 +0200
+++ Portfile	2016-06-07 09:14:38.000000000 +0200
@@ -64,7 +64,10 @@
 configure.args      -DBUILD_SHARED_LIBS=on \
                     -DGLFW_BUILD_EXAMPLES=off \
                     -DGLFW_BUILD_TESTS=off \
-                    -DGLFW_CMAKE_CONFIG_PATH=share
+                    -DGLFW_CMAKE_CONFIG_PATH=share \
+                    -DGLFW_USE_CHDIR=on \
+                    -DGLFW_USE_MENUBAR=on \
+                    -DGLFW_USE_RETINA=on
 
 # remove top-level library path, such that internal libraries are used
 # instead of any already-installed ones.
@@ -82,6 +85,23 @@
     configure.args-append -DDOXYGEN_EXECUTABLE=
 }
 
+variant nochdir description {disable chdir} {
+    configure.args-delete -DGLFW_USE_CHDIR=on
+    configure.args-append -DGLFW_USE_CHDIR=off
+}
+
+variant nomenubar description {disable menubar} {
+    configure.args-delete -DGLFW_USE_MENUBAR=on
+    configure.args-append -DGLFW_USE_MENUBAR=off
+}
+
+variant noretina description {disable retina} {
+    configure.args-delete -DGLFW_USE_RETINA=on
+    configure.args-append -DGLFW_USE_RETINA=off
+}
+
+variant natron conflicts noretina requires nochdir nomenubar description {Natron version} {}
+
 platform darwin {
     pre-fetch {
         if {${os.major} < 10} {
