diff -Naur macosx.orig/English.lproj/Attributes.xib macosx/English.lproj/Attributes.xib
--- macosx.orig/English.lproj/Attributes.xib	2016-08-09 23:02:10.000000000 +0200
+++ macosx/English.lproj/Attributes.xib	2019-02-18 13:44:23.000000000 +0100
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <archive type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="7.01">
 	<data>
-		<int key="IBDocument.SystemTarget">1050</int>
+		<int key="IBDocument.SystemTarget">1060</int>
 		<string key="IBDocument.SystemVersion">9B18</string>
 		<string key="IBDocument.InterfaceBuilderVersion">629</string>
 		<string key="IBDocument.AppKitVersion">949</string>
diff -Naur macosx.orig/English.lproj/Document.xib macosx/English.lproj/Document.xib
--- macosx.orig/English.lproj/Document.xib	2016-08-09 23:02:10.000000000 +0200
+++ macosx/English.lproj/Document.xib	2019-02-18 13:44:44.000000000 +0100
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <archive type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="7.01">
 	<data>
-		<int key="IBDocument.SystemTarget">1050</int>
+		<int key="IBDocument.SystemTarget">1060</int>
 		<string key="IBDocument.SystemVersion">9C7010</string>
 		<string key="IBDocument.InterfaceBuilderVersion">629</string>
 		<string key="IBDocument.AppKitVersion">949.26</string>
diff -Naur macosx.orig/English.lproj/Export.xib macosx/English.lproj/Export.xib
--- macosx.orig/English.lproj/Export.xib	2016-08-09 23:02:10.000000000 +0200
+++ macosx/English.lproj/Export.xib	2019-02-18 13:44:38.000000000 +0100
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <archive type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="7.01">
 	<data>
-		<int key="IBDocument.SystemTarget">1050</int>
+		<int key="IBDocument.SystemTarget">1060</int>
 		<string key="IBDocument.SystemVersion">9C7010</string>
 		<string key="IBDocument.InterfaceBuilderVersion">629</string>
 		<string key="IBDocument.AppKitVersion">949.26</string>
diff -Naur macosx.orig/English.lproj/MainMenu.xib macosx/English.lproj/MainMenu.xib
--- macosx.orig/English.lproj/MainMenu.xib	2016-08-09 23:02:10.000000000 +0200
+++ macosx/English.lproj/MainMenu.xib	2019-02-18 13:44:30.000000000 +0100
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <archive type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="7.01">
 	<data>
-		<int key="IBDocument.SystemTarget">1050</int>
+		<int key="IBDocument.SystemTarget">1060</int>
 		<string key="IBDocument.SystemVersion">9E17</string>
 		<string key="IBDocument.InterfaceBuilderVersion">629</string>
 		<string key="IBDocument.AppKitVersion">949.33</string>
