diff -ur meson-0.57.0.orig/mesonbuild/mesonlib/universal.py meson-0.57.0/mesonbuild/mesonlib/universal.py
--- meson-0.57.0.orig/mesonbuild/mesonlib/universal.py  2021-02-15 19:57:57.698044600 +0530
+++ meson-0.57.0/mesonbuild/mesonlib/universal.py       2021-02-15 20:04:07.519197200 +0530
@@ -865,6 +865,8 @@


 def default_prefix() -> str:
+    if is_windows() and 'MSYSTEM' in os.environ:
+        return sys.prefix
     return 'c:/' if is_windows() else '/usr/local'
