--- meson-0.50.1/mesonbuild/mlog.py.orig	2019-03-06 21:48:10.000000000 +0100
+++ meson-0.50.1/mesonbuild/mlog.py	2019-04-17 17:46:10.229092600 +0200
@@ -31,7 +31,7 @@
     stdout = kernel.GetStdHandle(-11)
     mode = DWORD()
     if not kernel.GetConsoleMode(stdout, byref(mode)):
-        return False
+        return os.isatty(sys.stdout.fileno())
     # ENABLE_VIRTUAL_TERMINAL_PROCESSING == 0x4
     # If the call to enable VT processing fails (returns 0), we fallback to
     # original behavior
