--- SeExpr-2.11/src/SeExpr/SeNoise.cpp.orig	2021-04-22 08:50:18.000000000 -0700
+++ SeExpr-2.11/src/SeExpr/SeNoise.cpp	2021-04-22 08:51:24.000000000 -0700
@@ -29,11 +29,11 @@
 
 #ifdef __SSE4_1__
 inline double floorSSE(double val) {
-    return _mm_floor_sd(_mm_set_sd(0.0), _mm_set_sd(val))[0];
+    return _mm_cvtsd_f64(_mm_floor_sd(_mm_set_sd(0.0), _mm_set_sd(val)));
 }
 
 inline double roundSSE(double val) {
-    return _mm_round_sd(_mm_set_sd(0.0), _mm_set_sd(val), _MM_FROUND_TO_NEAREST_INT)[0];
+    return _mm_cvtsd_f64(_mm_round_sd(_mm_set_sd(0.0), _mm_set_sd(val), _MM_FROUND_TO_NEAREST_INT));
 }
 #else
 #define floorSSE floor
