--- Portfile.orig	2015-12-11 01:33:16.000000000 -0800
+++ Portfile	2015-12-11 01:33:23.000000000 -0800
@@ -77,6 +77,10 @@
                 ${dir}/curl-config \
                 ${dir}/libcurl.pc
             reinplace -E {s/ '(host_alias|--host)=[^']+'//g} ${dir}/curl-config
+            reinplace -E "s/-lssl -lcrypto -lssl -lcrypto/-lssl -lcrypto/g" ${dir}/curl-config ${dir}/libcurl.pc
+            reinplace -E "s|-L${prefix}/lib -L${prefix}/lib  -L${prefix}/lib -L${prefix}/lib|-L${prefix}/lib|g" ${dir}/curl-config ${dir}/libcurl.pc
+            reinplace -E "s|-L${prefix}/lib  -L${prefix}/lib|-L${prefix}/lib|g" ${dir}/curl-config ${dir}/libcurl.pc
+            reinplace -E "s|-L${prefix}/lib -L${prefix}/lib|-L${prefix}/lib|g" ${dir}/curl-config ${dir}/libcurl.pc
         }
     }
 
