--- libcaca-0.99.beta19/caca/figfont.c.orig	2012-12-28 17:34:08.000000000 +0000
+++ libcaca-0.99.beta19/caca/figfont.c	2014-07-30 10:38:17.765662500 +0000
@@ -29,7 +29,7 @@
 #include "caca.h"
 #include "caca_internals.h"
 
-#if defined _WIN32 && defined __GNUC__ && __GNUC__ >= 3
+#if defined _MSC_VER && defined __GNUC__ && __GNUC__ >= 3
 int sprintf_s(char *s, size_t n, const char *fmt, ...) CACA_WEAK;
 int vsnprintf(char *s, size_t n, const char *fmt, va_list ap) CACA_WEAK;
 #endif
@@ -636,7 +636,7 @@
  * Functions for the mingw32 runtime
  */
 
-#if defined _WIN32 && defined __GNUC__ && __GNUC__ >= 3
+#if defined _MSC_VER && defined __GNUC__ && __GNUC__ >= 3
 int sprintf_s(char *s, size_t n, const char *fmt, ...)
 {
     va_list args;
--- libcaca-0.99.beta19/caca/string.c.orig	2012-12-28 17:34:08.000000000 +0000
+++ libcaca-0.99.beta19/caca/string.c	2014-07-30 10:38:28.801063800 +0000
@@ -36,7 +36,7 @@
 #include "caca.h"
 #include "caca_internals.h"
 
-#if defined _WIN32 && defined __GNUC__ && __GNUC__ >= 3
+#if defined _MSC_VER && defined __GNUC__ && __GNUC__ >= 3
 int vsnprintf_s(char *s, size_t n, size_t c,
                 const char *fmt, va_list ap) CACA_WEAK;
 int vsnprintf(char *s, size_t n, const char *fmt, va_list ap) CACA_WEAK;
@@ -610,7 +610,7 @@
  * Functions for the mingw32 runtime
  */
 
-#if defined _WIN32 && defined __GNUC__ && __GNUC__ >= 3
+#if defined _MSC_VER && defined __GNUC__ && __GNUC__ >= 3
 int vsnprintf_s(char *s, size_t n, size_t c, const char *fmt, va_list ap)
 {
     return vsnprintf(s, n, fmt, ap);
