--- libcaca-0.99.beta19/configure.ac.orig	2014-05-16 19:07:53.000000000 +0000
+++ libcaca-0.99.beta19/configure.ac	2014-07-30 11:04:20.605117900 +0000
@@ -467,8 +467,8 @@
   if test "${RUBY}" != "no"; then
     RUBY_CFLAGS="$(ruby -r mkmf -e 'conf = RbConfig::CONFIG.merge("hdrdir" => $hdrdir, "arch_hdrdir" => $arch_hdrdir); print RbConfig::expand("#$INCFLAGS", conf)')"
     RUBY_LIBS="-L$(ruby -r rbconfig -e 'print RbConfig::CONFIG@<:@"libdir"@:>@') -l$(ruby -r rbconfig -e 'print RbConfig::CONFIG@<:@"RUBY_SO_NAME"@:>@')"
-    RUBY_SITEARCHDIR=`ruby -r rbconfig -e 'print RbConfig::CONFIG@<:@"sitearchdir"@:>@'`
-    RUBY_SITELIBDIR=`ruby -r rbconfig -e 'print RbConfig::CONFIG@<:@"sitelibdir"@:>@'`
+    RUBY_SITEARCHDIR=$(cygpath `ruby -r rbconfig -e 'print RbConfig::CONFIG@<:@"sitearchdir"@:>@'`)
+    RUBY_SITELIBDIR=$(cygpath `ruby -r rbconfig -e 'print RbConfig::CONFIG@<:@"sitelibdir"@:>@'`)
     AC_SUBST(RUBY_CFLAGS)
     AC_SUBST(RUBY_LIBS)
     AC_SUBST(RUBY_SITEARCHDIR)
