--- boost_1_74_0/boost/config/compiler/gcc.hpp.orig	2020-10-07 19:30:01.735299600 +0300
+++ boost_1_74_0/boost/config/compiler/gcc.hpp	2020-10-07 19:30:05.597735500 +0300
@@ -323,13 +323,6 @@
 #  define BOOST_FALLTHROUGH __attribute__((fallthrough))
 #endif
 
-#if defined(__MINGW32__) && !defined(__MINGW64__)
-// Currently (March 2019) thread_local is broken on mingw for all current 32bit compiler releases, see
-// https://sourceforge.net/p/mingw-w64/bugs/527/
-// Not setting this causes program termination on thread exit.
-#define BOOST_NO_CXX11_THREAD_LOCAL
-#endif
-
 //
 // Unused attribute:
 #if __GNUC__ >= 4
