--- boost_1_68_0/libs/pool/test/Jamfile.v2.orig	2018-08-15 19:53:32.028325900 -0400
+++ boost_1_68_0/libs/pool/test/Jamfile.v2	2018-08-15 20:00:08.714006800 -0400
@@ -22,19 +22,20 @@
 import testing ;
 
 test-suite pool :
-    [ run test_simple_seg_storage.cpp : : : <toolset>msvc:<cxxflags>-wd4267 ]
-    [ run test_pool_alloc.cpp ]
-    [ run pool_msvc_compiler_bug_test.cpp : : : <toolset>msvc:<cxxflags>-wd4512 ]
-    [ run test_msvc_mem_leak_detect.cpp ]
-    [ run test_bug_3349.cpp ]
-    [ run test_bug_4960.cpp ]
+    [ run test_simple_seg_storage.cpp : : : <toolset>msvc:<cxxflags>-wd4267 <library>/boost/system//boost_system ]
+    [ run test_pool_alloc.cpp <library>/boost/system//boost_system ]
+    [ run pool_msvc_compiler_bug_test.cpp : : : <toolset>msvc:<cxxflags>-wd4512 <library>/boost/system//boost_system ]
+    [ run test_msvc_mem_leak_detect.cpp <library>/boost/system//boost_system ]
+    [ run test_bug_3349.cpp <library>/boost/system//boost_system ]
+    [ run test_bug_4960.cpp <library>/boost/system//boost_system ]
     [ run test_bug_1252.cpp : : :
         <toolset>clang:<cxxflags>-Wno-c++11-long-long
         <toolset>gcc:<cxxflags>-Wno-long-long
-        <toolset>pathscale:<cxxflags>-Wno-long-long ]
-    [ run test_bug_2696.cpp ]
-    [ run test_bug_5526.cpp ]
-    [ run test_threading.cpp : : : <threading>multi <library>/boost/thread//boost_thread ]
+        <toolset>pathscale:<cxxflags>-Wno-long-long
+        <library>/boost/system//boost_system ]
+    [ run test_bug_2696.cpp <library>/boost/system//boost_system ]
+    [ run test_bug_5526.cpp <library>/boost/system//boost_system ]
+    [ run test_threading.cpp : : : <threading>multi <library>/boost/thread//boost_thread <library>/boost/system//boost_system ]
     [ compile test_poisoned_macros.cpp ]
     ;
 
