--- boost_1_70_0.orig/tools/build/src/build/virtual-target.jam	2019-04-12 16:15:08.325230800 -0400
+++ boost_1_70_0/tools/build/src/build/virtual-target.jam	2019-04-12 16:53:35.810087000 -0400
@@ -1137,19 +1137,6 @@
             properties-added = [ set.difference $(p2) : $(p1) ] ;
             properties-added ?= "none" ;
         }
-        import errors : user-error : errors.user-error ;
-        errors.user-error "Name clash for '$(actual-name)'"
-            : ""
-            : "Tried to build the target twice, with property sets having "
-            : "these incompatible properties:"
-            : ""
-            : "    - " $(properties-removed)
-            : "    - " $(properties-added)
-            : ""
-            : "Please make sure to have consistent requirements for these "
-            : "properties everywhere in your project, especially for install"
-            : "targets."
-            ;
     }
     else
     {
