--- boost_1_73_0/libs/regex/build/Jamfile.v2.orig	2020-04-22 16:35:55.000000000 +0300
+++ boost_1_73_0/libs/regex/build/Jamfile.v2	2020-05-24 16:51:35.262191700 +0300
@@ -33,11 +33,6 @@
 #
 if ! $(disable-icu)
 {
-   if [ modules.peek : ICU_LINK ]
-   {    
-       errors.user-error : "The ICU_LINK option is no longer supported by the Boost.Regex build - please refer to the documentation for equivalent options" ;
-   }
-
    if [ modules.peek : ICU_PATH ]
    {    
        ICU_PATH =  [ modules.peek : ICU_PATH ] ;
