Index: libmad-0.15.1b/configure.ac
===================================================================
--- libmad-0.15.1b.orig/configure.ac	2008-03-07 20:33:05.000000000 +0000
+++ libmad-0.15.1b/configure.ac	2008-03-07 20:33:31.000000000 +0000
@@ -233,6 +233,7 @@
 then
     case "$host" in
 	i?86-*)     FPM="INTEL"  ;;
+	x86_64*)    FPM="64BIT"  ;;
 	arm*-*)     FPM="ARM"    ;;
 	mips*-*)    FPM="MIPS"   ;;
 	sparc*-*)   FPM="SPARC"  ;;
