--- configure.orig	2018-04-20 22:08:11.000000000 -0500
+++ configure	2018-07-22 16:21:00.000000000 -0500
@@ -13938,8 +13938,8 @@
     darwin*)
       # Special case for OSX builds. Append these to give the user a chance to
       # override with --with-gl*
-      glut_cflags="$glut_cflags|-framework GLUT -framework OpenGL"
-      glut_ldflags="$glut_ldflags|-framework GLUT -framework OpenGL"
+      glut_cflags="-framework GLUT -framework OpenGL"
+      glut_ldflags="-framework GLUT -framework OpenGL"
       # quiet deprecation warnings for glut
       SAVED_CFLAGS="$CFLAGS"
           CFLAGS="-Werror -Wno-deprecated-declarations"
@@ -13975,7 +13975,7 @@
       none) ;;
       *) CPPFLAGS="$flag $CPPFLAGS";;
     esac
-    for ac_header in GL/glut.h GLUT/glut.h OpenGL/glut.h
+    for ac_header in GLUT/glut.h OpenGL/glut.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
