--- configure.orig	2015-11-19 08:19:13.000000000 -0700
+++ configure	2018-10-07 07:33:54.000000000 -0700
@@ -5889,7 +5889,7 @@
 
     CFLAGS="${_isl_saved_CFLAGS} ${islinc} ${gmpinc}"
     LDFLAGS="${_isl_saved_LDFLAGS} ${isllibs}"
-    LIBS="${_isl_saved_LIBS} -lisl"
+    LIBS="${_isl_saved_LIBS} -lisl.15"
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.10 of ISL" >&5
 $as_echo_n "checking for version 0.10 of ISL... " >&6; }
@@ -5937,7 +5937,7 @@
 
     CFLAGS="${_isl_saved_CFLAGS} ${islinc} ${gmpinc}"
     LDFLAGS="${_isl_saved_LDFLAGS} ${isllibs}"
-    LIBS="${_isl_saved_LIBS} -lisl"
+    LIBS="${_isl_saved_LIBS} -lisl.15"
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.11 of ISL" >&5
 $as_echo_n "checking for version 0.11 of ISL... " >&6; }
@@ -5985,7 +5985,7 @@
 
     CFLAGS="${_isl_saved_CFLAGS} ${islinc} ${gmpinc}"
     LDFLAGS="${_isl_saved_LDFLAGS} ${isllibs}"
-    LIBS="${_isl_saved_LIBS} -lisl"
+    LIBS="${_isl_saved_LIBS} -lisl.15"
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.12 of ISL" >&5
 $as_echo_n "checking for version 0.12 of ISL... " >&6; }
@@ -6033,7 +6033,7 @@
 
     CFLAGS="${_isl_saved_CFLAGS} ${islinc} ${gmpinc}"
     LDFLAGS="${_isl_saved_LDFLAGS} ${isllibs}"
-    LIBS="${_isl_saved_LIBS} -lisl"
+    LIBS="${_isl_saved_LIBS} -lisl.15"
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.14 of ISL" >&5
 $as_echo_n "checking for version 0.14 of ISL... " >&6; }
@@ -6081,7 +6081,7 @@
 
     CFLAGS="${_isl_saved_CFLAGS} ${islinc} ${gmpinc}"
     LDFLAGS="${_isl_saved_LDFLAGS} ${isllibs}"
-    LIBS="${_isl_saved_LIBS} -lisl"
+    LIBS="${_isl_saved_LIBS} -lisl.15"
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.15 of ISL" >&5
 $as_echo_n "checking for version 0.15 of ISL... " >&6; }
@@ -6207,7 +6207,7 @@
   fi
 
   clooginc="-DCLOOG_INT_GMP ${clooginc}"
-  clooglibs="${clooglibs} -lcloog-isl ${isllibs} -lisl"
+  clooglibs="${clooglibs} -lcloog-isl ${isllibs} -lisl.15"
