From 810a0235837d6f31d8b3c05544f99461df71403e Mon Sep 17 00:00:00 2001
From: Larry Gritz <lg@larrygritz.com>
Date: Mon, 23 Apr 2018 15:50:14 -0700
Subject: [PATCH] Changes to allow building against the new ffmpeg 4.0 (#1926)

---
 src/ffmpeg.imageio/ffmpeginput.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/ffmpeg.imageio/ffmpeginput.cpp b/src/ffmpeg.imageio/ffmpeginput.cpp
index 4e0cb3b96..548ddba4e 100644
--- a/src/ffmpeg.imageio/ffmpeginput.cpp
+++ b/src/ffmpeg.imageio/ffmpeginput.cpp
@@ -124,6 +124,15 @@ inline int receive_frame(AVCodecContext *avctx, AVFrame *picture,
 }
 #endif
 
+
+// Changes for ffmpeg 4.0
+#define USE_FFMPEG_4_0 (LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(58, 18, 100))
+
+#if USE_FFMPEG_4_0
+#  define CODEC_CAP_DELAY AV_CODEC_CAP_DELAY
+#endif
+
+
 #include <boost/thread/once.hpp>
 
 #include <OpenImageIO/imageio.h>
