--- PKGBUILD.orig	2017-11-03 11:44:53.000000000 +0100
+++ PKGBUILD	2017-11-07 08:37:48.000000000 +0100
@@ -4,28 +4,24 @@
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=1.7.17
-pkgrel=1
+pkgrel=1.2
 pkgdesc="A library for reading and writing images, including classes, utilities, and applications (mingw-w64)"
 arch=('any')
 url="http://www.openimageio.org/"
 license=("custom")
 depends=("${MINGW_PACKAGE_PREFIX}-boost"
-         "${MINGW_PACKAGE_PREFIX}-field3d"
          "${MINGW_PACKAGE_PREFIX}-freetype"
          "${MINGW_PACKAGE_PREFIX}-jasper"
          "${MINGW_PACKAGE_PREFIX}-giflib"
          "${MINGW_PACKAGE_PREFIX}-glew"
-         "${MINGW_PACKAGE_PREFIX}-hdf5"
          "${MINGW_PACKAGE_PREFIX}-libjpeg"
          "${MINGW_PACKAGE_PREFIX}-libpng"
-         "${MINGW_PACKAGE_PREFIX}-LibRaw"
+         "${MINGW_PACKAGE_PREFIX}-LibRaw-gpl2"
          "${MINGW_PACKAGE_PREFIX}-libwebp"
          "${MINGW_PACKAGE_PREFIX}-libtiff"
          "${MINGW_PACKAGE_PREFIX}-opencolorio"
-         "${MINGW_PACKAGE_PREFIX}-opencv"
          "${MINGW_PACKAGE_PREFIX}-openexr"
          "${MINGW_PACKAGE_PREFIX}-openjpeg"
-         "${MINGW_PACKAGE_PREFIX}-openssl"
          "${MINGW_PACKAGE_PREFIX}-zlib")
 makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
              #"${MINGW_PACKAGE_PREFIX}-qt5"
@@ -40,14 +36,24 @@
         0001-MinGW-w64-include-winbase-h-early-for-TCHAR-types.patch
         0003-boost-latest.patch
         0004-guard-msvc-stuff.patch
-        0005-fix-initialization.patch)
+        0005-fix-initialization.patch
+        oiio-1.5.23-invalidatespec.patch
+        oiio-1.7.10-c99math.patch
+        oiio-1.7.10-nothreads.patch
+        oiio-1.7.12-tiffposition.patch)
 sha256sums=('669c59d06399dd882c3e2469bb3a4b33d537db150f6fe056495c3429b5c3838f'
             '138b21e2c21fc43023f5788fa107f904454fda7812d1ab4f762de18b323b8649'
             'ab6bde02c0c05495a9cb13441535d42f4d5929bd930859d75f589d12be9a182b'
             'b7cc78d0f535ceb1aba68b11970c32ffbf04815d229b12e64e2c2c0b3403b4b2'
             'ac7039e5e9ba7e1a4e92d5f674282d270d0dee7afd96d3899f12544a0f6392d1'
             '3a73e95486a5a50685eed753f11c433bcf9f1b1a46947d6ae1350c8f68568536'
-            'bddb6c328e13d7b53e511acba4e21cb6a3b2a09a961c34a566bf38c84622fed4')
+            'bddb6c328e13d7b53e511acba4e21cb6a3b2a09a961c34a566bf38c84622fed4'
+            'skip'
+            'skip'
+            'skip'
+            'skip'
+            'skip'
+            'skip')
 
 prepare() {
   cd ${srcdir}/oiio-Release-${pkgver}
@@ -57,6 +63,10 @@
   patch -p1 -i ${srcdir}/0003-boost-latest.patch
   patch -p1 -i ${srcdir}/0004-guard-msvc-stuff.patch
   patch -p1 -i ${srcdir}/0005-fix-initialization.patch
+  patch -p1 -i ${srcdir}/oiio-1.5.23-invalidatespec.patch
+  patch -p1 -i ${srcdir}/oiio-1.7.10-nothreads.patch
+  patch -p1 -i ${srcdir}/oiio-1.7.10-c99math.patch
+  patch -p1 -i ${srcdir}/oiio-1.7.12-tiffposition.patch
 }
 
 build() {
@@ -72,13 +82,25 @@
     -G"MSYS Makefiles" \
     -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
     -DPYLIB_INSTALL_DIR=lib/python2.7/site-packages \
+    -DUSE_QT=OFF \
+    -DUSE_FIELD3D=OFF \
+    -DUSE_FFMPEG=OFF \
+    -DUSE_OPENJPEG=ON \
+    -DUSE_OPENCV=OFF \
+    -DUSE_OPENSSL=OFF \
     -DUSE_FREETYPE=ON \
-    -DUSE_OPENCV=ON \
-    -DUSE_FIELD3D=ON \
-    -DUSE_OPENSSL=ON \
+    -DUSE_GIF=ON \
+    -DUSE_LIBRAW=ON \
+    -DUSE_NUKE=OFF \
+    -DSTOP_ON_WARNING=OFF \
+    -DOPENEXR_HOME=${MINGW_PREFIX} \
+    -DILMBASE_HOME=${MINGW_PREFIX} \
+    -DBOOST_ROOT=${MINGW_PREFIX} \
+    -DTHIRD_PARTY_TOOLS_HOME=${MINGW_PREFIX} \
     -DFREETYPE_INCLUDE_PATH=${MINGW_PREFIX}/include/freetype2 \
-    -DOPENJPEG_INCLUDE_DIR=${MINGW_PREFIX}/include/openjpeg-2.2 \
-    -DOPENJPEG_OPENJPEG_LIBRARIES=${MINGW_PREFIX}/lib/libopenjp2.dll.a \
+    -DOPENJPEG_HOME=${MINGW_PREFIX} \
+    -DOPENJPEG_INCLUDE_DIR=`pkg-config --variable=includedir libopenjp2` \
+    -DLIBRAW_PATH=${MINGW_PREFIX}/libraw-gpl2 \
     -DCMAKE_SHARED_LINKER_FLAGS=" -Wl,--export-all-symbols -Wl,--enable-auto-import " \
     -DOIIO_BUILD_TESTS=ON \
     ../oiio-Release-${pkgver}
