--- Portfile.orig	2021-10-25 20:02:31.000000000 -0700
+++ Portfile	2021-10-25 20:06:45.000000000 -0700
@@ -1,7 +1,7 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem              1.0
-PortGroup               cmake 1.1
+PortGroup               cmake 1.0
 if {${os.major} <= 17} {
     PortGroup           muniversal 1.0
 }
@@ -30,17 +30,17 @@
 
 if {${os.platform} eq "darwin" && ${configure.cxx_stdlib} ne "libstdc++"} {
     if { ${subport} eq "libomp-devel" } {
-        version         13.0.0
-        checksums       rmd160  1ecfcc0f1d18e208f99ce8390281ed6b5f92a28e \
-                        sha256  4930ae7a1829a53b698255c2c6b6ee977cc364b37450c14ee458793c0d5e493c \
-                        size    1139488
-        livecheck.regex {"llvmorg-([0-9.rc-]+)".*}
+        version                 11.1.0
+        checksums               rmd160  8869b60844901466aed4e462ae8dac831bf519c3 \
+                                sha256  d187483b75b39acb3ff8ea1b7d98524d95322e3cb148842957e9b0fbb866052e \
+                                size    971628
+        livecheck.regex         {"llvmorg-([0-9.rc-]+)".*}
     } else {
-        version         13.0.0
-        checksums       rmd160  1ecfcc0f1d18e208f99ce8390281ed6b5f92a28e \
-                        sha256  4930ae7a1829a53b698255c2c6b6ee977cc364b37450c14ee458793c0d5e493c \
-                        size    1139488
-        livecheck.regex {"llvmorg-([0-9.]+)".*}
+        version                 11.1.0
+        checksums               rmd160  8869b60844901466aed4e462ae8dac831bf519c3 \
+                                sha256  d187483b75b39acb3ff8ea1b7d98524d95322e3cb148842957e9b0fbb866052e \
+                                size    971628
+        livecheck.regex         {"llvmorg-([0-9.]+)".*}
     }
 
     if {${os.major} <= 12} {
@@ -53,34 +53,38 @@
     configure.cxxflags-append -std=c++11
     compiler.cxx_standard   2011
 
-    master_sites        https://github.com/llvm/llvm-project/releases/download/llvmorg-${version} \
-                        https://releases.llvm.org/${version}
-    distname            openmp-[strsed ${version} {s/-//}].src
-    use_xz              yes
-    dist_subdir         openmp-release
-    worksrcdir          ${distname}
-    set rtpath          "runtime/"
+    master_sites \
+        https://github.com/llvm/llvm-project/releases/download/llvmorg-${version} \
+        https://releases.llvm.org/${version}
+    distname                openmp-[strsed ${version} {s/-//}].src
+    use_xz                  yes
+    dist_subdir             openmp-release
+    worksrcdir              ${distname}
+    set rtpath              "runtime/"
     
-    patchfiles-append   patch-libomp-use-gettid-on-Leopard.diff
+    # D88252 is to enable arm64 support
+    patchfiles-append       patch-libomp-use-gettid-on-Leopard.diff \
+                            reviews.llvm.org_D88252.diff
 
-    livecheck.url       https://api.github.com/repos/llvm/llvm-project/tags
+    livecheck.url           https://api.github.com/repos/llvm/llvm-project/tags
 } else {
     if { ${subport} eq "libomp-devel" } {
-        version         291764
-        replaced_by     libomp
+        version                 291764
+        replaced_by             libomp
     } else {
-        PortGroup       github 1.0
+        PortGroup               github 1.0
         # Last version working on libstdc++
-        github.setup    llvm-mirror openmp 381 svn-tags/RELEASE_
-        name            libomp
+        github.setup            llvm-mirror openmp 381 svn-tags/RELEASE_
+        name                    libomp
 
-        worksrcdir      openmp-${version}/final/runtime
-        version         3.8.1
-        checksums       rmd160  a41054068a127ef84610afef8090109078cb6c46 \
-                        sha256  4c46b5946fe9b2a701661746d11c7c85c51a7f18673194a7ebd2a43470948a34
-        set rtpath      "./"
+        worksrcdir              openmp-${version}/final/runtime
+        version                 3.8.1
+        checksums \
+            rmd160  a41054068a127ef84610afef8090109078cb6c46 \
+            sha256  4c46b5946fe9b2a701661746d11c7c85c51a7f18673194a7ebd2a43470948a34
+        set rtpath              "./"
     }
-    livecheck.type      none
+    livecheck.type          none
 }
 
 depends_build-append    port:perl5
@@ -139,7 +143,7 @@
     }
 
     set fpath ${worksrcpath}/${rtpath}
-    file copy ${fpath}/README.txt ${fpath}/../LICENSE.TXT \
+    file copy ${fpath}/README.txt ${fpath}/../LICENSE.txt \
         ${fpath}/../www ${instdest}/share/doc/libomp/
 
 }
