--- Portfile.orig	2022-01-30 14:36:18.000000000 -0800
+++ Portfile	2022-01-30 14:39:37.000000000 -0800
@@ -1,7 +1,7 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem              1.0
-PortGroup               cmake 1.1
+PortGroup               cmake 1.0
 if {${os.major} <= 17} {
     PortGroup           muniversal 1.0
 }
@@ -30,21 +30,19 @@
 
 if {${os.platform} eq "darwin" && ${configure.cxx_stdlib} ne "libstdc++"} {
     if { ${subport} eq "libomp-devel" } {
-        version         13.0.0
-        checksums       rmd160  1ecfcc0f1d18e208f99ce8390281ed6b5f92a28e \
-                        sha256  4930ae7a1829a53b698255c2c6b6ee977cc364b37450c14ee458793c0d5e493c \
-                        size    1139488
-        livecheck.regex {"llvmorg-([0-9.rc-]+)".*}
+        version                 11.1.0
+        checksums               rmd160  8869b60844901466aed4e462ae8dac831bf519c3 \
+                                sha256  d187483b75b39acb3ff8ea1b7d98524d95322e3cb148842957e9b0fbb866052e \
+                                size    971628
+        livecheck.regex         {"llvmorg-([0-9.rc-]+)".*}
     } else {
-        version         13.0.0
-        checksums       rmd160  1ecfcc0f1d18e208f99ce8390281ed6b5f92a28e \
-                        sha256  4930ae7a1829a53b698255c2c6b6ee977cc364b37450c14ee458793c0d5e493c \
-                        size    1139488
-        livecheck.regex {"llvmorg-([0-9.]+)".*}
+        version                 11.1.0
+        checksums               rmd160  8869b60844901466aed4e462ae8dac831bf519c3 \
+                                sha256  d187483b75b39acb3ff8ea1b7d98524d95322e3cb148842957e9b0fbb866052e \
+                                size    971628
+        livecheck.regex         {"llvmorg-([0-9.]+)".*}
     }
 
-    revision            1
-
     if {${os.major} <= 12} {
         # kmp_alloc.c includes <atomic> but libc++ is not the default on
         # these systems. https://trac.macports.org/ticket/52554
@@ -55,34 +53,38 @@
     configure.cxxflags-append -std=c++11
     compiler.cxx_standard   2011
 
-    master_sites        https://github.com/llvm/llvm-project/releases/download/llvmorg-${version} \
-                        https://releases.llvm.org/${version}
-    distname            openmp-[strsed ${version} {s/-//}].src
-    use_xz              yes
-    dist_subdir         openmp-release
-    worksrcdir          ${distname}
-    set rtpath          "runtime/"
-
-    patchfiles-append   patch-libomp-use-gettid-on-Leopard.diff
+    master_sites \
+        https://github.com/llvm/llvm-project/releases/download/llvmorg-${version} \
+        https://releases.llvm.org/${version}
+    distname                openmp-[strsed ${version} {s/-//}].src
+    use_xz                  yes
+    dist_subdir             openmp-release
+    worksrcdir              ${distname}
+    set rtpath              "runtime/"
+    
+    # D88252 is to enable arm64 support
+    patchfiles-append       patch-libomp-use-gettid-on-Leopard.diff \
+                            reviews.llvm.org_D88252.diff
 
-    livecheck.url       https://api.github.com/repos/llvm/llvm-project/tags
+    livecheck.url           https://api.github.com/repos/llvm/llvm-project/tags
 } else {
     if { ${subport} eq "libomp-devel" } {
-        version         291764
-        replaced_by     libomp
+        version                 291764
+        replaced_by             libomp
     } else {
-        PortGroup       github 1.0
+        PortGroup               github 1.0
         # Last version working on libstdc++
-        github.setup    llvm-mirror openmp 381 svn-tags/RELEASE_
-        name            libomp
+        github.setup            llvm-mirror openmp 381 svn-tags/RELEASE_
+        name                    libomp
 
-        worksrcdir      openmp-${version}/final/runtime
-        version         3.8.1
-        checksums       rmd160  a41054068a127ef84610afef8090109078cb6c46 \
-                        sha256  4c46b5946fe9b2a701661746d11c7c85c51a7f18673194a7ebd2a43470948a34
-        set rtpath      "./"
+        worksrcdir              openmp-${version}/final/runtime
+        version                 3.8.1
+        checksums \
+            rmd160  a41054068a127ef84610afef8090109078cb6c46 \
+            sha256  4c46b5946fe9b2a701661746d11c7c85c51a7f18673194a7ebd2a43470948a34
+        set rtpath              "./"
     }
-    livecheck.type      none
+    livecheck.type          none
 }
 
 depends_build-append    port:perl5
@@ -108,14 +110,8 @@
 # With this, cmake sets the correct library name in the dylibs for the
 # final destination we move them to
 configure.args-append   -DCMAKE_INSTALL_RPATH=${prefix}/lib/libomp \
-                        -DCMAKE_INSTALL_NAME_DIR=${prefix}/lib/libomp
-
-if {${os.major} <= 10} {
-    set hnames              {omp.h}
-    configure.args-append   -DLIBOMP_OMPT_SUPPORT=FALSE
-} else {
-    set hnames              {omp-tools.h omp.h ompt.h}
-}
+                        -DCMAKE_INSTALL_NAME_DIR=${prefix}/lib/libomp \
+                        -DLIBOMP_OMPT_SUPPORT=FALSE
 
 variant top_level description \
     "Install (links to) omp.h and libs into ${prefix}/(include|lib)" {}
@@ -129,22 +125,17 @@
 post-destroot {
     set instdest ${destroot}${prefix}
     xinstall -d ${instdest}/share/doc/libomp
-
     xinstall -d ${instdest}/include/libomp
-    foreach h ${hnames} {
-        move ${instdest}/tmp/include/${h} ${instdest}/include/libomp/
-    }
-
+    move ${instdest}/tmp/include/omp.h ${instdest}/include/libomp/
     xinstall -d ${instdest}/lib/libomp
+
     foreach p {libiomp5.dylib libomp.dylib libgomp.dylib} {
         move ${instdest}/tmp/lib/${p} ${instdest}/lib/libomp/
     }
 
     if {[variant_isset top_level]} {
-        foreach h ${hnames} {
-            system -W ${instdest}/include \
-              "ln -s libomp/${h}"
-        }
+        system -W ${instdest}/include \
+          "ln -s libomp/omp.h"
         foreach p {libiomp5.dylib libomp.dylib libgomp.dylib} {
             system -W ${instdest}/lib/ \
               "ln -s libomp/${p}"
@@ -152,7 +143,7 @@
     }
 
     set fpath ${worksrcpath}/${rtpath}
-    file copy ${fpath}/README.txt ${fpath}/../LICENSE.TXT \
+    file copy ${fpath}/README.txt ${fpath}/../LICENSE.txt \
         ${fpath}/../www ${instdest}/share/doc/libomp/
 
 }
