diff -Naur libvpx-1.8.0.orig/configure libvpx-1.8.0/configure
--- libvpx-1.8.0.orig/configure	2019-04-04 20:21:43.685074600 -0400
+++ libvpx-1.8.0/configure	2019-04-04 20:23:43.924370800 -0400
@@ -265,7 +265,6 @@
 HAVE_LIST="
     ${ARCH_EXT_LIST}
     vpx_ports
-    pthread_h
     unistd_h
 "
 EXPERIMENT_LIST="
@@ -582,11 +581,6 @@
 
     # Use both check_header and check_lib here, since check_lib
     # could be a stub that always returns true.
-    check_header pthread.h && check_lib -lpthread <<EOF || disable_feature pthread_h
-#include <pthread.h>
-#include <stddef.h>
-int main(void) { return pthread_create(NULL, NULL, NULL, NULL); }
-EOF
     check_header unistd.h # for sysconf(3) and friends.
 
     check_header vpx/vpx_integer.h -I${source_path} && enable_feature vpx_ports
diff -Naur libvpx-1.8.0.orig/vp8/common/generic/systemdependent.c libvpx-1.8.0/vp8/common/generic/systemdependent.c
--- libvpx-1.8.0.orig/vp8/common/generic/systemdependent.c	2019-04-04 20:21:43.903881700 -0400
+++ libvpx-1.8.0/vp8/common/generic/systemdependent.c	2019-04-04 20:24:40.116494200 -0400
@@ -23,7 +23,8 @@
 #if CONFIG_MULTITHREAD
 #if HAVE_UNISTD_H && !defined(__OS2__)
 #include <unistd.h>
-#elif defined(_WIN32)
+#endif
+#if defined(_WIN32)
 #include <windows.h>
 typedef void(WINAPI *PGNSI)(LPSYSTEM_INFO);
 #elif defined(__OS2__)
