--- misc/Makefile.orig	2015-01-10 20:43:30.000000000 -0800
+++ misc/Makefile	2015-01-10 20:51:27.000000000 -0800
@@ -201,14 +201,18 @@ indr.NEW: indr.o
 		$(OFILE_DIR)/indr.private.o
 	$(DSYMUTIL) $(SYMROOT)/indr.NEW
 
-strip.NEW: strip.o
+strip.NEW: strip.o PruneTrie.o
 	$(CC) $(RC_CFLAGS) -nostdlib -r \
 		-o $(OBJROOT)/strip.private.o \
 		$(OFILE_DIR)/strip.o $(LIBSTUFF)
 	$(CXX) $(RC_CFLAGS) $(SDK) -o $(SYMROOT)/strip.NEW \
-		$(OFILE_DIR)/strip.private.o $(LIB_PRUNETRIE) $(CXXLIB)
+		$(OFILE_DIR)/strip.private.o $(OFILE_DIR)/PruneTrie.o $(CXXLIB)
 	$(DSYMUTIL) $(SYMROOT)/strip.NEW
 
+PruneTrie.o: PruneTrie.cpp
+	$(CXX) $(CXXFLAGS) $(RC_CFLAGS) $(SDK) $(CXXLIB) -c -o \
+		$(OBJROOT)/PruneTrie.o $(SRCROOT)/PruneTrie.cpp
+
 nmedit.NEW: nmedit.o
 	$(CC) $(RC_CFLAGS) -nostdlib -r \
 		-o $(OBJROOT)/nmedit.private.o \
--- misc/strip.c.orig	2015-01-10 21:25:17.000000000 -0800
+++ misc/strip.c	2015-01-10 21:25:40.000000000 -0800
@@ -47,7 +47,7 @@
 #include "stuff/unix_standard_mode.h"
 #include "stuff/execute.h"
 #ifdef TRIE_SUPPORT
-#include <mach-o/prune_trie.h>
+#include "prune_trie.h"
 #endif /* TRIE_SUPPORT */
 
 /* These are set from the command line arguments */
