--- a/config/Imath.pc.in
+++ b/config/Imath.pc.in
@@ -7,11 +7,11 @@
 exec_prefix=${prefix}
 libdir=@PKG_CONFIG_INSTALL_LIBDIR@
 includedir=@PKG_CONFIG_INSTALL_INCLUDEDIR@
-libsuffix=@LIB_SUFFIX_DASH@
+
 Name: Imath
 Description: Imath library: vector/matrix and math operations, plus the half type.
 Version: @IMATH_VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} -lImath${libsuffix}
+Libs: -L${libdir} -lImath
 Cflags: -I${includedir} -I${includedir}/Imath
+++ b/src/Imath/CMakeLists.txt
--- a/src/Imath/CMakeLists.txt
@@ -146,7 +146,8 @@
       COMMENT "Copying resource files to Imath.framework/Resources in build directory"
     )
 else()
-    set_target_properties(${IMATH_LIBRARY} PROPERTIES OUTPUT_NAME "Imath${IMATH_LIB_SUFFIX}" )
+    set_target_properties(${IMATH_LIBRARY} PROPERTIES OUTPUT_NAME "Imath"
+                                                      RUNTIME_OUTPUT_NAME "Imath${IMATH_LIB_SUFFIX}" )
 endif()
   
 #
--- a/src/python/PyImath/CMakeLists.txt
+++ b/src/python/PyImath/CMakeLists.txt
@@ -123,7 +123,8 @@
     MACOSX_RPATH TRUE)
 else()
   set_target_properties(${PYIMATH_LIBRARY} PROPERTIES
-    OUTPUT_NAME "${PYIMATH_LIBRARY}${PYIMATH_LIB_SUFFIX}"
+    OUTPUT_NAME "${PYIMATH_LIBRARY}_Python${Python3_VERSION_MAJOR}_${Python3_VERSION_MINOR}"
+    RUNTIME_OUTPUT_NAME "${PYIMATH_LIBRARY}${PYIMATH_LIB_SUFFIX}"
     SOVERSION ${IMATH_LIB_SOVERSION}
     VERSION ${IMATH_LIB_VERSION}
     POSITION_INDEPENDENT_CODE ON
--- a/src/python/PyImath/PyImath.pc.in
+++ b/src/python/PyImath/PyImath.pc.in
@@ -7,8 +7,6 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-libsuffix=@lib_suffix@
-pylibsuffix=@pylib_suffix@
 
 boost_includedir=@Boost_INCLUDE_DIRS@
 boost_libdir=@boost_libdir@
@@ -21,5 +19,5 @@
 Name: PyImath
 Description: Python wrappings of Imath classes/functions
 Version: @IMATH_VERSION@
-Libs: -L${libdir} -lImath${libsuffix} -lPyImath${pylibsuffix} -L${boost_libdir} -l${boost_libname} -L${python_libdir} -l${python_libname}
+Libs: -L${libdir} -lImath -lPyImath_Python@Python3_VERSION_MAJOR@_@Python3_VERSION_MINOR@ -L${boost_libdir} -l${boost_libname} -L${python_libdir} -l${python_libname}
 Cflags: -I${includedir} -I${includedir}/@IMATH_OUTPUT_SUBDIR@ -I${includedir}/@PYIMATH_OUTPUT_SUBDIR@ -I${boost_includedir} -I${python_includedir} 
