--- a/src/Imath/ImathVec.h
+++ b/src/Imath/ImathVec.h
@@ -873,7 +873,7 @@
 //----------------------------------------------------------------------------
 
 /// @cond Doxygen_Suppress
-
+/*
 // Vec2<short>
 template <> IMATH_HOSTDEVICE short Vec2<short>::length() const IMATH_NOEXCEPT = delete;
 template <> IMATH_HOSTDEVICE const Vec2<short>& Vec2<short>::normalize() IMATH_NOEXCEPT = delete;
@@ -954,7 +954,7 @@
 template <> IMATH_HOSTDEVICE Vec4<int64_t> Vec4<int64_t>::normalized() const IMATH_NOEXCEPT = delete;
 template <> Vec4<int64_t> Vec4<int64_t>::normalizedExc() const = delete;
 template <> IMATH_HOSTDEVICE Vec4<int64_t> Vec4<int64_t>::normalizedNonNull() const IMATH_NOEXCEPT = delete;
-
+*/
 /// @endcond Doxygen_Suppress
 
 //------------------------
