diff -rubw libpyside/PySideConfig-spec.cmake.in libpyside/PySideConfig-spec.cmake.in
--- libpyside/PySideConfig-spec.cmake.in	2014-04-22 10:45:29.000000000 -0500
+++ libpyside/PySideConfig-spec.cmake.in	2014-07-05 21:38:17.000000000 -0500
@@ -3,7 +3,7 @@
 #  PYSIDE_PYTHONPATH    - Path to where the PySide Python module files could be found
 #  PYSIDE_TYPESYSTEMS   - Type system files that should be used by other bindings extending PySide
 
-SET(PYSIDE_INCLUDE_DIR "@CMAKE_INSTALL_PREFIX@/include/PySide@pyside_SUFFIX@")
+SET(PYSIDE_INCLUDE_DIR "@CMAKE_INSTALL_PREFIX@/include/PySide-@PYTHONBRANCH@")
 # Platform specific library names
 if(MSVC)
     SET(PYSIDE_LIBRARY "@LIB_INSTALL_DIR@/@CMAKE_SHARED_LIBRARY_PREFIX@pyside@pyside_SUFFIX@@LIBRARY_OUTPUT_SUFFIX@@SHIBOKEN_PYTHON_SUFFIX@.lib")
@@ -15,4 +15,4 @@
     SET(PYSIDE_LIBRARY "@LIB_INSTALL_DIR@/@CMAKE_SHARED_LIBRARY_PREFIX@pyside@pyside_SUFFIX@@LIBRARY_OUTPUT_SUFFIX@@SHIBOKEN_PYTHON_SUFFIX@@CMAKE_SHARED_LIBRARY_SUFFIX@")
 endif()
 SET(PYSIDE_PYTHONPATH "@SITE_PACKAGE@")
-SET(PYSIDE_TYPESYSTEMS "@CMAKE_INSTALL_PREFIX@/share/PySide@pyside_SUFFIX@/typesystems")
+SET(PYSIDE_TYPESYSTEMS "@CMAKE_INSTALL_PREFIX@/share/PySide-@PYTHONBRANCH@/typesystems")
diff -rubw libpyside/PySideConfig.cmake.in libpyside/PySideConfig.cmake.in
--- libpyside/PySideConfig.cmake.in	2014-04-22 10:45:29.000000000 -0500
+++ libpyside/PySideConfig.cmake.in	2014-07-05 21:38:11.000000000 -0500
@@ -2,4 +2,4 @@
     message(STATUS "Using default python: @SHIBOKEN_PYTHON_SUFFIX@")
     SET(PYTHON_BASENAME @SHIBOKEN_PYTHON_SUFFIX@)
 endif()
-include(@LIB_INSTALL_DIR@/cmake/PySide-@BINDING_API_VERSION@/PySideConfig${PYTHON_BASENAME}.cmake)
+include(@PYTHONPREFIX@/lib/cmake/PySide-@BINDING_API_VERSION@/PySideConfig${PYTHON_BASENAME}.cmake)
diff -rubw libpyside/pyside.pc.in libpyside/pyside.pc.in
--- libpyside/pyside.pc.in	2014-04-22 10:45:29.000000000 -0500
+++ libpyside/pyside.pc.in	2014-07-05 21:38:07.000000000 -0500
@@ -1,8 +1,8 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@LIB_INSTALL_DIR@
-includedir=@CMAKE_INSTALL_PREFIX@/include/PySide@pyside_SUFFIX@
-typesystemdir=@CMAKE_INSTALL_PREFIX@/share/PySide@pyside_SUFFIX@/typesystems
+includedir=@CMAKE_INSTALL_PREFIX@/include/PySide-@PYTHONBRANCH@
+typesystemdir=@CMAKE_INSTALL_PREFIX@/share/PySide-@PYTHONBRANCH@/typesystems
 pythonpath=@SITE_PACKAGE@
 
 Name: PySide@pyside_SUFFIX@
