From 54aa9c8f7cf32337c021ccc5b4c9d96eba02c7cb Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Fri, 16 Jan 2015 21:11:37 -0800
Subject: [PATCH 3002/3005] buildit: Set compatibility version to
 RC_ProjectSourceVersion

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 lib/buildit | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git llvm_master/projects/libcxx/lib/buildit macports_master/projects/libcxx/lib/buildit
index 237e5a7..d313efe 100755
--- llvm_master/projects/libcxx/lib/buildit
+++ macports_master/projects/libcxx/lib/buildit
@@ -54,7 +54,8 @@ case $TRIPLE in
     then
         EXTRA_FLAGS="-nostdinc++ -std=c++11 -U__STRICT_ANSI__"
         LDSHARED_FLAGS="-o libc++.1.dylib \
-            -dynamiclib -nodefaultlibs -current_version 1 \
+            -dynamiclib -nodefaultlibs \
+            -current_version ${RC_ProjectSourceVersion} \
             -compatibility_version 1 \
             -install_name /usr/lib/libc++.1.dylib \
             -Wl,-reexport_library,/usr/lib/libc++abi.dylib \
-- 
2.10.1

