From fe1df1c458109ae896067ea3cc1ae79e176b40fb Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Tue, 20 Dec 2016 01:08:20 -0800
Subject: [PATCH 4001/4002] https://llvm.org/bugs/show_bug.cgi?id=31433

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 cmake/modules/AddLLDB.cmake | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git llvm_release_40/tools/lldb/cmake/modules/AddLLDB.cmake macports_release_40/tools/lldb/cmake/modules/AddLLDB.cmake
index 47ee25da5..97a22d726 100644
--- llvm_release_40/tools/lldb/cmake/modules/AddLLDB.cmake
+++ macports_release_40/tools/lldb/cmake/modules/AddLLDB.cmake
@@ -152,6 +152,12 @@ function(add_lldb_executable name)
                                 -DCMAKE_INSTALL_COMPONENT=${name}
                                 -P "${CMAKE_BINARY_DIR}/cmake_install.cmake")
     endif()
+  else()
+    if(ARG_GENERATE_INSTALL)
+      install(TARGETS ${name}
+            COMPONENT ${name}
+            RUNTIME DESTINATION ${install_dir})
+    endif()
   endif()
 
   if(ARG_INCLUDE_IN_FRAMEWORK AND LLDB_BUILD_FRAMEWORK)
-- 
2.13.3

