--- Portfile.orig	2017-10-04 10:22:55.000000000 +0200
+++ Portfile	2017-10-04 10:24:50.000000000 +0200
@@ -1,19 +1,9 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem 1.0
-PortGroup  cxx11 1.1
-
-# NOTE: As of cppunit 1.14.0 dated April 13, 2017, the configure
-# script now checks internally to make sure the requested compiler is
-# C++11 compliant, and will error out if not. So, all we need to do
-# here is to use the cxx11 PG to make sure the compiler is compliant.
-
-# NOTE: CppUnit's public API also required C++11 for compiling, hence
-# any project using CppUnit must be compiled with C++11 support,
-# regardless of whether the project itself requires C++11.
 
 name                cppunit
-version             1.14.0
+version             1.13.2
 categories          devel
 platforms           darwin
 maintainers         {michaelld @michaelld} openmaintainer
@@ -26,8 +16,12 @@
 homepage            https://www.freedesktop.org/wiki/Software/cppunit/
 master_sites        http://dev-www.libreoffice.org/src/
 
-checksums           rmd160  9d2bec93f3cee98b95430b60d801f5ed886688aa \
-                    sha256  3d569869d27b48860210c758c4f313082103a5e58219a7669b52bfd29d674780
+checksums           rmd160  2a86e06409b53b3477842354b21dd17fc4943b09 \
+                    sha256  3f47d246e3346f2ba4d7c9e882db3ad9ebd3fcbd2e8b732f946e0e3eeb9f429f
+
+post-extract {
+    move ${worksrcpath}/configure.in ${worksrcpath}/configure.ac
+}
 
 configure.args      --disable-dot \
                     --disable-doxygen \
