From 90de52ef5856c276848dc2c85dd4985f9e2639f4 Mon Sep 17 00:00:00 2001
From: Stephen <stephen.kiazyk@thinkboxsoftware.com>
Date: Wed, 24 Feb 2016 14:32:49 -0600
Subject: [PATCH] Add missing OpenEXR headers to CMake install rule

---
 IlmImf/CMakeLists.txt | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/OpenEXR/IlmImf/CMakeLists.txt b/OpenEXR/IlmImf/CMakeLists.txt
index 18d90ed..11f7a1a 100644
--- a/OpenEXR/IlmImf/CMakeLists.txt
+++ b/OpenEXR/IlmImf/CMakeLists.txt
@@ -350,6 +350,29 @@ INSTALL ( FILES
     ImfDeepImageState.h
     ImfDeepImageStateAttribute.h
     ImfFloatVectorAttribute.h
+    ImfAutoArray.h
+    ImfCheckedArithmetic.h
+    ImfCompressor.h
+    ImfDwaCompressor.h
+    ImfDwaCompressorSimd.h
+    ImfFastHuf.h
+    ImfInputPartData.h
+    ImfInputStreamMutex.h
+    ImfOptimizedPixelReading.h
+    ImfOutputPartData.h
+    ImfOutputStreamMutex.h
+    ImfPizCompressor.h
+    ImfPxr24Compressor.h
+    ImfRle.h
+    ImfRleCompressor.h
+    ImfScanLineInputFile.h
+    ImfSimd.h
+    ImfStdIO.h
+    ImfSystemSpecific.h
+    ImfTiledMisc.h
+    ImfTileOffsets.h
+    ImfZip.h
+    ImfZipCompressor.h
   DEPENDENCIES
     OpenEXR::Config
     IlmBase::Iex
