--- openexr-2.4.0/IlmBase/config/IlmBaseSetup.cmake.orig	2019-10-29 15:34:57.792136700 +0300
+++ openexr-2.4.0/IlmBase/config/IlmBaseSetup.cmake	2019-10-29 15:35:30.172182000 +0300
@@ -31,7 +31,7 @@
 set(ILMBASE_PACKAGE_NAME "IlmBase ${ILMBASE_VERSION}" CACHE STRING "Public string / label for displaying package")
 
 # Whether to generate and install a pkg-config file IlmBase.pc on
-if(WIN32)
+if(MSVC)
 option(ILMBASE_INSTALL_PKG_CONFIG "Install IlmBase.pc file" OFF)
 else()
 option(ILMBASE_INSTALL_PKG_CONFIG "Install IlmBase.pc file" ON)
--- openexr-2.4.0/IlmBase/config/LibraryDefine.cmake.orig	2019-10-29 15:37:25.812343900 +0300
+++ openexr-2.4.0/IlmBase/config/LibraryDefine.cmake	2019-10-29 15:38:12.164408900 +0300
@@ -84,7 +84,9 @@
     )
   endif()
   set_target_properties(${libname} PROPERTIES
-      OUTPUT_NAME "${libname}${ILMBASE_LIB_SUFFIX}"
+      OUTPUT_NAME "${libname}"
+      RUNTIME_OUTPUT_NAME "${libname}${ILMBASE_LIB_SUFFIX}"
+      ARCHIVE_OUTPUT_NAME "${libname}"
       RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
   )
   add_library(${PROJECT_NAME}::${libname} ALIAS ${libname})
--- openexr-2.4.0/OpenEXR/config/OpenEXRSetup.cmake.orig	2019-10-29 15:39:38.034529100 +0300
+++ openexr-2.4.0/OpenEXR/config/OpenEXRSetup.cmake	2019-10-29 15:39:43.874537300 +0300
@@ -21,7 +21,7 @@
 set(OPENEXR_PACKAGE_NAME "IlmBase ${ILMBASE_VERSION}" CACHE STRING "Public string / label for displaying package")
 
 # Whether to generate and install a pkg-config file OpenEXR.pc
-if (WIN32)
+if (MSVC)
 option(OPENEXR_INSTALL_PKG_CONFIG "Install OpenEXR.pc file" OFF)
 else()
 option(OPENEXR_INSTALL_PKG_CONFIG "Install OpenEXR.pc file" ON)
--- openexr-2.4.0/oPENexr/config/LibraryDefine.cmake.orig	2019-10-29 15:38:56.304470700 +0300
+++ openexr-2.4.0/oPENexr/config/LibraryDefine.cmake	2019-10-29 15:39:22.864507900 +0300
@@ -78,7 +78,9 @@
     )
   endif()
   set_target_properties(${libname} PROPERTIES
-      OUTPUT_NAME "${libname}${OPENEXR_LIB_SUFFIX}"
+      OUTPUT_NAME "${libname}"
+      RUNTIME_OUTPUT_NAME "${libname}${OPENEXR_LIB_SUFFIX}"
+      ARCHIVE_OUTPUT_NAME "${libname}"
       RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
   )
   add_library(${PROJECT_NAME}::${libname} ALIAS ${libname})
--- openexr-2.4.0/IlmBase/IlmBase.pc.in.orig	2019-10-29 16:09:29.326136000 +0300
+++ openexr-2.4.0/IlmBase/IlmBase.pc.in	2019-10-29 16:09:50.616346500 +0300
@@ -13,5 +13,5 @@
 Version: @ILMBASE_VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} -lImath${libsuffix} -lHalf${libsuffix} -lIex${libsuffix} -lIexMath${libsuffix} -lIlmThread${libsuffix} @PTHREAD_LIBS@
+Libs: -L${libdir} -lImath -lHalf -lIex -lIexMath -lIlmThread @PTHREAD_LIBS@
 Cflags: @PTHREAD_CFLAGS@ -I${includedir} -I${includedir}/OpenEXR
--- openexr-2.4.0/OpenEXR/OpenEXR.pc.in.orig	2019-10-29 16:10:13.784661600 +0300
+++ openexr-2.4.0/OpenEXR/OpenEXR.pc.in	2019-10-29 16:10:17.744888100 +0300
@@ -14,7 +14,7 @@
 Description: OpenEXR image library
 Version: @OPENEXR_VERSION@
 
-Libs: -L${libdir} -lIlmImf${libsuffix}
+Libs: -L${libdir} -lIlmImf
 Cflags: -I${OpenEXR_includedir}
 Requires: IlmBase
 Libs.private: -lz
