--- Portfile.orig	2019-12-22 20:23:54.000000000 -0800
+++ Portfile	2019-12-22 20:24:13.000000000 -0800
@@ -21,7 +21,7 @@
 depends_build-append        port:pkgconfig
 
 depends_lib-append          port:gdk-pixbuf2 \
-                            port:jpeg \
+                            path:lib/libjpeg.dylib:jpeg \
                             port:libde265 \
                             port:libpng \
                             port:x265
