--- Portfile.orig	2021-04-01 21:33:34.000000000 -0700
+++ Portfile	2021-04-01 21:33:57.000000000 -0700
@@ -4,7 +4,7 @@
 PortGroup                   github 1.0
 
 github.setup                strukturag libheif 1.11.0 v
-revision                    2
+revision                    102
 
 checksums                   rmd160  e6c17b0c1016c10da0b1aec74fd9ae09c42dbdf7 \
                             sha256  c550938f56ff6dac83702251a143f87cb3a6c71a50d8723955290832d9960913 \
@@ -27,7 +27,6 @@
                             path:include/turbojpeg.h:libjpeg-turbo \
                             port:libde265 \
                             port:libpng \
-                            port:rav1e \
                             port:x265
 
 compiler.cxx_standard       2011
