--- cmake/modules/PopplerMacros.cmake.orig
+++ cmake/modules/PopplerMacros.cmake
@@ -145,7 +145,7 @@ if(CMAKE_COMPILER_IS_GNUCXX)
   endif(GCC_HAS_AS_NEEDED)
 endif (CMAKE_COMPILER_IS_GNUCXX)
 
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
 # set the default compile warnings
   set(_warn "-Wall -Wextra -Wpedantic")
   set(_warn "${_warn} -Wno-unused-parameter")
