--- Portfile.orig	2016-05-09 10:08:47.000000000 +0200
+++ Portfile	2016-05-09 10:09:14.000000000 +0200
@@ -29,7 +29,7 @@
                         sha1    d5b075ccffc0068015f74f78e4bc39138bcfe2d4 \
                         rmd160  530615c061dccdcb64d75c6fab148dd565bb6c91
 
-depends_lib             port:tiff port:jpeg port:zlib
+depends_lib             port:tiff path:lib/libjpeg.dylib:jpeg port:zlib
 
 patchfiles              lp64.patch
 
