--- Portfile.orig	2016-11-10 17:53:21.000000000 +0100
+++ Portfile	2016-11-10 17:53:28.000000000 +0100
@@ -21,7 +21,7 @@
 checksums       rmd160 f082cb4febf1670cb234d03fab6f04a38544855b \
                 sha256 e9c8a241f80d1cc190c308f5efc6669a98776ed27458643553abee823fadd7b3
 
-depends_lib     port:jpeg
+depends_lib     path:lib/libjpeg.dylib:jpeg
 
 configure.args  --disable-opengl --disable-strict
 
