--- src/libtexture/imagecache.cpp.orig	2019-07-09 09:14:23.000000000 +0200
+++ src/libtexture/imagecache.cpp	2019-07-09 09:14:54.000000000 +0200
@@ -350,7 +350,7 @@
 std::shared_ptr<ImageInput>
 ImageCacheFile::get_imageinput(ImageCachePerThreadInfo* thread_info)
 {
-#if defined(__GLIBCXX__) && __GLIBCXX__ < 20160822
+#if (defined(__GLIBCXX__) && __GLIBCXX__ < 20160822) || (defined(_LIBCPP_VERSION) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED < 1090)
     // Older gcc libstdc++ does not properly support std::atomic
     // operations on std::shared_ptr, despite it being a C++11
     // feature. No choice but to lock.
@@ -369,7 +369,7 @@
 {
     if (newval)
         imagecache().incr_open_files();
-#if defined(__GLIBCXX__) && __GLIBCXX__ < 20160822
+#if (defined(__GLIBCXX__) && __GLIBCXX__ < 20160822) || (defined(_LIBCPP_VERSION) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED < 1090)
     // Older gcc libstdc++ does not properly support std::atomic
     // operations on std::shared_ptr, despite it being a C++11
     // feature. No choice but to lock.
