--- src/libtexture/imagecache_pvt.h.orig	2019-05-03 17:56:26.000000000 -0700
+++ src/libtexture/imagecache_pvt.h	2019-05-23 20:53:57.000000000 -0700
@@ -337,7 +337,8 @@
     void invalidate_spec()
     {
         m_validspec = false;
-        m_subimages.clear();
+        //Do not clear the subimages here because another thread might be using it
+        //m_subimages.clear ();
     }
 
     /// Should we print an error message? Keeps track of whether the
