--- oiio-Release-2.3.20.0/src/libutil/strutil.cpp.orig	2022-10-13 23:22:41.000000000 -0700
+++ oiio-Release-2.3.20.0/src/libutil/strutil.cpp	2022-10-13 23:23:32.000000000 -0700
@@ -48,6 +48,20 @@
 #include "stb_sprintf.h"
 
 
+#ifdef __APPLE__
+#include <Availability.h>
+#if __MAC_OS_X_VERSION_MIN_REQUIRED < 1070
+static size_t strnlen(const char *s, size_t maxlen) {
+  size_t l = 0;
+  while (l < maxlen && *s) {
+    l++;
+    s++;
+  }
+  return l;
+}
+#endif
+#endif
+
 OIIO_NAMESPACE_BEGIN
 
 
