diff -ur oiio-Release-1.7.10-exif/src/libOpenImageIO/exif.cpp oiio-Release-1.7.10/src/libOpenImageIO/exif.cpp
--- oiio-Release-1.7.10-exif/src/libOpenImageIO/exif.cpp	2017-01-01 22:55:38.000000000 +0100
+++ oiio-Release-1.7.10/src/libOpenImageIO/exif.cpp	2017-01-26 10:32:51.000000000 +0100
@@ -74,8 +74,8 @@
 #include "OpenImageIO/imageio.h"
 
 
-#define DEBUG_EXIF_READ  1
-#define DEBUG_EXIF_WRITE 1
+#define DEBUG_EXIF_READ  0
+#define DEBUG_EXIF_WRITE 0
 
 OIIO_NAMESPACE_BEGIN
 
