diff -ur oiio-Release-1.8.7.orig/CMakeLists.txt oiio-Release-1.8.7/CMakeLists.txt
--- oiio-Release-1.8.7.orig/CMakeLists.txt	2018-01-02 06:08:01.000000000 +0100
+++ oiio-Release-1.8.7/CMakeLists.txt	2018-01-22 13:48:48.000000000 +0100
@@ -110,6 +110,12 @@
     add_definitions ("-DOIIO_TEXTURE_SIMD_BATCH_WIDTH=${TEX_BATCH_SIZE}")
 endif ()
 
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    # C99 math function declarations are missing from macOS <= 10.6
+    # just make sure <cmath> gets included with the right defines
+    add_definitions("-D_GLIBCXX_USE_C99_MATH_TR1" "-include" "OpenImageIO/missing_math.h")
+endif()
+
 # Set the default namespace
 set (PROJ_NAMESPACE "${${PROJ_NAME}_NAMESPACE}")
 set (PROJ_NAMESPACE_V "${PROJ_NAMESPACE}_v${PROJECT_VERSION_MAJOR}_${PROJECT_VERSION_MINOR}")
diff -ur oiio-Release-1.8.7.orig/src/include/OpenImageIO/missing_math.h oiio-Release-1.8.7/src/include/OpenImageIO/missing_math.h
--- oiio-Release-1.8.7.orig/src/include/OpenImageIO/missing_math.h	2018-01-02 06:08:01.000000000 +0100
+++ oiio-Release-1.8.7/src/include/OpenImageIO/missing_math.h	2018-01-22 13:45:34.000000000 +0100
@@ -38,6 +38,41 @@
 
 #pragma once
 
+// GCC doesn't import C99 math func into the std namespace, because
+// a few declarations are missing.
+// fix from https://github.com/jeremyhu/libcxx/commit/720feba4874d4ca3131753dc31af127e3c509c36
+#ifdef __APPLE__
+#ifndef __APPLE_BAD_MATH_H
+# if __has_include(<Availability.h>)
+#  include <Availability.h>
+#  if __MAC_OS_X_VERSION_MAX_ALLOWED < 1070
+#    define __APPLE_BAD_MATH_H 1
+#  else
+#    define __APPLE_BAD_MATH_H 0
+#  endif
+# else
+#  define __APPLE_BAD_MATH_H 1
+# endif
+#endif
+
+# if __APPLE_BAD_MATH_H
+/* These prototypes are incorrectly omitted from <math.h> on Snow Leopard despite being available */
+extern "C" {
+    extern long long int llrintl(long double);
+    extern long long int llrint(double);
+    extern long long int llrintf(float);
+
+    extern long long int llroundl(long double);
+    extern long long int llround(double);
+    extern long long int llroundf(float);
+}
+# endif
+#if (OIIO_CPLUSPLUS_VERSION >= 11) && !defined(_GLIBCXX_USE_C99_MATH_TR1)
+// import C99 math into std:: in libstdc++ (it won't harm libc++)
+#define _GLIBCXX_USE_C99_MATH_TR1 1
+#endif
+#endif // __APPLE__
+
 #include <cmath>
 
 #include <OpenImageIO/oiioversion.h>
