--- nss-3.17.1/nss/cmd/crmftest/Makefile	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/cmd/crmftest/Makefile	2014-10-03 12:57:19.715776517 +0200
@@ -58,7 +58,7 @@
 LDDIST = $(DIST)/lib
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
-EXTRA_LIBS += $(LDDIST)/sectool.lib
+#EXTRA_LIBS += $(LDDIST)/sectool.lib
 endif
 
 include ../platrules.mk
--- nss-3.17.1/nss/cmd/shlibsign/Makefile	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/cmd/shlibsign/Makefile	2014-10-03 12:58:03.115792965 +0200
@@ -56,8 +56,8 @@
 endif
 CHECKLOC = $(CHECKLIBS:.$(DLL_SUFFIX)=.chk)
 
-MD_LIB_RELEASE_FILES = $(CHECKLOC)
-ALL_TRASH += $(CHECKLOC)
+MD_LIB_RELEASE_FILES = # $(CHECKLOC)
+#ALL_TRASH += $(CHECKLOC)
 endif
 
 #######################################################################
@@ -95,5 +95,5 @@
     endif
 endif
 
-libs install :: $(CHECKLOC)
+libs install :: #$(CHECKLOC)
 
--- nss-3.17.1/nss/coreconf/rules.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/coreconf/rules.mk	2014-10-03 12:57:19.714776541 +0200
@@ -366,7 +366,7 @@
 PWD := $(shell pwd)
 ifeq (,$(findstring ;,$(PATH)))
 ifndef USE_MSYS
-PWD := $(subst \,/,$(shell cygpath -w $(PWD)))
+#PWD := $(subst \,/,$(shell cygpath -w $(PWD)))
 endif
 endif
 endif
--- nss-3.17.1/nss/coreconf/WIN32.mk
+++ nss-3.17.1/nss/coreconf/WIN32.mk
@@ -97,7 +97,7 @@ MAKE_OBJDIR += $(OBJDIR)
 GARBAGE     += $(OBJDIR)/vc20.pdb $(OBJDIR)/vc40.pdb
 XP_DEFINE   += -DXP_PC
 ifdef NS_USE_GCC
-LIB_SUFFIX   = a
+LIB_SUFFIX   = dll.a
 else
 LIB_SUFFIX   = lib
 endif
--- nss-3.17.1/nss/coreconf/WINNT.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/coreconf/WINNT.mk	2014-10-03 12:57:19.715776517 +0200
@@ -14,7 +14,7 @@
 #
 # Win NT needs -GT so that fibers can work
 #
-OS_CFLAGS += -GT
+#OS_CFLAGS += -GT
 
 # WINNT uses the lib prefix, Win95 doesn't
 NSPR31_LIB_PREFIX = lib
--- nss-3.17.1/nss/lib/ckfw/builtins/config.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/ckfw/builtins/config.mk	2014-10-03 12:57:19.716776496 +0200
@@ -15,7 +15,7 @@
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
     SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-    RES = $(OBJDIR)/$(LIBRARY_NAME).res
+    RES = $(OBJDIR)/$(LIBRARY_NAME).res.o
     RESNAME = $(LIBRARY_NAME).rc
 endif
 
--- nss-3.17.1/nss/lib/ckfw/capi/Makefile	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/ckfw/capi/Makefile	2014-10-03 12:57:19.716776496 +0200
@@ -42,6 +42,9 @@
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
+	-lcrypt32 \
+	-ladvapi32 \
+	-lrpcrt4 \
 	$(NULL)
 endif
 
--- nss-3.17.1/nss/lib/freebl/config.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/freebl/config.mk	2014-10-03 12:57:19.716776496 +0200
@@ -50,7 +50,7 @@
 # don't want the 32 in the shared library name
 SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 
-RES     = $(OBJDIR)/$(LIBRARY_NAME).res
+RES     = $(OBJDIR)/$(LIBRARY_NAME).res.o
 RESNAME = freebl.rc
 
 ifdef NS_USE_GCC
--- nss-3.17.1/nss/lib/freebl/Makefile	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/freebl/Makefile	2014-10-03 12:57:19.717776476 +0200
@@ -147,6 +147,17 @@
     endif
 endif
 else
+# 64-bit Windows
+ifdef NS_USE_GCC
+    # win64/gcc - use the x86 code for now, skipping optimization
+    ifdef BUILD_OPT
+        OPTIMIZER += -Os
+    endif
+    ASFILES  =
+    DEFINES += -DMPI_AMD64 -DMP_USE_UINT_DIGIT
+    DEFINES += -DMP_CHAR_STORE_SLOW -DMP_IS_LITTLE_ENDIAN
+else
+# MSVC
     # -DMP_NO_MP_WORD
     DEFINES += -DMP_CHAR_STORE_SLOW -DMP_IS_LITTLE_ENDIAN
 ifdef NS_USE_GCC
@@ -171,6 +182,7 @@
 endif
 endif
 endif
+endif
 
 ifeq ($(OS_TARGET),IRIX)
 ifeq ($(USE_N32),1)
--- nss-3.17.1/nss/lib/freebl/mpi/mpi-priv.h	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/freebl/mpi/mpi-priv.h	2014-10-03 12:57:19.717776476 +0200
@@ -220,7 +220,7 @@
 #define MPI_ASM_DECL
 #endif
 
-#ifdef MPI_AMD64
+#if defined(MPI_AMD64) && defined(MP_ASSEMBLY_MULTIPLY)
 
 mp_digit MPI_ASM_DECL s_mpv_mul_set_vec64(mp_digit*, mp_digit *, mp_size, mp_digit);
 mp_digit MPI_ASM_DECL s_mpv_mul_add_vec64(mp_digit*, const mp_digit*, mp_size, mp_digit);
--- nss-3.17.1/nss/lib/nss/config.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/nss/config.mk	2014-10-03 12:57:19.717776476 +0200
@@ -10,7 +10,7 @@
 SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
 
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
+RES = $(OBJDIR)/$(LIBRARY_NAME).res.o
 RESNAME = $(LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
--- nss-3.17.1/nss/lib/smime/config.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/smime/config.mk	2014-10-03 12:57:19.718776454 +0200
@@ -11,7 +11,7 @@
 SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
 
-RES = $(OBJDIR)/smime.res
+RES = $(OBJDIR)/smime.res.o
 RESNAME = smime.rc
 
 ifdef NS_USE_GCC
--- nss-3.17.1/nss/lib/softoken/config.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/softoken/config.mk	2014-10-03 12:57:19.718776454 +0200
@@ -17,7 +17,7 @@
 SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
 
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
+RES = $(OBJDIR)/$(LIBRARY_NAME).res.o
 RESNAME = $(LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
--- nss-3.17.1/nss/lib/softoken/legacydb/config.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/softoken/legacydb/config.mk	2014-10-03 12:57:19.718776454 +0200
@@ -18,7 +18,7 @@
 SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
 
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
+RES = $(OBJDIR)/$(LIBRARY_NAME).res.o
 RESNAME = $(LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
--- nss-3.17.1/nss/lib/ssl/config.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/ssl/config.mk	2014-10-03 12:57:19.718776454 +0200
@@ -28,7 +28,7 @@
 SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
 
-RES = $(OBJDIR)/ssl.res
+RES = $(OBJDIR)/ssl.res.o
 RESNAME = ssl.rc
 
 ifdef NS_USE_GCC
--- nss-3.17.1/nss/lib/sysinit/config.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/sysinit/config.mk	2014-10-03 12:57:19.719776431 +0200
@@ -15,8 +15,8 @@
 SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
 
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
-RESNAME = $(LIBRARY_NAME).rc
+#RES = $(OBJDIR)/$(LIBRARY_NAME).res
+#RESNAME = $(LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
 EXTRA_SHARED_LIBS += \
--- nss-3.17.1/nss/lib/util/config.mk	2014-09-23 21:09:22.000000000 +0200
+++ nss-3.17.1/nss/lib/util/config.mk	2014-10-03 12:57:19.719776431 +0200
@@ -10,7 +10,7 @@
 SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
 
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
+RES = $(OBJDIR)/$(LIBRARY_NAME).res.o
 RESNAME = $(LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
--- nss-3.33/nss/cmd/pk11ectest/manifest.mn	2017-11-13 09:34:13.417188600 +0300
+++ nss-3.33/nss/cmd/pk11ectest/manifest.mn	2017-11-13 09:34:19.366799400 +0300
@@ -13,4 +13,4 @@
 
 PROGRAM = pk11ectest
 
-USE_STATIC_LIBS = 1
+#USE_STATIC_LIBS = 1
--- nss-3.33/nss/cmd/rsaperf/manifest.mn	2017-11-13 09:41:34.539284900 +0300
+++ nss-3.33/nss/cmd/rsaperf/manifest.mn	2017-11-13 09:41:38.458892000 +0300
@@ -21,4 +21,4 @@
 
 PROGRAM	= rsaperf
 
-USE_STATIC_LIBS = 1
+#USE_STATIC_LIBS = 1
--- nss-3.33/nss/cmd/platlibs.mk.orig	2017-11-13 10:14:28.414785800 +0300
+++ nss-3.33/nss/cmd/platlibs.mk	2017-11-13 10:14:56.929045300 +0300
@@ -199,6 +199,7 @@
 	$(DIST)/lib/$(IMPORT_LIB_PREFIX)smime3$(IMPORT_LIB_SUFFIX) \
 	$(DIST)/lib/$(IMPORT_LIB_PREFIX)ssl3$(IMPORT_LIB_SUFFIX) \
 	$(DIST)/lib/$(IMPORT_LIB_PREFIX)nss3$(IMPORT_LIB_SUFFIX) \
+	$(DIST)/lib/$(IMPORT_LIB_PREFIX)softokn3$(IMPORT_LIB_SUFFIX) \
 	$(NSPR_LIB_DIR)/$(NSPR31_LIB_PREFIX)plc4$(IMPORT_LIB_SUFFIX) \
 	$(NSPR_LIB_DIR)/$(NSPR31_LIB_PREFIX)plds4$(IMPORT_LIB_SUFFIX) \
 	$(NSPR_LIB_DIR)/$(NSPR31_LIB_PREFIX)nspr4$(IMPORT_LIB_SUFFIX) \
--- nss-3.47/nss/lib/ckfw/builtins/testlib/certdata-testlib.txt.orig	2019-11-07 14:27:03.372328600 +0300
+++ nss-3.47/nss/lib/ckfw/builtins/testlib/certdata-testlib.txt	2019-11-07 14:27:11.531142900 +0300
@@ -27,7 +27,7 @@
 # 3 - err_distrust:
 #   - The server/tls distrust field is set with CK_TRUE. These fields must be
 #     CK_FALSE when no schedule is set. Otherwise, must hold a valid encoded
-      timestamp.
+#     timestamp.
 #   - The email distrust field is set with an incomplete and invalid encoded
 #     timestamp.
 #
--- nss-3.47/nss/lib/ckfw/builtins/testlib/config.mk.orig	2019-11-07 14:31:24.907788000 +0300
+++ nss-3.47/nss/lib/ckfw/builtins/testlib/config.mk	2019-11-07 14:31:30.664198100 +0300
@@ -15,7 +15,7 @@
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 	SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-	RES = $(OBJDIR)/$(LIBRARY_NAME).res
+	RES = $(OBJDIR)/$(LIBRARY_NAME).res.o
 	RESNAME = $(LIBRARY_NAME).rc
 endif
 
