--- oiio-Release-1.7.12/src/tiff.imageio/tiffinput.cpp	2017-03-01 23:09:16.000000000 +0100
+++ oiio-Release-1.7.12-tiffposition/src/tiff.imageio/tiffinput.cpp	2017-03-03 22:59:51.000000000 +0100
@@ -676,12 +676,13 @@
     float x = 0, y = 0;
     TIFFGetField (m_tif, TIFFTAG_XPOSITION, &x);
     TIFFGetField (m_tif, TIFFTAG_YPOSITION, &y);
-    m_spec.x = (int)x;
-    m_spec.y = (int)y;
+    float xdensity = 1, ydensity = 1;
+    TIFFGetField (m_tif, TIFFTAG_XRESOLUTION, &xdensity);
+    TIFFGetField (m_tif, TIFFTAG_YRESOLUTION, &ydensity);
+
+    m_spec.x = (int)(x * xdensity + 0.5);
+    m_spec.y = (int)(y * ydensity + 0.5);
     m_spec.z = 0;
-    // FIXME? - TIFF spec describes the positions as in resolutionunit.
-    // What happens if this is not unitless pixels?  Are we interpreting
-    // it all wrong?
 
     // Start by assuming the "full" (aka display) window is the same as the
     // data window. That's what we'll stick to if there is no further
@@ -943,8 +944,6 @@
     case RESUNIT_INCH : m_spec.attribute ("ResolutionUnit", "in"); break;
     case RESUNIT_CENTIMETER : m_spec.attribute ("ResolutionUnit", "cm"); break;
     }
-    float xdensity = m_spec.get_float_attribute ("XResolution", 0.0f);
-    float ydensity = m_spec.get_float_attribute ("YResolution", 0.0f);
     if (xdensity && ydensity)
         m_spec.attribute ("PixelAspectRatio", ydensity/xdensity);
 
--- oiio-Release-1.7.12/src/tiff.imageio/tiffoutput.cpp	2017-03-01 23:09:16.000000000 +0100
+++ oiio-Release-1.7.12-tiffposition/src/tiff.imageio/tiffoutput.cpp	2017-03-03 23:00:50.000000000 +0100
@@ -296,8 +296,14 @@
 
     // N.B. Clamp position at 0... TIFF is internally incapable of having
     // negative origin.
-    TIFFSetField (m_tif, TIFFTAG_XPOSITION, (float)std::max (0, m_spec.x));
-    TIFFSetField (m_tif, TIFFTAG_YPOSITION, (float)std::max (0, m_spec.y));
+    float X_density = m_spec.get_float_attribute ("XResolution", 1.0f);
+    float Y_density = m_spec.get_float_attribute ("YResolution", 1.0f);
+    if (X_density <=  0.)
+        X_density = 1;
+    if (Y_density <=  0.)
+        Y_density = 1;
+    TIFFSetField (m_tif, TIFFTAG_XPOSITION, std::max (0.f, m_spec.x / X_density));
+    TIFFSetField (m_tif, TIFFTAG_YPOSITION, std::max (0.f, m_spec.y / Y_density));
 
     TIFFSetField (m_tif, TIFFTAG_IMAGEWIDTH, m_spec.width);
     TIFFSetField (m_tif, TIFFTAG_IMAGELENGTH, m_spec.height);
@@ -520,8 +526,6 @@
 
     // Deal with missing XResolution or YResolution, or a PixelAspectRatio
     // that contradicts them.
-    float X_density = m_spec.get_float_attribute ("XResolution", 1.0f);
-    float Y_density = m_spec.get_float_attribute ("YResolution", 1.0f);
     float aspect = m_spec.get_float_attribute ("PixelAspectRatio", 1.0f);
     if (X_density < 1.0f || Y_density < 1.0f || aspect*X_density != Y_density) {
         if (X_density < 1.0f || Y_density < 1.0f) {
