--- a/projects/libcxx/lib/CMakeLists.txt.orig	2019-03-30 20:06:20.000000000 -0700
+++ b/projects/libcxx/lib/CMakeLists.txt	2019-03-30 20:07:28.000000000 -0700
@@ -149,9 +149,6 @@
   endif()
 
   if ( CMAKE_OSX_DEPLOYMENT_TARGET STREQUAL "10.6" )
-    message(FATAL_ERROR "Mac OSX 10.6 is not supported anymore as a deployment "
-                        "target. If you need support for this, please contact "
-                        "the libc++ maintainers.")
   else()
     if ("armv7" IN_LIST CMAKE_OSX_ARCHITECTURES)
       set(RE_EXPORT_LIST "${CMAKE_CURRENT_SOURCE_DIR}/libc++sjlj-abi.exp")
