--- a/tools/clang/tools/extra/CMakeLists.txt	2018-04-03 01:26:00.000000000 -0700
+++ b/tools/clang/tools/extra/CMakeLists.txt	2018-04-03 14:05:29.000000000 -0700
@@ -10,7 +10,6 @@ add_subdirectory(change-namespace)
 add_subdirectory(clang-doc)
 add_subdirectory(clang-query)
 add_subdirectory(clang-move)
-add_subdirectory(clangd)
 add_subdirectory(include-fixer)
 add_subdirectory(pp-trace)
 add_subdirectory(tool-template)
