From 46543272dbe7655821e1c794ca3cc22e8cb4cef9 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Tue, 20 Mar 2018 23:44:23 +0100
Subject: gfile: Fix windows build

---
 goo/gfile.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/goo/gfile.cc b/goo/gfile.cc
index 218882d..06014b3 100644
--- a/goo/gfile.cc
+++ b/goo/gfile.cc
@@ -65,6 +65,8 @@
 #define PATH_MAX 1024
 #endif
 
+#ifndef _WIN32
+
 namespace {
 
 template< typename... >
@@ -99,6 +101,8 @@ inline const struct timespec& mtim(const struct stat& stbuf) {
 
 }
 
+#endif
+
 //------------------------------------------------------------------------
 
 GooString *getCurrentDir() {
-- 
cgit v1.1
