--- PKGBUILD.orig	2018-05-16 09:12:38.000000000 +0200
+++ PKGBUILD	2018-05-16 09:12:38.000000000 +0200
@@ -2,8 +2,8 @@
 # Contributer: Zach Bacon <11doctorwhocanada@gmail.com>
 
 _realname=ffmpeg
-pkgbase="mingw-w64-${_realname}"
-pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgbase="mingw-w64-${_realname}-gpl3"
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}-gpl3"
 pkgver=4.0
 pkgrel=2
 pkgdesc="Complete and free Internet live audio and video broadcasting solution (mingw-w64)"
@@ -22,16 +22,19 @@
         "${MINGW_PACKAGE_PREFIX}-libbluray"
         "${MINGW_PACKAGE_PREFIX}-libcaca"
         "${MINGW_PACKAGE_PREFIX}-libmodplug"
+        "${MINGW_PACKAGE_PREFIX}-librsvg"
         "${MINGW_PACKAGE_PREFIX}-libtheora"
         "${MINGW_PACKAGE_PREFIX}-libvorbis"
         "${MINGW_PACKAGE_PREFIX}-libvpx"
         "${MINGW_PACKAGE_PREFIX}-libwebp"
         "${MINGW_PACKAGE_PREFIX}-openal"
         "${MINGW_PACKAGE_PREFIX}-opencore-amr"
+        "${MINGW_PACKAGE_PREFIX}-openh264"
         "${MINGW_PACKAGE_PREFIX}-openjpeg2"
         "${MINGW_PACKAGE_PREFIX}-opus"
         "${MINGW_PACKAGE_PREFIX}-rtmpdump"
         "${MINGW_PACKAGE_PREFIX}-SDL2"
+        "${MINGW_PACKAGE_PREFIX}-snappy"
         "${MINGW_PACKAGE_PREFIX}-speex"
         "${MINGW_PACKAGE_PREFIX}-wavpack"
         "${MINGW_PACKAGE_PREFIX}-x264"
@@ -57,7 +60,7 @@
   cd "${srcdir}/build-${MINGW_CHOST}"
 
   ../${_realname}-${pkgver}/configure \
-    --prefix=${MINGW_PREFIX} \
+    --prefix=${MINGW_PREFIX}/ffmpeg-gpl3 \
     --target-os=mingw32 \
     --arch=${CARCH%%-*} \
     --disable-debug \
@@ -77,9 +80,12 @@
     --enable-libmp3lame \
     --enable-libopencore_amrnb \
     --enable-libopencore_amrwb \
+    --enable-libopenh264 \
     --enable-libopenjpeg \
     --enable-libopus \
+    --enable-librsvg \
     --enable-librtmp \
+    --enable-libsnappy \
     --enable-libspeex \
     --enable-libtheora \
     --enable-libvorbis \
@@ -108,18 +114,18 @@
   cd "${srcdir}/build-${MINGW_CHOST}"
   # workaround for conflict with SDL main(), use it if you have SDL installed
   #make check CC_C="-c -Umain"
-  make check
+  #make check
 }
 
 package() {
   cd "${srcdir}/build-${MINGW_CHOST}"
   make DESTDIR="${pkgdir}" install
-  mv "${pkgdir}${MINGW_PREFIX}"/bin/*.lib "${pkgdir}${MINGW_PREFIX}"/lib/
+  mv "${pkgdir}${MINGW_PREFIX}/ffmpeg-gpl3"/bin/*.lib "${pkgdir}${MINGW_PREFIX}/ffmpeg-gpl3"/lib/
 
-  #find ${pkgdir}${MINGW_PREFIX}/bin -type f -name "*.exe" -exec objcopy --subsystem console {} \;
-  rm -f ${pkgdir}${MINGW_PREFIX}/lib/*.def
-  rm -f ${pkgdir}${MINGW_PREFIX}/lib/*.lib
+  #find ${pkgdir}${MINGW_PREFIX}/ffmpeg-gpl3/bin -type f -name "*.exe" -exec objcopy --subsystem console {} \;
+  rm -f ${pkgdir}${MINGW_PREFIX}/ffmpeg-gpl3/lib/*.def
+  rm -f ${pkgdir}${MINGW_PREFIX}/ffmpeg-gpl3/lib/*.lib
   
   local PREFIX_DEPS=$(cygpath -am ${MINGW_PREFIX})
-  find ${pkgdir}${MINGW_PREFIX}/lib/pkgconfig -name *.pc -exec sed -i -e"s|${PREFIX_DEPS}|${MINGW_PREFIX}|g" {} \;
+  find ${pkgdir}${MINGW_PREFIX}/ffmpeg-gpl3/lib/pkgconfig -name *.pc -exec sed -i -e"s|${PREFIX_DEPS}|${MINGW_PREFIX}/ffmpeg-gpl3|g" {} \;
 }
