--- Modules/CMakeFindFrameworks.cmake.orig
+++ Modules/CMakeFindFrameworks.cmake
@@ -24,6 +24,7 @@
     if(APPLE)
       foreach(dir
           ~/Library/Frameworks/${fwk}.framework
+          __PREFIX__/Library/Frameworks/${fwk}.framework
           /Library/Frameworks/${fwk}.framework
           /System/Library/Frameworks/${fwk}.framework
           /Network/Library/Frameworks/${fwk}.framework)
