--- Portfile.orig	2020-02-08 08:54:55.000000000 -0800
+++ Portfile	2020-02-08 08:59:19.000000000 -0800
@@ -60,6 +60,16 @@
     patchfiles-append   patch-search_order.diff
     configure.ldflags-delete \
                         -L${prefix}/lib
+    # code generated with -Os on clang-3.9 crashes openexr in
+    # #0  Imf_2_2::DwaCompressor::initializeFuncs () at ImfDwaCompressor.cpp:2865
+    # -O, -O2, -O3 work, though
+    if { ${configure.compiler} eq "macports-clang-3.9" } {
+        configure.optflags "-O2"
+    }
+    # clang-4.0 and later are not yet tested, use -O2 just to be on the safe side
+    if { ${configure.compiler} eq "macports-clang-4.0" || ${configure.compiler} eq "macports-clang-5.0" || ${configure.compiler} eq "macports-clang-6.0" || ${configure.compiler} eq "macports-clang-7.0" || ${configure.compiler} eq "macports-clang-8.0" || ${configure.compiler} eq "macports-clang-9.0"} {
+        configure.optflags "-O2"
+    }
 }
 
 subport ilmbase {
@@ -70,6 +80,8 @@
                 sha256  456978d1a978a5f823c7c675f3f36b0ae14dba36638aeaa3c4b0e784f12a3862 \
                 size    595490
 
+    depends_build   port:gsed
+
     # Bitrotted ucontext usage is buggy, don't bother with it.
     patchfiles-append \
                 patch-configure.diff
