--- Portfile.orig	2018-06-22 17:51:30.000000000 +0200
+++ Portfile	2018-06-22 17:52:10.000000000 +0200
@@ -29,7 +29,7 @@
 }
 
 if {${subport} eq ${name}} {
-    #revision 0
+    revision 1
 
     depends_lib-append port:ilmbase
 
@@ -61,8 +61,17 @@
     # avoid "can't find a register in class 'BREG' while reloading 'asm'
     # see #51470 for MacPorts report
     # see https://github.com/openexr/openexr/issues/128 for upstream report
-    if { ${build_arch} eq "i386" && [string match *gcc-4* ${configure.compiler}] } {
         patchfiles-append patch-pic.diff
+
+    # code generated with -Os on clang-3.9 crashes openexr in
+    # #0  Imf_2_2::DwaCompressor::initializeFuncs () at ImfDwaCompressor.cpp:2865
+    # -O, -O2, -O3 work, though
+    if { ${configure.compiler} eq "macports-clang-3.9" } {
+        configure.optflags "-O2"
+    }
+    # clang-4.0 and later are not yet tested, use -O2 just to be on the safe side
+    if { ${configure.compiler} eq "macports-clang-4.0" || ${configure.compiler} eq "macports-clang-5.0" || ${configure.compiler} eq "macports-clang-6.0" } {
+        configure.optflags "-O2"
     }
 }
 
@@ -74,6 +83,11 @@
         sha256  cac206e63be68136ef556c2b555df659f45098c159ce24804e9d5e9e0286609e \
         size    573073
 
+    depends_build   port:gsed
+
+    # https://github.com/openexr/openexr/pull/170
+    patchfiles-append ilmbase-threadpool.diff
+
     # Bitrotted ucontext usage is buggy, don't bother with it.
     patchfiles-append patch-configure.diff
 }
