--- Portfile.orig	2017-05-06 17:53:02.000000000 +0200
+++ Portfile	2017-06-02 13:28:53.000000000 +0200
@@ -28,7 +28,7 @@
 }
 
 if {${subport} eq ${name}} {
-    revision 1
+    revision 2
 
     depends_lib-append port:ilmbase
 
@@ -60,8 +60,17 @@
     # avoid "can't find a register in class 'BREG' while reloading 'asm'
     # see #51470 for MacPorts report
     # see https://github.com/openexr/openexr/issues/128 for upstream report
-    if { ${build_arch} eq "i386" && [string match *gcc-4* ${configure.compiler}] } {
         patchfiles-append patch-pic.diff
+
+    # code generated with -Os on clang-3.9 crashes openexr in
+    # #0  Imf_2_2::DwaCompressor::initializeFuncs () at ImfDwaCompressor.cpp:2865
+    # -O, -O2, -O3 work, though
+    if { ${configure.compiler} eq "macports-clang-3.9" } {
+        configure.optflags "-O2"
+    }
+    # clang-4.0 was not yet tested, use -O2 just to be on the safe side
+    if { ${configure.compiler} eq "macports-clang-4.0" } {
+        configure.optflags "-O2"
     }
 }
 
@@ -71,6 +80,11 @@
         rmd160  b1adb0c96eff6b3554ddaf8f8bafe2cba8c114e3 \
         sha256  ecf815b60695555c1fbc73679e84c7c9902f4e8faa6e8000d2f905b8b86cedc7
 
+    depends_build   port:gsed
+
+    # https://github.com/openexr/openexr/pull/170
+    patchfiles-append ilmbase-threadpool.diff
+
     # Bitrotted ucontext usage is buggy, don't bother with it.
     patchfiles-append patch-configure.diff
 }
