--- Portfile.orig	2022-06-13 13:14:08.000000000 -0700
+++ Portfile	2022-06-13 13:18:44.000000000 -0700
@@ -175,12 +175,12 @@
 if {${universal_possible} && [variant_isset universal]} {
     set merger_host(x86_64) x86_64-apple-${os.platform}${os.major}
     set merger_host(i386) i686-apple-${os.platform}${os.major}
-    set merger_configure_args(x86_64) --build=x86_64-apple-${os.platform}${os.major}
-    set merger_configure_args(i386) --build=i686-apple-${os.platform}${os.major}
+    set merger_configure_args(x86_64) "--build=${build_arch}-apple-${os.platform}${os.major} --host=x86_64-apple-${os.platform}${os.major}"
+    set merger_configure_args(i386)   "--build=${build_arch}-apple-${os.platform}${os.major} --host=i686-apple-${os.platform}${os.major}"
     set merger_host(ppc) powerpc-apple-${os.platform}${os.major}
     set merger_host(ppc64) powerpc64-apple-${os.platform}${os.major}
-    set merger_configure_args(ppc) --build=powerpc-apple-${os.platform}${os.major}
-    set merger_configure_args(ppc64) --build=powerpc64-apple-${os.platform}${os.major}
+    set merger_configure_args(ppc)    "--build=${build_arch}-apple-${os.platform}${os.major} --host=ppc-apple-${os.platform}${os.major}"
+    set merger_configure_args(ppc64)  "--build=${build_arch}-apple-${os.platform}${os.major} --host=ppc64-apple-${os.platform}${os.major}"
 } elseif {${build_arch} eq "i386"} {
     configure.args-append \
         --host=i686-apple-${os.platform}${os.major} \
