--- mkspecs/macx-g++/qmake.conf.orig	2012-04-23 14:49:04.000000000 -0400
+++ mkspecs/macx-g++/qmake.conf	2012-04-17 14:38:16.000000000 -0400
@@ -9,7 +9,7 @@
 MAKEFILE_GENERATOR	= UNIX
 TARGET_PLATFORM		= macx
 TEMPLATE		= app
-CONFIG			+= qt warn_on release app_bundle incremental global_init_link_order lib_version_first plugin_no_soname link_prl
+CONFIG			+= qt warn_on release app_bundle incremental global_init_link_order lib_version_first plugin_no_soname link_prl absolute_library_soname
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
