--- tools/assistant/lib/lib.pro.orig	2012-03-14 10:01:12.000000000 -0400
+++ tools/assistant/lib/lib.pro	2012-04-25 12:32:49.000000000 -0400
@@ -12,16 +12,15 @@
     application \
     framework.
 DEFINES -= QT_ASCII_CAST_WARNINGS
-qclucene = QtCLucene$${QT_LIBINFIX}
-if(!debug_and_release|build_pass):CONFIG(debug, debug|release) { 
-    mac:qclucene = $${qclucene}_debug
-    win32:qclucene = $${qclucene}d
-}
-linux-lsb-g++:LIBS_PRIVATE += --lsb-shared-libs=$$qclucene
+
+qtAddLibrary(QtCLucene)
+
 unix|win32-g++*:QMAKE_PKGCONFIG_REQUIRES += QtNetwork \
     QtSql \
-    QtXml
-LIBS_PRIVATE += -l$$qclucene
+    QtGui \
+    QtCore \
+    QtCLucene
+
 RESOURCES += helpsystem.qrc
 SOURCES += qhelpenginecore.cpp \
     qhelpengine.cpp \
