--- src/tools/bootstrap/bootstrap.pro.orig	2015-05-07 10:14:44.000000000 -0400
+++ src/tools/bootstrap/bootstrap.pro	2015-05-30 13:30:53.000000000 -0400
@@ -103,7 +103,7 @@
 else:win32:SOURCES += ../../corelib/tools/qlocale_win.cpp
 
 macx: {
-   QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.4 #enables weak linking for 10.4 (exported)
+   QMAKE_MACOSX_DEPLOYMENT_TARGET = @MACOSX_DEPLOYMENT_TARGET@
    SOURCES += ../../corelib/kernel/qcore_mac.cpp
    LIBS += -framework CoreServices -framework ApplicationServices
 }
