diff --git a/libs/libmv/libmv/numeric/numeric.h b/libs/libmv/libmv/numeric/numeric.h
index 20a4a29..50181f0 100644
--- a/libs/libmv/libmv/numeric/numeric.h
+++ b/libs/libmv/libmv/numeric/numeric.h
@@ -34,9 +34,9 @@
 #include <Eigen/SVD>
 
 #if !defined(__MINGW64__)
-#  if defined(_WIN32) || defined(__APPLE__) || \
-      defined(__FreeBSD__) || defined(__NetBSD__)
-static void sincos(double x, double *sinx, double *cosx) {
+#  if (defined(_WIN32) || defined(__APPLE__) || \
+      defined(__FreeBSD__) || defined(__NetBSD__))  && !defined(__MINGW32__)
+inline void sincos(double x, double *sinx, double *cosx) {
   *sinx = sin(x);
   *cosx = cos(x);
 }
