--- libdjvu/ByteStream.cpp.orig	2012-05-08 11:56:53.000000000 +0900
+++ libdjvu/ByteStream.cpp	2012-09-29 16:14:17.000000000 +0900
@@ -648,7 +648,7 @@
 static FILE *
 urlfopen(const GURL &url,const char mode[])
 {
-#ifdef WIN32
+#if defined(WIN32)
   FILE *retval=0;
   const GUTF8String filename(url.UTF8Filename());
   wchar_t *wfilename;
@@ -666,6 +666,8 @@
 	}
   }
   return retval?retval:fopen((const char *)url.NativeFilename(),mode);
+#elif defined(__APPLE__)
+  return fopen((const char *)url.UTF8Filename(),mode);
 #else
   return fopen((const char *)url.NativeFilename(),mode);
 #endif
@@ -675,7 +677,11 @@
 static int
 urlopen(const GURL &url, const int mode, const int perm)
 {
+#if defined(__APPLE__)
+  return open((const char *)url.UTF8Filename(),mode,perm);
+#else
   return open((const char *)url.NativeFilename(),mode,perm);
+#endif
 }
 #endif /* UNIX */
 
