--- src/libjasper/jpc/jpc_dec.c.orig	2007-01-19 13:43:07.000000000 -0800
+++ src/libjasper/jpc/jpc_dec.c	2008-03-24 12:17:14.000000000 -0700
@@ -1069,12 +1069,18 @@
 	/* Apply an inverse intercomponent transform if necessary. */
 	switch (tile->cp->mctid) {
 	case JPC_MCT_RCT:
-		assert(dec->numcomps == 3);
+		if (dec->numcomps != 3 && dec->numcomps != 4) {
+			jas_eprintf("dec->numcomps (%d) != 3\n", dec->numcomps);
+			return -2;
+		}
 		jpc_irct(tile->tcomps[0].data, tile->tcomps[1].data,
 		  tile->tcomps[2].data);
 		break;
 	case JPC_MCT_ICT:
-		assert(dec->numcomps == 3);
+		if (dec->numcomps != 3 && dec->numcomps != 4) {
+			jas_eprintf("dec->numcomps (%d) != 3\n", dec->numcomps);
+			return -3;
+		}
 		jpc_iict(tile->tcomps[0].data, tile->tcomps[1].data,
 		  tile->tcomps[2].data);
 		break;
