--- Portfile.orig	2017-11-18 20:07:00.000000000 +0100
+++ Portfile	2017-11-19 22:54:00.000000000 +0100
@@ -7,6 +7,7 @@
 
 name                x265
 version             2.5
+revision            1
 
 checksums           rmd160  612e545199dcaadef782ca0a71a5ade679876509 \
                     sha256  2e53259b504a7edb9b21b9800163b1ff4c90e60c74e23e7001d423c69c5d3d17
@@ -31,15 +32,27 @@
 cmake.out_of_source yes
 compiler.blacklist  *llvm-gcc-4.2
 
+configure.env-append CONFIGURE='${configure.cmd}'
+build.env-append     BUILD='${build.cmd}'
+destroot.env-append  BUILD='${build.cmd}' ARCH=none
+configure.cmd        ${portpath}/${filesdir}/configure.sh
+build.cmd            ${portpath}/${filesdir}/build.sh
+
 #    error: illegal text-relocation to '_x265_pw_1'
 #   report: https://mailman.videolan.org/pipermail/x265-devel/2014-May/004476.html
 # solution: http://lists.apple.com/archives/unix-porting/2008/Jan/msg00027.html
 if {[variant_exists universal] && [variant_isset universal]} {
     lappend merger_configure_ldflags(i386) -Wl,-read_only_relocs,suppress
+    foreach arch ${configure.universal_archs} {
+        lappend merger_configure_env(${arch}) ARCH='${arch}'
+        lappend merger_build_env(${arch}) ARCH='${arch}'
+    }
 } else {
     if {${build_arch} eq "i386"} {
         configure.ldflags-append -Wl,-read_only_relocs,suppress
     }
+    configure.env-append ARCH='${build_arch}'
+    build.env-append ARCH='${build_arch}'
 }
 
 livecheck.regex     "${name}_(\\d+(?:\\.\\d+)*)${extract.suffix}"
