--- Portfile.orig	2019-11-04 22:48:59.000000000 -0800
+++ Portfile	2019-11-04 22:50:45.000000000 -0800
@@ -8,7 +8,7 @@
 
 name                x265
 version             3.2
-revision            1
+revision            2
 
 checksums           rmd160  f9069ae6504a9f04702a5e3ea78cf1985cd63646 \
                     sha256  364d79bcd56116a9e070fdeb1d9d2aaef1a786b4970163fb56ff0991a183133b \
@@ -40,15 +40,27 @@
 # To be reviewed once Xcode 11.2 comes out.
 compiler.blacklist-append {clang > 1099}
 
+configure.env-append CONFIGURE='${configure.cmd}' BUILD='${build.cmd}'
+build.env-append     BUILD='${build.cmd}'
+destroot.env-append  BUILD='${build.cmd}' ARCH=none
+configure.cmd        ${portpath}/${filesdir}/configure.sh
+build.cmd            ${portpath}/${filesdir}/build.sh
+
 #    error: illegal text-relocation to '_x265_pw_1'
 #   report: https://mailman.videolan.org/pipermail/x265-devel/2014-May/004476.html
 # solution: http://lists.apple.com/archives/unix-porting/2008/Jan/msg00027.html
 if {[variant_exists universal] && [variant_isset universal]} {
     lappend merger_configure_ldflags(i386) -Wl,-read_only_relocs,suppress
+    foreach arch ${configure.universal_archs} {
+        lappend merger_configure_env(${arch}) ARCH='${arch}'
+        lappend merger_build_env(${arch}) ARCH='${arch}'
+    }
 } else {
     if {${build_arch} eq "i386"} {
         configure.ldflags-append -Wl,-read_only_relocs,suppress
     }
+    configure.env-append ARCH='${build_arch}'
+    build.env-append ARCH='${build_arch}'
 }
 
 livecheck.regex     "${name}_(\\d+(?:\\.\\d+)*)${extract.suffix}"
