--- CMakeLists.txt.orig	2020-05-29 23:24:35.000000000 +0545
+++ CMakeLists.txt	2022-08-04 01:19:02.000000000 +0545
@@ -43,7 +43,7 @@
 set(ARM_ALIASES armv6l armv7l aarch64)
 list(FIND X86_ALIASES "${SYSPROC}" X86MATCH)
 list(FIND ARM_ALIASES "${SYSPROC}" ARMMATCH)
-set(POWER_ALIASES ppc64 ppc64le)
+set(POWER_ALIASES ppc ppc64 ppc64le)
 list(FIND POWER_ALIASES "${SYSPROC}" POWERMATCH)
 if("${SYSPROC}" STREQUAL "" OR X86MATCH GREATER "-1")
     set(X86 1)
@@ -469,7 +469,7 @@
     endif(WINXP_SUPPORT)
 endif()
 
-if(POWER)
+if(POWER AND NOT APPLE)
     # IBM Power8
     option(ENABLE_ALTIVEC "Enable ALTIVEC profiling instrumentation" ON)
     if(ENABLE_ALTIVEC)
