--- CMakeLists.txt	2012-08-08 11:29:50.000000000 -0400
+++ CMakeLists.txt	2012-08-08 11:30:08.000000000 -0400
@@ -38,7 +38,7 @@
     set(CMAKE_CXX_FLAGS "/Zc:wchar_t- /GR /EHsc /DWIN32 /D_WINDOWS /D_SCL_SECURE_NO_WARNINGS")
 else()
     if(CMAKE_HOST_UNIX AND NOT CYGWIN)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -fvisibility=hidden -Wno-strict-aliasing")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -fvisibility=default -Wno-strict-aliasing")
     endif()
     set(CMAKE_CXX_FLAGS_DEBUG "-g")
     option(ENABLE_GCC_OPTIMIZATION "Enable specific GCC flags to optimization library size and performance. Only available on Release Mode" 0)
