diff -rubw data/ShibokenConfig-spec.cmake.in data/ShibokenConfig-spec.cmake.in
--- data/ShibokenConfig-spec.cmake.in	2014-04-22 10:42:48.000000000 -0500
+++ data/ShibokenConfig-spec.cmake.in	2014-07-05 18:42:33.000000000 -0500
@@ -5,7 +5,7 @@
 #  SHIBOKEN_PYTHON_INTERPRETER - Python interpreter (regular or debug) to be used with the bindings.
 #  SHIBOKEN_PYTHON_LIBRARIES   - Python libraries (regular or debug) Shiboken is linked against.
 
-SET(SHIBOKEN_INCLUDE_DIR "@CMAKE_INSTALL_PREFIX@/include/shiboken@shiboken_SUFFIX@")
+SET(SHIBOKEN_INCLUDE_DIR "@CMAKE_INSTALL_PREFIX@/include/shiboken-@PYTHONBRANCH@")
 if(MSVC)
     SET(SHIBOKEN_LIBRARY "@LIB_INSTALL_DIR@/@CMAKE_SHARED_LIBRARY_PREFIX@shiboken@shiboken_SUFFIX@@PYTHON_SUFFIX@@LIBRARY_OUTPUT_SUFFIX@.lib")
 elseif(CYGWIN)
@@ -23,4 +23,4 @@
 message(STATUS "libshiboken built for @SHIBOKEN_BUILD_TYPE@")
 @SBK_ADD_PY_DEBUG_DEFINITION@
 
-set(SHIBOKEN_BINARY "@CMAKE_INSTALL_PREFIX@/bin/@SHIBOKEN_GENERATOR@")
+set(SHIBOKEN_BINARY "@CMAKE_INSTALL_PREFIX@/bin/@SHIBOKEN_GENERATOR@-@PYTHONBRANCH@")
diff -rubw data/ShibokenConfig.cmake.in data/ShibokenConfig.cmake.in
--- data/ShibokenConfig.cmake.in	2014-04-22 10:42:48.000000000 -0500
+++ data/ShibokenConfig.cmake.in	2014-07-05 18:42:37.000000000 -0500
@@ -2,4 +2,4 @@
     message(STATUS "Using default python: @PYTHON_SUFFIX@")
     SET(PYTHON_SUFFIX @PYTHON_SUFFIX@)
 endif()
-include(@LIB_INSTALL_DIR@/cmake/Shiboken-@shiboken_VERSION@/ShibokenConfig${PYTHON_SUFFIX}.cmake)
+include(@PYTHONPREFIX@/lib/cmake/Shiboken-@shiboken_VERSION@/ShibokenConfig${PYTHON_SUFFIX}.cmake)
diff -rubw data/shiboken.pc.in data/shiboken.pc.in
--- data/shiboken.pc.in	2014-04-22 10:42:48.000000000 -0500
+++ data/shiboken.pc.in	2014-07-05 18:42:28.000000000 -0500
@@ -1,8 +1,8 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@LIB_INSTALL_DIR@
-includedir=@CMAKE_INSTALL_PREFIX@/include/shiboken
-generator_location=@CMAKE_INSTALL_PREFIX@/bin/@SHIBOKEN_GENERATOR@
+includedir=@CMAKE_INSTALL_PREFIX@/include/shiboken-@PYTHONBRANCH@
+generator_location=@CMAKE_INSTALL_PREFIX@/bin/@SHIBOKEN_GENERATOR@-@PYTHONBRANCH@
 python_interpreter=@PYTHON_EXECUTABLE@
 python_include_dir=@SBK_PYTHON_INCLUDE_DIR@
 
