--- gobject-introspection-1.66.1/giscanner/ccompiler.py.orig	2021-04-28 20:23:22.145710000 -0700
+++ gobject-introspection-1.66.1/giscanner/ccompiler.py	2021-04-28 20:23:26.473853400 -0700
@@ -208,7 +208,7 @@
                 args.append('-libpath:' + library_path)
             else:
                 args.append('-L' + library_path)
-                if os.path.isabs(library_path):
+                if os.name != 'nt' and os.path.isabs(library_path):
                     if libtool:
                         args.append('-rpath')
                         args.append(library_path)
