--- configure.ac.orig	2015-03-30 19:21:24.000000000 +1100
+++ configure.ac	2015-05-20 12:57:22.000000000 +1000
@@ -329,8 +329,8 @@
 if test $ac_cv_prog_gcc = yes; then
     cflags_to_try="-Wall -Wstrict-prototypes -Wundef \
 -Wmissing-declarations -Wmissing-prototypes -Wwrite-strings \
--Wno-strict-aliasing -Werror=declaration-after-statement \
--fno-builtin -fno-common -Werror=return-type"
+-Wno-strict-aliasing \
+-fno-builtin -fno-common"
     optflags_to_try="$CC_OPT_FLAGS_TO_TRY"
     dbgflags_to_try="$CC_DBG_FLAGS_TO_TRY"
 else
