--- configure.ac.orig	2012-06-25 02:15:47.000000000 +0200
+++ configure.ac	2012-06-25 02:16:31.000000000 +0200
@@ -1739,24 +1739,16 @@
 dnl check for big/little endian for LCMS
 dnl --------------------------------------------------
 
-AC_MSG_CHECKING([for big endian])
-
-AC_RUN_IFELSE(
-  [AC_LANG_PROGRAM([#include <stdio.h>], [
-   static const int one = 1;
-   return (*(char*)&one == 0 ? 0 : 1);
-  ])],
+AC_C_BIGENDIAN(
   [LCMS_BIGENDIAN=1],
   [LCMS_BIGENDIAN=0])
 
 if test "x$LCMS_BIGENDIAN" != "x0"; then
   LCMS_ENDIAN="-DUSE_BIG_ENDIAN=$LCMS_BIGENDIAN"
   LCMS2_ENDIAN="-DCMS_USE_BIG_ENDIAN=$LCMS_BIGENDIAN"
-  AC_MSG_RESULT(yes)
 else
   LCMS_ENDIAN=
   LCMS2_ENDIAN=
-  AC_MSG_RESULT(no)
 fi
 
 
