doc symlink should not point into DESTDIR
--- base/unixinst.mak.orig	2018-03-21 03:48:06.000000000 -0500
+++ base/unixinst.mak	2018-03-27 06:26:37.000000000 -0500
@@ -165,7 +165,7 @@
 	$(SH) -c 'for f in $(DOC_PAGES) ;\
 	do if ( test -f $(PSDOCDIR)/$$f ); then $(INSTALL_DATA) $(PSDOCDIR)/$$f $(DESTDIR)$(docdir); fi;\
 	done'
-	ln -s $(DESTDIR)$(docdir) $(DESTDIR)$(gsdatadir)/doc
+	ln -s $(docdir) $(DESTDIR)$(gsdatadir)/doc
 
 # install the man pages for each locale
 MAN_LCDIRS=. de
