diff -burN Magick++.orig/lib/Functions.cpp Magick++/lib/Functions.cpp
--- Magick++.orig/lib/Functions.cpp	2014-06-18 00:12:25.000000000 +0200
+++ Magick++/lib/Functions.cpp	2015-05-25 17:47:46.617371888 +0200
@@ -56,6 +56,12 @@
     magick_initialized=true;
 }
 
+// Random seed
+MagickPPExport void Magick::SetRandomSeed(const unsigned long seed) 
+{ 
+  MagickCore::SetRandomSecretKey(seed); 
+} 
+
 //
 // Create a local wrapper around MagickCoreTerminus
 //
diff -burN Magick++.orig/lib/Magick++/Functions.h Magick++/lib/Magick++/Functions.h
--- Magick++.orig/lib/Magick++/Functions.h	2014-04-09 00:59:33.000000000 +0200
+++ Magick++/lib/Magick++/Functions.h	2015-05-25 17:48:18.169855728 +0200
@@ -26,5 +26,7 @@
 
   // C library initialization routine
   MagickPPExport void InitializeMagick(const char *path_);
+  // Seed a new sequence of pseudo-random numbers 
+  MagickPPExport void SetRandomSeed(const unsigned long seed); 
 }
 #endif // Magick_Functions_header
