--- x264-snapshot-20140208-2245-stable/configure.orig	2014-02-09 16:10:32.011010400 +0000
+++ x264-snapshot-20140208-2245-stable/configure	2014-02-09 16:11:59.730649400 +0000
@@ -1249,9 +1249,9 @@
 Name: x264
 Description: H.264 (MPEG4 AVC) encoder library
 Version: $(grep POINTVER < x264_config.h | sed -e 's/.* "//; s/".*//')
-Libs: -L$libdir -lx264 $([ "$shared" = "yes" ] || echo $libpthread $libm $libdl)
+Libs: -L\${libdir} -lx264 $([ "$shared" = "yes" ] || echo $libpthread $libm $libdl)
 Libs.private: $([ "$shared" = "yes" ] && echo $libpthread $libm $libdl)
-Cflags: -I$includedir
+Cflags: -I\${includedir}
 EOF
 
 filters="crop select_every"
