diff -ur /Users/devernay/Development/blender/extern/glog/src/windows/port.h ./src/windows/port.h
--- /Users/devernay/Development/blender/extern/glog/src/windows/port.h	2016-03-21 18:40:14.000000000 +0100
+++ ./src/windows/port.h	2016-05-20 11:55:42.000000000 +0200
@@ -136,17 +136,22 @@
 #endif  // _MSC_VER
 
 // ----------------------------------- THREADS
+#ifndef __MINGW32__ // MingW
 typedef DWORD pthread_t;
 typedef DWORD pthread_key_t;
 typedef LONG pthread_once_t;
 enum { PTHREAD_ONCE_INIT = 0 };   // important that this be 0! for SpinLock
+#else
 #define pthread_self  GetCurrentThreadId
 #define pthread_equal(pthread_t_1, pthread_t_2)  ((pthread_t_1)==(pthread_t_2))
+#endif
 
+#ifndef __MINGW32__ // MingW
 inline struct tm* localtime_r(const time_t* timep, struct tm* result) {
   localtime_s(result, timep);
   return result;
 }
+#endif
 
 inline char* strerror_r(int errnum, char* buf, size_t buflen) {
 #ifdef FREE_WINDOWS
--- CMakeLists.txt.orig	2022-08-08 10:34:58.000000000 -0400
+++ CMakeLists.txt	2022-08-08 10:34:54.000000000 -0400
@@ -72,8 +72,8 @@
     )
 endif()
 add_library(glog STATIC ${glog_HEADERS} ${glog_SOURCES})
-target_include_directories(glog src)
-target_compile_definitions(glog GOOGLE_GLOG_DLL_DECL=)
+target_include_directories(glog PRIVATE src)
+target_compile_definitions(glog PRIVATE GOOGLE_GLOG_DLL_DECL=)
 if(WINDOWS)
     target_include_directories(APPEND glog src/windows)
     target_compile_definitions(APPEND glog GLOG_NO_ABBREVIATED_SEVERITIES)
