--- base/gsgdata.h.orig	2011-08-05 20:12:20.000000000 +0900
+++ base/gsgdata.h	2011-08-13 08:40:25.000000000 +0900
@@ -127,4 +127,8 @@
 /* from_null(pgd) is a shortcut for from_string(pgd, NULL, 0, NULL). */
 void gs_glyph_data_from_null(gs_glyph_data_t *pgd);
 
+extern const gs_glyph_data_procs_t gs_glyph_cache_elem_procs;
+static const gs_glyph_data_procs_t no_free_procs;
+static const gs_glyph_data_procs_t free_by_font_procs;
+
 #endif /* gsgdata_INCLUDED */
