--- base/gsgcache.c.orig	2011-08-05 20:12:20.000000000 +0900
+++ base/gsgcache.c	2011-10-31 13:41:24.000000000 +0900
@@ -162,7 +162,7 @@
     return_error(gs_error_unregistered); /* Unsupported; should not happen. */
 }
 
-static const gs_glyph_data_procs_t gs_glyph_cache_elem_procs = {
+       const gs_glyph_data_procs_t gs_glyph_cache_elem_procs = {
     gs_glyph_cache_elem__free_data, gs_glyph_cache_elem__substring
 };
 
