--- base/unixinst.mak.orig	2012-02-08 17:48:48.000000000 +0900
+++ base/unixinst.mak	2012-02-14 07:57:08.000000000 +0900
@@ -127,17 +127,17 @@
 	   C-style.htm Changes.htm Commprod.htm Copying.htm \
 	   Current.htm Deprecated.htm \
 	   DLL.htm Deprecated.htm Develop.htm Devices.htm Drivers.htm \
-	   Fonts.htm Helpers.htm Hershey.htm \
+	   Fonts.htm GS9_Color_Management.pdf Helpers.htm Hershey.htm \
 	   History1.htm History2.htm History3.htm History4.htm \
 	   History5.htm History6.htm History7.htm History8.htm \
 	   History9.htm \
 	   Details.htm Details8.htm Details9.htm \
 	   Htmstyle.htm Humor.htm Issues.htm Install.htm Language.htm \
 	   Lib.htm Maintain.htm Make.htm New-user.htm \
-	   News.htm Projects.htm Ps-style.htm Ps2epsi.htm Ps2pdf.htm \
+	   News.htm Projects.htm Ps-style.htm Ps2epsi.htm Ps2pdf.htm Ps2ps2.htm \
 	   Psfiles.htm Public.htm Readme.htm Release.htm \
 	   Source.htm Testing.htm Unix-lpr.htm \
-	   Use.htm Xfonts.htm
+	   Use.htm WhatIsGS.htm Xfonts.htm thirdparty.htm
 install-doc: $(PSDOCDIR)/News.htm
 	-mkdir -p $(DESTDIR)$(docdir)
 	$(SH) -c 'for f in $(DOC_PAGES) ;\
