diff --git a/src/libtexture/imagecache_pvt.h b/src/libtexture/imagecache_pvt.h
index 804d5d2..90b7c6f 100644
--- a/src/libtexture/imagecache_pvt.h
+++ b/src/libtexture/imagecache_pvt.h
@@ -290,7 +290,8 @@ public:
     /// Forget the specs we know
     void invalidate_spec () {
         m_validspec = false;
-        m_subimages.clear ();
+        //Do not clear the subimages here because another thread might be using it
+        //m_subimages.clear ();
     }
     
 private:
