--- build_scripts/utils.py.orig	2018-07-26 23:19:42.000000000 -0700
+++ build_scripts/utils.py	2018-08-24 10:32:58.000000000 -0700
@@ -674,7 +674,7 @@
     version_re = re.compile(r'(\d+)\.(\d+)\.(\d+)')
     result = None
     last_version_string = '000000'
-    for llvm_config in find_glob_in_path('llvm-config*'):
+    for llvm_config in find_glob_in_path('llvm-config'):
         try:
             output = run_process_output([llvm_config, '--version'])
             if output:
