--- configure.ac.orig	2019-07-05 02:28:51.961544100 +0200
+++ configure.ac	2019-07-05 02:30:40.799213700 +0200
@@ -1719,7 +1719,7 @@
 dnl ===========================================================================
 
 # whether modules are built or not.
-with_ltdl='yes'
+with_ltdl='no'
 have_ltdl='no'
 LTDL_LIBS=''
 if test "$build_modules" != 'no' || test "X$ax_cv_check_cl_libcl" != Xno; then
@@ -1746,9 +1746,9 @@
   else
     AC_MSG_RESULT([no])
   fi
-  if test "$have_ltdl" != 'yes' ; then
-    AC_MSG_FAILURE([libltdl is required for modules and OpenCL builds],[1])
-  fi
+#  if test "$have_ltdl" != 'yes' ; then
+#    AC_MSG_FAILURE([libltdl is required for modules and OpenCL builds],[1])
+#  fi
 fi
 AM_CONDITIONAL(WITH_LTDL, test "$have_ltdl" != 'no')
 
