--- MagickCore/distribute-cache.c.orig	2018-12-26 08:36:39.624517664 +0100
+++ MagickCore/distribute-cache.c	2018-12-26 08:37:27.240625981 +0100
@@ -89,6 +89,8 @@
 #define LENGTH_TYPE size_t
 #define MAGICKCORE_HAVE_DISTRIBUTE_CACHE
 #elif defined(MAGICKCORE_WINDOWS_SUPPORT) && !defined(__MINGW32__)
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #define CHAR_TYPE_CAST (char *)
 #define CLOSE_SOCKET(socket) (void) closesocket(socket)
 #define HANDLER_RETURN_TYPE DWORD WINAPI
--- MagickCore/nt-base.h.orig	2018-12-26 08:40:27.833033590 +0100
+++ MagickCore/nt-base.h	2018-12-26 08:40:44.985072087 +0100
@@ -39,6 +39,8 @@
 #include <errno.h>
 #include <malloc.h>
 #include <sys/utime.h>
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #if defined(_DEBUG) && !defined(__MINGW32__)
 #include <crtdbg.h>
 #endif
