--- libde265-1.0.3/libde265.pc.in.orig	2016-06-23 11:31:00.000000000 +0300
+++ libde265-1.0.3/libde265.pc.in	2019-06-11 09:48:52.508268000 +0300
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/@PACKAGE@
 
 Name: libde265
 Description: H.265/HEVC video decoder.
@@ -10,4 +10,4 @@
 Requires: 
 Libs: -lde265 -L@libdir@
 Libs.private: @LIBS@ -lstdc++
-Cflags: -I@includedir@
+Cflags: -I@includedir@/@PACKAGE@
