--- config/autoconf.mk.in	Tue Nov 12 04:52:34 2013
+++ config/autoconf.mk.in	Tue Nov 26 11:22:37 2013
@@ -31,6 +31,7 @@
 OBJ_SUFFIX	= $(error config/config.mk needs to be included before using OBJ_SUFFIX)
 _OBJ_SUFFIX	= @OBJ_SUFFIX@
 LIB_SUFFIX	= @LIB_SUFFIX@
+IMPORT_LIB_SUFFIX = @IMPORT_LIB_SUFFIX@
 DLL_SUFFIX	= @DLL_SUFFIX@
 ASM_SUFFIX	= @ASM_SUFFIX@
 MOD_NAME	= @NSPR_MODNAME@
--- config/nspr.pc.in	Tue Nov 12 04:52:34 2013
+++ config/nspr.pc.in	Tue Nov 26 11:22:46 2013
@@ -6,5 +6,5 @@
 Name: NSPR
 Description: The Netscape Portable Runtime
 Version: @MOD_MAJOR_VERSION@.@MOD_MINOR_VERSION@.@MOD_PATCH_VERSION@
-Libs: -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@
-Cflags: -I@includedir@
+Libs: -L${libdir} -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@
+Cflags: -I${includedir} 
--- config/rules.mk	Tue Nov 12 04:52:34 2013
+++ config/rules.mk	Tue Nov 26 11:22:37 2013
@@ -85,16 +85,16 @@
 SHARED_LIB_PDB	= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).pdb
 ifdef MSC_VER
 LIBRARY         = $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION)_s.$(LIB_SUFFIX)
-IMPORT_LIBRARY  = $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
+IMPORT_LIBRARY  = $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).$(IMPORT_LIB_SUFFIX)
 else
 LIBRARY         = $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION)_s.$(LIB_SUFFIX)
-IMPORT_LIBRARY  = $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
+IMPORT_LIBRARY  = $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(IMPORT_LIB_SUFFIX)
 endif
 else
 SHARED_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
 SHARED_LIB_PDB	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).pdb
 LIBRARY         = $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION)_s.$(LIB_SUFFIX)
-IMPORT_LIBRARY  = $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
+IMPORT_LIBRARY  = $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(IMPORT_LIB_SUFFIX)
 endif
 
 else
--- configure	Tue Nov 12 04:54:52 2013
+++ configure	Tue Nov 26 11:22:37 2013
@@ -665,6 +665,7 @@
 ASM_SUFFIX
 DLL_SUFFIX
 LIB_SUFFIX
+IMPORT_LIB_SUFFIX
 OBJ_SUFFIX
 CPU_ARCH
 PR_MD_ARCH_DIR
@@ -5615,6 +5615,7 @@
 
 OBJ_SUFFIX=o
 LIB_SUFFIX=a
+IMPORT_LIB_SUFFIX=a
 DLL_SUFFIX=so
 ASM_SUFFIX=s
 MKSHLIB='$(LD) $(DSO_LDOPTS) -o $@'
@@ -7112,6 +7112,7 @@
         CC="$CC -mwindows"
         CXX="$CXX -mwindows"
         DLL_SUFFIX=dll
+        IMPORT_LIB_SUFFIX=dll.a
         MKSHLIB='$(CC) -shared -Wl,--export-all-symbols -Wl,--out-implib -Wl,$(IMPORT_LIBRARY) $(DLLBASE) -o $(subst $(OBJDIR)/,,$(SHARED_LIBRARY))'
         RC=$WINDRES
         # Use temp file for windres (bug 213281)
@@ -7127,6 +7127,7 @@
         GARBAGE='$(OBJDIR)/vc20.pdb $(OBJDIR)/vc40.pdb'
         OBJ_SUFFIX=obj
         LIB_SUFFIX=lib
+        IMPORT_LIB_SUFFIX=lib
         DLL_SUFFIX=dll
 
         # Determine compiler version
@@ -7246,11 +7246,11 @@
         
         if test "$OS_TARGET" = "WINNT"; then
             CFLAGS="$CFLAGS -GT"
-            LIBNSPR='$(dist_libdir)/libnspr$(MOD_MAJOR_VERSION).$(LIB_SUFFIX)'
-            LIBPLC='$(dist_libdir)/libplc$(MOD_MAJOR_VERSION).$(LIB_SUFFIX)'
+            LIBNSPR='$(dist_libdir)/libnspr$(MOD_MAJOR_VERSION).$(IMPORT_LIB_SUFFIX)'
+            LIBPLC='$(dist_libdir)/libplc$(MOD_MAJOR_VERSION).$(IMPORT_LIB_SUFFIX)'
         else
-            LIBNSPR='$(dist_libdir)/nspr$(MOD_MAJOR_VERSION).$(LIB_SUFFIX)'
-            LIBPLC='$(dist_libdir)/plc$(MOD_MAJOR_VERSION).$(LIB_SUFFIX)'
+            LIBNSPR='$(dist_libdir)/nspr$(MOD_MAJOR_VERSION).$(IMPORT_LIB_SUFFIX)'
+            LIBPLC='$(dist_libdir)/plc$(MOD_MAJOR_VERSION).$(IMPORT_LIB_SUFFIX)'
         fi
     fi # GNU_CC
 
@@ -7736,6 +7736,7 @@
 
     OBJ_SUFFIX=obj
     LIB_SUFFIX=lib
+    IMPORT_LIB_SUFFIX=lib
     DLL_SUFFIX=dll
     MKSHLIB='$(LD) -DLL $(DSO_LDOPTS) -OUT:$@'
 
@@ -7409,6 +7409,7 @@
     $as_echo "#define TCPV40HDRS 1" >>confdefs.h
 
     LIB_SUFFIX=lib
+    IMPORT_LIB_SUFFIX=lib
     DLL_SUFFIX=dll
     RC=rc.exe
     PR_MD_ARCH_DIR=os2
--- configure.in	Tue Nov 12 04:54:52 2013
+++ configure.in	Tue Nov 26 11:22:37 2013
@@ -845,6 +845,7 @@
 dnl ========================================================
 OBJ_SUFFIX=o
 LIB_SUFFIX=a
+IMPORT_LIB_SUFFIX=.a
 DLL_SUFFIX=so
 ASM_SUFFIX=s
 MKSHLIB='$(LD) $(DSO_LDOPTS) -o $@'
@@ -1912,6 +1913,7 @@
         CC="$CC -mwindows"
         CXX="$CXX -mwindows"
         DLL_SUFFIX=dll
+        IMPORT_LIB_SUFFIX=dll.a
         MKSHLIB='$(CC) -shared -Wl,--export-all-symbols -Wl,--out-implib -Wl,$(IMPORT_LIBRARY) $(DLLBASE) -o $(subst $(OBJDIR)/,,$(SHARED_LIBRARY))'
         RC=$WINDRES
         # Use temp file for windres (bug 213281)
@@ -1926,6 +1928,7 @@
         GARBAGE='$(OBJDIR)/vc20.pdb $(OBJDIR)/vc40.pdb'
         OBJ_SUFFIX=obj
         LIB_SUFFIX=lib
+        IMPORT_LIB_SUFFIX=lib
         DLL_SUFFIX=dll
 
         # Determine compiler version
@@ -1945,11 +1948,11 @@
         
         if test "$OS_TARGET" = "WINNT"; then
             CFLAGS="$CFLAGS -GT"
-            LIBNSPR='$(dist_libdir)/libnspr$(MOD_MAJOR_VERSION).$(LIB_SUFFIX)'
-            LIBPLC='$(dist_libdir)/libplc$(MOD_MAJOR_VERSION).$(LIB_SUFFIX)'
+            LIBNSPR='$(dist_libdir)/libnspr$(MOD_MAJOR_VERSION).$(IMPORT_LIB_SUFFIX)'
+            LIBPLC='$(dist_libdir)/libplc$(MOD_MAJOR_VERSION).$(IMPORT_LIB_SUFFIX)'
         else
-            LIBNSPR='$(dist_libdir)/nspr$(MOD_MAJOR_VERSION).$(LIB_SUFFIX)'
-            LIBPLC='$(dist_libdir)/plc$(MOD_MAJOR_VERSION).$(LIB_SUFFIX)'
+            LIBNSPR='$(dist_libdir)/nspr$(MOD_MAJOR_VERSION).$(IMPORT_LIB_SUFFIX)'
+            LIBPLC='$(dist_libdir)/plc$(MOD_MAJOR_VERSION).$(IMPORT_LIB_SUFFIX)'
         fi
     fi # GNU_CC
 
@@ -2455,6 +2458,7 @@
     AC_DEFINE(HAVE_SOCKLEN_T)
     USE_PTHREADS=1
     LIB_SUFFIX=lib
+    IMPORT_LIB_SUFFIX=lib
     DLL_SUFFIX=dll
     MKSHLIB=
     DSO_LDOPTS=
@@ -2211,6 +2211,7 @@
     AC_DEFINE(BSD_SELECT)
     AC_DEFINE(TCPV40HDRS)
     LIB_SUFFIX=lib
+    IMPORT_LIB_SUFFIX=lib
     DLL_SUFFIX=dll
     RC=rc.exe
     PR_MD_ARCH_DIR=os2
@@ -3083,6 +3087,7 @@
 
 AC_SUBST(OBJ_SUFFIX)
 AC_SUBST(LIB_SUFFIX)
+AC_SUBST(IMPORT_LIB_SUFFIX)
 AC_SUBST(DLL_SUFFIX)
 AC_SUBST(ASM_SUFFIX)
 AC_SUBST(WRAP_LDFLAGS)
