diff -aur ImageMagick-7.0.1-3/MagickCore/nt-base.h.orig ImageMagick-7.0.1-3/MagickCore/nt-base.h > patch
--- ImageMagick-7.0.1-3/MagickCore/nt-base.h.orig	2016-05-11 18:03:33 -0400
+++ ImageMagick-7.0.1-3/MagickCore/nt-base.h	2016-05-11 18:04:48 -0400
@@ -39,6 +39,8 @@
 #include <errno.h>
 #include <malloc.h>
 #include <sys/utime.h>
+#include <winsock2.h>
+#include <ws2tcpip.h>
 #if defined(_DEBUG) && !defined(__MINGW32__)
 #include <crtdbg.h>
 #endif
diff -aur ImageMagick-7.0.1-3/MagickCore/nt-base.c.orig ImageMagick-7.0.1-3/MagickCore/nt-base.c > patch
--- ImageMagick-7.0.1-3/MagickCore/nt-base.c.orig	2016-05-11 18:05:52 -0400
+++ ImageMagick-7.0.1-3/MagickCore/nt-base.c	2016-05-11 18:09:20 -0400
@@ -1687,12 +1687,16 @@
   wchar_t
     file_specification[MagickPathExtent];
 
+  wchar_t WCDirectorySeparator[strlen(DirectorySeparator)+1];
+  MultiByteToWideChar(CP_UTF8,0,DirectorySeparator,-1,WCDirectorySeparator,
+                     strlen(DirectorySeparator)+1);
+
   assert(path != (const char *) NULL);
   length=MultiByteToWideChar(CP_UTF8,0,path,-1,file_specification,
     MagickPathExtent);
   if (length == 0)
     return((DIR *) NULL);
-  if(wcsncat(file_specification,(const wchar_t*) DirectorySeparator,
+  if(wcsncat(file_specification, WCDirectorySeparator,
        MagickPathExtent-wcslen(file_specification)-1) == (wchar_t*) NULL)
     return((DIR *) NULL);
   entry=(DIR *) AcquireMagickMemory(sizeof(DIR));
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Blob.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Blob.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Blob.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Blob.cpp	2016-06-03 13:22:18.419489600 +0300
@@ -13,8 +13,6 @@
 #include "Magick++/Blob.h"
 #include "Magick++/BlobRef.h"
 
-#include <string.h>
-
 Magick::Blob::Blob(void)
   : _blobRef(new Magick::BlobRef(0,0))
 {
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/BlobRef.cpp ImageMagick-7.0.1-8-p/Magick++/lib/BlobRef.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/BlobRef.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/BlobRef.cpp	2016-06-03 13:23:05.051825900 +0300
@@ -9,12 +9,8 @@
 #define MAGICKCORE_IMPLEMENTATION  1
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 
-#include "Magick++/Include.h"
 #include "Magick++/BlobRef.h"
 #include "Magick++/Exception.h"
-#include "Magick++/Thread.h"
-
-#include <string.h>
 
 Magick::BlobRef::BlobRef(const void* data_,const size_t length_)
   : allocator(Magick::Blob::NewAllocator),
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/CoderInfo.cpp ImageMagick-7.0.1-8-p/Magick++/lib/CoderInfo.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/CoderInfo.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/CoderInfo.cpp	2016-06-03 13:23:18.049526000 +0300
@@ -9,7 +9,6 @@
 #define MAGICKCORE_IMPLEMENTATION  1
 #define MAGICK_PLUSPLUS_IMPLEMENTATION  1
 
-#include "Magick++/Include.h"
 #include "Magick++/CoderInfo.h"
 #include "Magick++/Exception.h"
 
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Color.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Color.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Color.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Color.cpp	2016-06-03 13:23:37.062624500 +0300
@@ -9,9 +9,6 @@
 #define MAGICKCORE_IMPLEMENTATION
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 
-#include "Magick++/Include.h"
-#include <string>
-
 using namespace std;
 
 #include "Magick++/Color.h"
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Drawable.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Drawable.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Drawable.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Drawable.cpp	2016-06-03 13:24:06.666044400 +0300
@@ -10,9 +10,7 @@
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 #define MAGICK_DRAWABLE_IMPLEMENTATION
 
-#include "Magick++/Include.h"
 #include <math.h>
-#include <string>
 
 #include "Magick++/Drawable.h"
 #include "Magick++/Image.h"
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Exception.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Exception.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Exception.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Exception.cpp	2016-06-03 13:24:23.897044400 +0300
@@ -9,10 +9,7 @@
 #define MAGICKCORE_IMPLEMENTATION  1
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 
-#include "Magick++/Include.h"
-#include <string>
 #include <errno.h>
-#include <string.h>
 
 using namespace std;
 
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Functions.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Functions.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Functions.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Functions.cpp	2016-06-03 13:24:53.376044400 +0300
@@ -9,9 +9,6 @@
 #define MAGICKCORE_IMPLEMENTATION  1
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 
-#include "Magick++/Include.h"
-#include <string>
-
 using namespace std;
 
 #include "Magick++/Functions.h"
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Geometry.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Geometry.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Geometry.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Geometry.cpp	2016-06-03 13:25:06.583044400 +0300
@@ -9,8 +9,6 @@
 #define MAGICKCORE_IMPLEMENTATION  1
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 
-#include "Magick++/Include.h"
-#include <string>
 #include <ctype.h> // for isdigit
 #if !defined(MAGICKCORE_WINDOWS_SUPPORT)
 #include <strings.h>
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Image.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Image.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Image.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Image.cpp	2016-06-03 13:25:17.758044400 +0300
@@ -9,10 +9,7 @@
 #define MAGICKCORE_IMPLEMENTATION  1
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 
-#include "Magick++/Include.h"
 #include <cstdlib>
-#include <string>
-#include <string.h>
 #include <errno.h>
 #include <math.h>
 
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Magick++/ImageRef.h ImageMagick-7.0.1-8-p/Magick++/lib/Magick++/ImageRef.h
--- ImageMagick-7.0.1-8/Magick++/lib/Magick++/ImageRef.h	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Magick++/ImageRef.h	2016-06-03 13:26:21.323044400 +0300
@@ -12,9 +12,9 @@
 #if !defined(Magick_ImageRef_header)
 #define Magick_ImageRef_header
 
-#include <string>
 #include "Magick++/Include.h"
 #include "Magick++/Thread.h"
+#include <string>
 
 namespace Magick
 {
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Magick++/Include.h ImageMagick-7.0.1-8-p/Magick++/lib/Magick++/Include.h
--- ImageMagick-7.0.1-8/Magick++/lib/Magick++/Include.h	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Magick++/Include.h	2016-06-03 13:20:45.313801100 +0300
@@ -24,6 +24,7 @@
 #include <stdarg.h>
 #include <stdlib.h>
 #include <time.h>
+#include <string.h>
 #include <math.h>
 #include <sys/types.h>
 
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Magick++/Options.h ImageMagick-7.0.1-8-p/Magick++/lib/Magick++/Options.h
--- ImageMagick-7.0.1-8/Magick++/lib/Magick++/Options.h	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Magick++/Options.h	2016-06-03 13:16:53.143983600 +0300
@@ -15,8 +15,8 @@
 #if !defined(Magick_Options_header)
 #define Magick_Options_header
 
-#include "Magick++/Include.h"
 #include <string>
+#include "Magick++/Include.h"
 #include "Magick++/Color.h"
 #include "Magick++/Geometry.h"
 #include "Magick++/Drawable.h"
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Montage.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Montage.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Montage.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Montage.cpp	2016-06-03 13:26:32.298044400 +0300
@@ -8,10 +8,6 @@
 #define MAGICKCORE_IMPLEMENTATION  1
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 
-#include "Magick++/Include.h"
-#include <string>
-#include <string.h>
-
 #include "Magick++/Montage.h"
 #include "Magick++/Functions.h"
 
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Options.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Options.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Options.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Options.cpp	2016-06-03 13:26:44.889242100 +0300
@@ -11,9 +11,6 @@
 #define MAGICKCORE_IMPLEMENTATION  1
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 
-#include "Magick++/Include.h"
-#include <string>
-#include <string.h>
 #include <stdlib.h>
 #include <math.h>
 
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Statistic.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Statistic.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Statistic.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Statistic.cpp	2016-06-03 13:26:54.561795400 +0300
@@ -8,7 +8,6 @@
 #define MAGICKCORE_IMPLEMENTATION  1
 #define MAGICK_PLUSPLUS_IMPLEMENTATION  1
 
-#include "Magick++/Include.h"
 #include "Magick++/Exception.h"
 #include "Magick++/Statistic.h"
 #include "Magick++/Image.h"
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/Thread.cpp ImageMagick-7.0.1-8-p/Magick++/lib/Thread.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/Thread.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/Thread.cpp	2016-06-03 13:27:44.874448000 +0300
@@ -12,8 +12,6 @@
 #include "Magick++/Thread.h"
 #include "Magick++/Exception.h"
 
-#include <string.h>
-
 // Default constructor
 Magick::MutexLock::MutexLock(void)
 #if defined(MAGICKCORE_HAVE_PTHREAD)
diff -Naur ImageMagick-7.0.1-8/Magick++/lib/TypeMetric.cpp ImageMagick-7.0.1-8-p/Magick++/lib/TypeMetric.cpp
--- ImageMagick-7.0.1-8/Magick++/lib/TypeMetric.cpp	2016-06-01 01:23:00.000000000 +0300
+++ ImageMagick-7.0.1-8-p/Magick++/lib/TypeMetric.cpp	2016-06-03 13:27:52.337701600 +0300
@@ -10,7 +10,6 @@
 #define MAGICK_PLUSPLUS_IMPLEMENTATION 1
 
 #include "Magick++/TypeMetric.h"
-#include <string.h>
 
 Magick::TypeMetric::TypeMetric(void)
 {
--- ImageMagick-7.0.1-2/MagickCore/Makefile.am.orig	2016-05-10 02:35:13 -0400
+++ ImageMagick-7.0.1-2/MagickCore/Makefile.am	2016-05-10 02:37:41 -0400
@@ -42,10 +42,10 @@
 
 if WITH_MODULES
 MagickCore_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_SOURCES = $(MAGICKCORE_BASE_SRCS) $(MAGICKCORE_PLATFORM_SRCS)
-MagickCore_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_LIBADD = $(MAGICK_DEP_LIBS)
+MagickCore_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_LIBADD = $(MAGICK_DEP_LIBS) -lws2_32
 else
 MagickCore_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_SOURCES = $(MAGICKCORE_BASE_SRCS) $(MAGICKCORE_PLATFORM_SRCS) $(MAGICKCORE_CODER_SRCS) $(MAGICKCORE_FILTER_SRCS)
-MagickCore_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_LIBADD = $(MAGICK_DEP_LIBS)
+MagickCore_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_LIBADD = $(MAGICK_DEP_LIBS) -lws2_32
 endif # WITH_MODULES
 MagickCore_libMagickCore_@MAGICK_MAJOR_VERSION@_@MAGICK_ABI_SUFFIX@_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBRARY_EXTRA_CPPFLAGS)
 
