--- opencolorio/src/core/CMakeLists.txt.orig	2018-01-09 17:22:14.511093200 +0300
+++ opencolorio/src/core/CMakeLists.txt	2018-01-09 17:22:19.297367000 +0300
@@ -48,7 +48,7 @@
         target_link_libraries(OpenColorIO ${EXTERNAL_LIBRARIES})
     endif()
 
-    if(WIN32)
+    if(MSVC)
         # Mute a design issue where the Exception public class inherits 
         # from a STL Exception. STL classes are never supposed to
         # be exported among different dynamic libraries.
--- opencolorio/src/core_gpu_tests/CMakeLists.txt.orig	2018-01-09 17:22:14.511093200 +0300
+++ opencolorio/src/core_gpu_tests/CMakeLists.txt	2018-01-09 17:22:19.297367000 +0300
@@ -38,7 +38,9 @@
     # Mute a design issue where the Exception public class inherits 
     # from a STL Exception. STL classes are never supposed to
     # be exported among different dynamic libraries.
+    if(MSVC)
     set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} /wd4275")
+    endif()
 
 endif()
 
--- opencolorio/src/pyglue/CMakeLists.txt.orig	2018-01-09 17:22:56.163475600 +0300
+++ opencolorio/src/pyglue/CMakeLists.txt	2018-01-09 17:23:01.249766500 +0300
@@ -13,7 +13,7 @@
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-strict-aliasing -Wno-missing-field-initializers")
 endif()
 
-if(WIN32)
+if(MSVC)
     # Mute a design issue where the Exception public class inherits 
     # from a STL Exception. STL classes are never supposed to
     # be exported among different dynamic libraries.
