diff --git a/tools/MINGW-packages/mingw-w64-opencolorio-git/PKGBUILD b/tools/MINGW-packages/mingw-w64-opencolorio-git/PKGBUILD
index 12804f5ee..5283dfa13 100644
--- a/tools/MINGW-packages/mingw-w64-opencolorio-git/PKGBUILD
+++ b/tools/MINGW-packages/mingw-w64-opencolorio-git/PKGBUILD
@@ -55,7 +55,7 @@ prepare() {
 
 build() {
   [[ -d "build-${MINGW_CHOST}" ]] && rm -rf "build-${MINGW_CHOST}"
-  mkdir -p ${srcdir}/build-${MINGW_CHOST} && cd ${srcdir}/build-${MINGW_CHOST}
+  mkdir -p build-${MINGW_CHOST} && cd build-${MINGW_CHOST}
 
   MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
   ${MINGW_PREFIX}/bin/cmake \
@@ -75,7 +75,7 @@ build() {
     -DOCIO_BUILD_TESTS=OFF \
     -DOCIO_BUILD_GPU_TESTS=OFF \
     -DOCIO_BUILD_PYGLUE=OFF \
-    -DOCIO_BUILD_JNIGLUE=OFF \    
+    -DOCIO_BUILD_JNIGLUE=OFF \
     ../${_realname}
 
   make
