--- PKGBUILD.orig	2018-07-12 19:30:04.000000000 +0200
+++ PKGBUILD	2018-07-12 19:59:03.000000000 +0200
@@ -3,8 +3,9 @@
 _realname=opencolorio
 pkgbase=mingw-w64-${_realname}-git
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}-git"
+gitcommit=75b4749
 pkgver=811.75b4749
-pkgrel=1
+pkgrel=1.1
 pkgdesc="A color management framework for visual effects and animation (mingw-w64)"
 arch=('any')
 url="http://opencolorio.org"
@@ -14,14 +15,13 @@
 depends=("${MINGW_PACKAGE_PREFIX}-boost"
          "${MINGW_PACKAGE_PREFIX}-glew"
          "${MINGW_PACKAGE_PREFIX}-lcms2"
-         "${MINGW_PACKAGE_PREFIX}-python2"
          "${MINGW_PACKAGE_PREFIX}-tinyxml"
          "${MINGW_PACKAGE_PREFIX}-yaml-cpp")
 makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
              "${MINGW_PACKAGE_PREFIX}-pkg-config"
              "git")
 options=('strip')
-source=("${_realname}"::"git+https://github.com/imageworks/OpenColorIO.git"
+source=("${_realname}"::"git+https://github.com/imageworks/OpenColorIO.git#commit=${gitcommit}"
         #${_realname}-${pkgver}.tar.gz::https://github.com/imageworks/OpenColorIO/tarball/v${pkgver}.tar.gz
         mingw-w64.patch
         fix-redefinitions.patch
@@ -68,10 +68,15 @@
     -DUSE_EXTERNAL_LCMS=ON \
     -DOCIO_USE_BOOST_PTR=ON \
     -DOCIO_BUILD_PYGLUE=ON \
-    -DOCIO_PYGLUE_LINK=ON \
-    -DPYTHON=${MINGW_PREFIX}/bin/python2.exe \
     -DPYTHON_EXECUTABLE=${MINGW_PREFIX}/bin/python2.exe \
     -DOCIO_BUILD_TRUELIGHT=OFF \
+    -DOCIO_BUILD_APPS=OFF \
+    -DOCIO_BUILD_NUKE=OFF \
+    -DOCIO_BUILD_DOCS=OFF \
+    -DOCIO_BUILD_TESTS=OFF \
+    -DOCIO_BUILD_GPU_TESTS=OFF \
+    -DOCIO_BUILD_PYGLUE=OFF \
+    -DOCIO_BUILD_JNIGLUE=OFF \
     ../${_realname}
 
   make
