--- PKGBUILD.orig	2019-03-10 17:08:13.000000000 -0700
+++ PKGBUILD	2019-03-10 17:12:00.000000000 -0700
@@ -3,8 +3,10 @@
 _realname=opencolorio
 pkgbase=mingw-w64-${_realname}-git
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}-git"
-pkgver=815.15e96c1f
-pkgrel=1
+# we do not support the new OCIO v2 GPU API (which appeared at commit 809)
+gitcommit=a80857f7
+pkgver=808.a80857f7
+pkgrel=1.1
 pkgdesc="A color management framework for visual effects and animation (mingw-w64)"
 arch=('any')
 url="http://opencolorio.org"
@@ -14,14 +16,13 @@
 depends=("${MINGW_PACKAGE_PREFIX}-boost"
          "${MINGW_PACKAGE_PREFIX}-glew"
          "${MINGW_PACKAGE_PREFIX}-lcms2"
-         "${MINGW_PACKAGE_PREFIX}-python3"
          "${MINGW_PACKAGE_PREFIX}-tinyxml"
          "${MINGW_PACKAGE_PREFIX}-yaml-cpp")
 makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
              "${MINGW_PACKAGE_PREFIX}-pkg-config"
              "git")
 options=('strip')
-source=("${_realname}"::"git+https://github.com/imageworks/OpenColorIO.git"
+source=("${_realname}"::"git+https://github.com/imageworks/OpenColorIO.git#commit=${gitcommit}"
         #${_realname}-${pkgver}.tar.gz::https://github.com/imageworks/OpenColorIO/tarball/v${pkgver}.tar.gz
         mingw-w64.patch
         fix-redefinitions.patch
@@ -55,8 +56,6 @@
 build() {
   [[ -d "build-${MINGW_CHOST}" ]] && rm -rf "build-${MINGW_CHOST}"
   mkdir -p ${srcdir}/build-${MINGW_CHOST} && cd ${srcdir}/build-${MINGW_CHOST}
-  
-  local _py3ver=$(${MINGW_PREFIX}/bin/python3 -c "import sys;sys.stdout.write('.'.join(map(str, sys.version_info[:2])))")
 
   MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
   ${MINGW_PREFIX}/bin/cmake \
@@ -69,11 +68,14 @@
     -DUSE_EXTERNAL_LCMS=ON \
     -DOCIO_USE_BOOST_PTR=ON \
     -DOCIO_BUILD_PYGLUE=ON \
-    -DOCIO_PYGLUE_LINK=ON \
-    -DPYTHON_VERSION=${_py3ver} \
-    -DPYTHON=${MINGW_PREFIX}/bin/python3.exe \
-    -DPYTHON_EXECUTABLE=${MINGW_PREFIX}/bin/python3.exe \
     -DOCIO_BUILD_TRUELIGHT=OFF \
+    -DOCIO_BUILD_APPS=OFF \
+    -DOCIO_BUILD_NUKE=OFF \
+    -DOCIO_BUILD_DOCS=OFF \
+    -DOCIO_BUILD_TESTS=OFF \
+    -DOCIO_BUILD_GPU_TESTS=OFF \
+    -DOCIO_BUILD_PYGLUE=OFF \
+    -DOCIO_BUILD_JNIGLUE=OFF \    
     ../${_realname}
 
   make
