--- Modules/Platform/Darwin-Initialize.cmake.orig
+++ Modules/Platform/Darwin-Initialize.cmake
@@ -94,11 +94,17 @@
         NOT CMAKE_OSX_DEPLOYMENT_TARGET VERSION_EQUAL ${_CMAKE_OSX_DEPLOYMENT_TARGET})
       set(_CMAKE_OSX_SDKS_VER ${CMAKE_OSX_DEPLOYMENT_TARGET}${_CMAKE_OSX_SDKS_VER_SUFFIX_${CMAKE_OSX_DEPLOYMENT_TARGET}})
       set(_CMAKE_OSX_SYSROOT_CHECK "${_CMAKE_OSX_SDKS_DIR}/MacOSX${_CMAKE_OSX_SDKS_VER}.sdk")
-      message(WARNING
-        "CMAKE_OSX_DEPLOYMENT_TARGET is '${CMAKE_OSX_DEPLOYMENT_TARGET}' "
-        "but the matching SDK does not exist at:\n \"${_CMAKE_OSX_SYSROOT_CHECK}\"\n"
-        "Instead using SDK:\n \"${_CMAKE_OSX_SYSROOT_DEFAULT}\"."
-        )
+      if(IS_DIRECTORY "${_CMAKE_OSX_SYSROOT_DEFAULT}")
+        if(NOT "${CMAKE_OSX_DEPLOYMENT_TARGET}" STREQUAL "${_CURRENT_OSX_VERSION}")
+	  message(WARNING
+            "CMAKE_OSX_DEPLOYMENT_TARGET is '${CMAKE_OSX_DEPLOYMENT_TARGET}' "
+            "but the matching SDK does not exist at:\n \"${_CMAKE_OSX_SYSROOT_CHECK}\"\n"
+            "Instead using SDK:\n \"${_CMAKE_OSX_SYSROOT_DEFAULT}\"."
+            )
+	endif()
+      else()
+        set(_CMAKE_OSX_SYSROOT_DEFAULT "")
+      endif()
     endif()
   else()
     # Assume developer files are in root (such as Xcode 4.5 command-line tools).
