diff --git a/src/plugins/platforms/cocoa/qcocoawindow.mm b/src/plugins/platforms/cocoa/qcocoawindow.mm
index c551999..5a82cfb 100644
--- a/src/plugins/platforms/cocoa/qcocoawindow.mm
+++ b/src/plugins/platforms/cocoa/qcocoawindow.mm
@@ -133,6 +133,10 @@ - (void)handleWindowEvent:(NSEvent *)theEvent
     // (e.g., when closing the window by pressing the title bar close button).
     [self retain];
     [self.window superSendEvent:theEvent];
+
+    if (theEvent.type == NSLeftMouseUp)
+        [QNSView clearLimbo];
+
     bool windowStillAlive = self.window != nil; // We need to read before releasing
     [self release];
     if (!windowStillAlive)
@@ -457,6 +461,9 @@ - (void)dealloc
         child->m_parentCocoaWindow = 0;
     }
 
+    if (m_qtView && m_qtView.leftButtonRetained) {
+        [m_qtView sendToLimbo]; // limbo retains
+    }
     [m_contentView release];
     [m_nsWindow release];
     [m_windowCursor release];
diff --git a/src/plugins/platforms/cocoa/qnsview.h b/src/plugins/platforms/cocoa/qnsview.h
index 2d4ad7a..820ac60 100644
--- a/src/plugins/platforms/cocoa/qnsview.h
+++ b/src/plugins/platforms/cocoa/qnsview.h
@@ -79,8 +79,13 @@ Q_FORWARD_DECLARE_OBJC_CLASS(QT_MANGLE_NAMESPACE(QNSViewMouseMoveHelper));
     bool m_exposedOnMoveToWindow;
     NSEvent *m_currentlyInterpretedKeyEvent;
     bool m_isMenuView;
+    BOOL m_leftButtonRetained;
+    BOOL m_isInLimbo;
 }
 
+@property (nonatomic, readonly) BOOL leftButtonRetained;
+@property (nonatomic, readonly) BOOL inLimbo;
+
 - (id)init;
 - (id)initWithQWindow:(QWindow *)window platformWindow:(QCocoaWindow *) platformWindow;
 - (void) clearQWindowPointers;
@@ -111,6 +116,9 @@ Q_FORWARD_DECLARE_OBJC_CLASS(QT_MANGLE_NAMESPACE(QNSViewMouseMoveHelper));
 
 - (void)resetMouseButtons;
 
+- (void)sendToLimbo;
++ (void)clearLimbo;
+
 - (void)handleMouseEvent:(NSEvent *)theEvent;
 - (void)mouseDown:(NSEvent *)theEvent;
 - (void)mouseDragged:(NSEvent *)theEvent;
diff --git a/src/plugins/platforms/cocoa/qnsview.mm b/src/plugins/platforms/cocoa/qnsview.mm
index bce0455..2d49442 100644
--- a/src/plugins/platforms/cocoa/qnsview.mm
+++ b/src/plugins/platforms/cocoa/qnsview.mm
@@ -72,6 +72,8 @@
 
 static bool _q_dontOverrideCtrlLMB = false;
 
+static NSMutableSet *_q_leftButtonLimbo = nil;
+
 @interface NSEvent (Qt_Compile_Leopard_DeviceDelta)
   - (CGFloat)deviceDeltaX;
   - (CGFloat)deviceDeltaY;
@@ -127,12 +129,17 @@ - (void)cursorUpdate:(NSEvent *)theEvent
 
 @implementation QT_MANGLE_NAMESPACE(QNSView)
 
+@synthesize leftButtonRetained = m_leftButtonRetained;
+@synthesize inLimbo = m_isInLimbo;
+
 + (void)initialize
 {
     NSString **notificationNameVar = (NSString **)dlsym(RTLD_NEXT, "NSWindowDidChangeOcclusionStateNotification");
     if (notificationNameVar)
         _q_NSWindowDidChangeOcclusionStateNotification = *notificationNameVar;
 
+    _q_leftButtonLimbo = [[NSMutableSet alloc] init];
+
     _q_dontOverrideCtrlLMB = qt_mac_resolveOption(false, "QT_MAC_DONT_OVERRIDE_CTRL_LMB");
 }
 
@@ -170,6 +177,8 @@ - (id) init
 
         m_isMenuView = false;
         self.focusRingType = NSFocusRingTypeNone;
+        m_leftButtonRetained = NO;
+        m_isInLimbo = NO;
     }
     return self;
 }
@@ -269,7 +278,7 @@ - (void) globalFrameChanged:(NSNotification*)notification
 
 - (void)viewDidMoveToSuperview
 {
-    if (!(m_platformWindow->m_contentViewIsToBeEmbedded))
+    if (!m_platformWindow || !m_platformWindow->m_contentViewIsToBeEmbedded)
         return;
 
     if ([self superview]) {
@@ -830,6 +839,9 @@ - (void)mouseDown:(NSEvent *)theEvent
         return [super mouseDown:theEvent];
     m_sendUpAsRightButton = false;
 
+    if (!m_leftButtonRetained)
+        m_leftButtonRetained = true;
+
     // Handle any active poup windows; clicking outside them should close them
     // all. Don't do anything or clicks inside one of the menus, let Cocoa
     // handle that case. Note that in practice many windows of the Qt::Popup type
@@ -888,6 +900,10 @@ - (void)mouseUp:(NSEvent *)theEvent
 {
     if (m_window && (m_window->flags() & Qt::WindowTransparentForInput) )
         return [super mouseUp:theEvent];
+
+    if (m_leftButtonRetained)
+        m_leftButtonRetained = false;
+
     if (m_sendUpAsRightButton) {
         m_buttons &= ~Qt::RightButton;
         m_sendUpAsRightButton = false;
@@ -2094,4 +2110,16 @@ - (void)draggedImage:(NSImage*) img endedAt:(NSPoint) point operation:(NSDragOpe
     QWindowSystemInterface::handleMouseEvent(target, mapWindowCoordinates(m_window, target, qtWindowPoint), qtScreenPoint, m_buttons);
 }
 
+- (void)sendToLimbo
+{
+    m_isInLimbo = YES;
+    [_q_leftButtonLimbo addObject:self];
+    [self removeFromSuperviewWithoutNeedingDisplay];
+}
+
++ (void)clearLimbo
+{
+    [_q_leftButtonLimbo removeAllObjects];
+}
+
 @end
