diff --git a/configure b/configure
index 4ce83be..7f84b3f 100755
--- a/configure
+++ b/configure
@@ -5553,6 +5553,12 @@ if [ "$CFG_XCB" != "no" ]; then
 
         if [ "$CFG_XCB" = "qt" ]; then
             QT_CONFIG="$QT_CONFIG xcb-qt"
+
+            # ADSK  QTBUG-43784  GLX is not enabled when building with -qt-xcb
+            if compileTest qpa/xcb-glx "xcb-glx" $QMAKE_CFLAGS_XCB $QMAKE_LIBS_XCB; then
+                CFG_XCB_GLX=yes
+                QT_CONFIG="$QT_CONFIG xcb-glx"
+            fi
         else
             CFG_XCB="system"
             if [ -n "$PKG_CONFIG" ] && $PKG_CONFIG --exists "xcb >= 1.5" 2>/dev/null; then
