diff --git a/src/widgets/widgets/qmenubar.cpp b/src/widgets/widgets/qmenubar.cpp
index cca0853..34b2811 100644
--- a/src/widgets/widgets/qmenubar.cpp
+++ b/src/widgets/widgets/qmenubar.cpp
@@ -1345,7 +1345,7 @@ void QMenuBarPrivate::handleReparent()
     QWidget *newWindow = newParent ? newParent->window() : 0;
 
     if (oldWindow != newWindow) {
-        if (oldParent && oldParent != oldWindow)
+        if (oldParent && oldWindow && oldParent != oldWindow)
             oldWindow->removeEventFilter(q);
 
         if (newParent && newParent != newWindow)
diff --git a/src/widgets/widgets/qmenubar_p.h b/src/widgets/widgets/qmenubar_p.h
index ee615e7..2b847da 100644
--- a/src/widgets/widgets/qmenubar_p.h
+++ b/src/widgets/widgets/qmenubar_p.h
@@ -128,8 +128,8 @@ class QMenuBarPrivate : public QWidgetPrivate
 
     // reparenting
     void handleReparent();
-    QWidget *oldParent;
-    QWidget *oldWindow;
+    QPointer<QWidget> oldParent;
+    QPointer<QWidget> oldWindow;
 
     QList<QAction*> hiddenActions;
     //default action
