diff --git a/src/opengl/qgl.cpp b/src/opengl/qgl.cpp
index d84b59e..cf8cce8 100644
--- a/src/opengl/qgl.cpp
+++ b/src/opengl/qgl.cpp
@@ -463,6 +463,8 @@ QGLFormat QGLFormat::fromSurfaceFormat(const QSurfaceFormat &format)
     retFormat.setStereo(format.stereo());
     retFormat.setVersion(format.majorVersion(), format.minorVersion());
     retFormat.setProfile(static_cast<QGLFormat::OpenGLContextProfile>(format.profile()));
+       if (format.testOption(QSurfaceFormat::DebugContext))
+        retFormat.setOption(QGL::DebugContext);
     return retFormat;
 }
 
@@ -492,11 +494,7 @@ QSurfaceFormat QGLFormat::toSurfaceFormat(const QGLFormat &format)
     retFormat.setMajorVersion(format.majorVersion());
     retFormat.setMinorVersion(format.minorVersion());
     retFormat.setProfile(static_cast<QSurfaceFormat::OpenGLContextProfile>(format.profile()));
-    // QGLFormat has no way to set DeprecatedFunctions, that is, to tell that forward
-    // compatibility should not be requested. Some drivers fail to ignore the fwdcompat
-    // bit with compatibility profiles so make sure it is not set.
-    if (format.profile() == QGLFormat::CompatibilityProfile)
-        retFormat.setOption(QSurfaceFormat::DeprecatedFunctions);
+    retFormat.setOption(QSurfaceFormat::DebugContext, format.testOption(QGL::DebugContext));
     return retFormat;
 }
 
diff --git a/src/opengl/qgl.h b/src/opengl/qgl.h
index 99abdd2..f2abb29 100644
--- a/src/opengl/qgl.h
+++ b/src/opengl/qgl.h
@@ -70,6 +70,7 @@ namespace QGL
         HasOverlay              = 0x0100,
         SampleBuffers           = 0x0200,
         DeprecatedFunctions     = 0x0400,
+        DebugContext            = 0x0800,
         SingleBuffer            = DoubleBuffer    << 16,
         NoDepthBuffer           = DepthBuffer     << 16,
         ColorIndex              = Rgba            << 16,
@@ -80,7 +81,8 @@ namespace QGL
         IndirectRendering       = DirectRendering << 16,
         NoOverlay               = HasOverlay      << 16,
         NoSampleBuffers         = SampleBuffers   << 16,
-        NoDeprecatedFunctions   = DeprecatedFunctions << 16
+        NoDeprecatedFunctions   = DeprecatedFunctions << 16,
+        NoDebugContext          = DebugContext    << 16
     };
     Q_DECLARE_FLAGS(FormatOptions, FormatOption)
 }
