--- ilmbase-2.1.0/CMakeLists.txt.orig	2014-05-28 14:46:01.760200000 +0400
+++ ilmbase-2.1.0/CMakeLists.txt	2014-05-28 22:42:50.512800000 +0400
@@ -20,7 +20,7 @@
 # Allow the developer to select if Dynamic or Static libraries are built
 OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" ON)
 
-IF ( NOT WIN32)
+IF ( NOT WIN32 OR MINGW)
   ADD_DEFINITIONS ( -pthread )
 ENDIF ()
 
@@ -96,7 +96,7 @@
 ADD_SUBDIRECTORY ( Imath )
 ADD_SUBDIRECTORY ( IlmThread )
 
-IF (WIN32)
+IF (WIN32 AND NOT MINGW)
   FILE ( COPY ${CMAKE_CURRENT_SOURCE_DIR}/config.windows/IlmBaseConfig.h
     DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/config
     )
@@ -127,30 +127,41 @@
   PROPERTIES
   VERSION 23.0.0
   SOVERSION 23
+  OUTPUT_NAME Half
+  RUNTIME_OUTPUT_NAME "Half${ILMBASE_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME Half
   )
 SET_TARGET_PROPERTIES ( Iex
   PROPERTIES
   VERSION 23.0.0
   SOVERSION 23
-  OUTPUT_NAME "Iex${ILMBASE_LIBSUFFIX}"
+  OUTPUT_NAME Iex
+  RUNTIME_OUTPUT_NAME "Iex${ILMBASE_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME Iex
   )
 SET_TARGET_PROPERTIES ( Imath
   PROPERTIES
   VERSION 23.0.0
   SOVERSION 23
-  OUTPUT_NAME "Imath${ILMBASE_LIBSUFFIX}"
+  OUTPUT_NAME Imath
+  RUNTIME_OUTPUT_NAME "Imath${ILMBASE_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME Imath
   )
 SET_TARGET_PROPERTIES ( IlmThread
   PROPERTIES
   VERSION 23.0.0
   SOVERSION 23
-  OUTPUT_NAME "IlmThread${ILMBASE_LIBSUFFIX}"
+  OUTPUT_NAME IlmThread
+  RUNTIME_OUTPUT_NAME "IlmThread${ILMBASE_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME IlmThread
   )
 SET_TARGET_PROPERTIES ( IexMath
   PROPERTIES
   VERSION 23.0.0
   SOVERSION 23
-  OUTPUT_NAME "IexMath${ILMBASE_LIBSUFFIX}"
+  OUTPUT_NAME IexMath
+  RUNTIME_OUTPUT_NAME "IexMath${ILMBASE_LIBSUFFIX}"
+  ARCHIVE_OUTPUT_NAME IexMath
   )
 
 IF ( NOT WIN32 )
@@ -190,7 +201,7 @@
 Name: IlmBase
 Description: Base math and exception libraries
 Version: ${ILMBASE_VERSION}
-Libs: -L\${libdir} -lImath${ILMBASE_LIBSUFFIX} -lIexMath${ILMBASE_LIBSUFFIX} -lHalf -lIex${ILMBASE_LIBSUFFIX} -lIlmThread${ILMBASE_LIBSUFFIX} -pthread
+Libs: -L\${libdir} -lImath -lIexMath -lHalf -lIex -lIlmThread -pthread
 Cflags: -pthread -I\${OpenEXR_includedir}
 ")
 
--- ilmbase-2.1.0/IlmThread/CMakeLists.txt.orig	2014-05-28 14:46:01.760200000 +0400
+++ ilmbase-2.1.0/IlmThread/CMakeLists.txt	2014-05-28 22:42:50.512800000 +0400
@@ -10,7 +10,7 @@
   IlmThreadSemaphorePosixCompat.cpp
   IlmThreadSemaphorePosix.cpp
 )
-IF (WIN32)
+IF (WIN32 AND NOT MINGW)
   SET( ILMTHREAD_LIBRARY_SOURCES ${ILMTHREAD_LIBRARY_SOURCES}
     IlmThreadMutexWin32.cpp
     IlmThreadSemaphoreWin32.cpp
