--- ilmbase-2.2.0/ImathTest/CMakeLists.txt.orig	2016-10-14 10:21:55.569780400 +0300
+++ ilmbase-2.2.0/ImathTest/CMakeLists.txt	2016-10-14 10:22:07.781402300 +0300
@@ -25,6 +25,11 @@
    testTinySVD.cpp
   testVec.cpp
   )
+
+if (MINGW AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 6.1.0 OR CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 6.1.0))  # 6.1.0 is the last known bad version
+      string (REPLACE -O3 -O2 CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+      string (REPLACE -O2 -O0 CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+endif ()
 
 ADD_TEST ( TestImath ImathTest )
 TARGET_LINK_LIBRARIES ( ImathTest Imath Iex )
