--- tests/pkcs11/pkcs11-mock.c.orig	2016-05-10 13:14:55.000000000 -0700
+++ tests/pkcs11/pkcs11-mock.c	2016-05-24 14:18:03.000000000 -0700
@@ -17,6 +17,40 @@
 #include "pkcs11-mock.h"
 #include <stdlib.h>
 
+#ifdef __APPLE__
+#ifndef __MAC_OS_X_VERSION_MIN_REQUIRED
+#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1050
+#include <Availability.h>
+#else
+#include <AvailabilityMacros.h>
+#endif
+#endif
+#if __MAC_OS_X_VERSION_MIN_REQUIRED <= 1060
+static char * strndup (char const *s, size_t n);
+static size_t strnlen (const char *string, size_t maxlen);
+
+static size_t
+strnlen (const char *string, size_t maxlen)
+{
+  const char *end = memchr (string, '\0', maxlen);
+  return end ? (size_t) (end - string) : maxlen;
+}
+
+static char *
+strndup (char const *s, size_t n)
+{
+  size_t len = strnlen (s, n);
+  char *new = malloc (len + 1);
+
+  if (new == NULL)
+    return NULL;
+
+  new[len] = '\0';
+  return memcpy (new, s, len);
+}
+#endif
+#endif
+
 /* This is a very basic mock PKCS #11 module that will return a given fixed
  * certificate, and public key for all searches. It will also provide a
  * CKO_X_CERTIFICATE_EXTENSION so that it can be used as a p11-kit trust
