--- configure.ac.orig	2020-06-03 08:00:25.000000000 -0500
+++ configure.ac	2020-06-04 09:31:30.000000000 -0500
@@ -145,10 +145,10 @@
     dnl Try to use -no_weak_imports if available. This makes sure we
     dnl error out when linking to a function that doesn't exist in the
     dnl intended minimum runtime version.
-    LDFLAGS="$LDFLAGS -Wl,-no_weak_imports"
-    AC_MSG_CHECKING([whether the linker supports -Wl,-no_weak_imports])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <sys/select.h>], [fd_set rfds; FD_ZERO(&rfds); FD_SET(0, &rfds);])],
-      [AC_MSG_RESULT(yes)], [AC_MSG_RESULT(no); LDFLAGS="$save_LDFLAGS"])
+#     LDFLAGS="$LDFLAGS -Wl,-no_weak_imports"
+#     AC_MSG_CHECKING([whether the linker supports -Wl,-no_weak_imports])
+#     AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <sys/select.h>], [fd_set rfds; FD_ZERO(&rfds); FD_SET(0, &rfds);])],
+#       [AC_MSG_RESULT(yes)], [AC_MSG_RESULT(no); LDFLAGS="$save_LDFLAGS"])
   ;;
   *solaris*)
     have_elf=yes
