--- Portfile.orig	2016-08-31 17:37:05.000000000 +0200
+++ Portfile	2016-07-02 07:31:59.000000000 +0200
@@ -1,11 +1,12 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
-# $Id: Portfile 151014 2016-08-05 15:44:29Z ryandesign@macports.org $
+# $Id$
 
 PortSystem                  1.0
 PortGroup                   muniversal 1.0
 
 name                        fontconfig
-version                     2.12.1
+version                     2.12.0
+revision                    1
 categories                  graphics
 maintainers                 ryandesign
 license                     fontconfig
@@ -21,8 +22,8 @@
 master_sites                https://www.freedesktop.org/software/fontconfig/release/
 use_bzip2                   yes
 
-checksums                   rmd160  66907dbb317309bcb5013ea45c53dbf5050e6376 \
-                            sha256  b449a3e10c47e1d1c7a6ec6e2016cca73d3bd68fbbd4f0ae5cc6b573f7d6c7f3
+checksums                   rmd160  00b9af80c2abf6c69bf542f4d21a30a32ae728f4 \
+                            sha256  b433e4efff1f68fdd8aac221ed1df3ff1e580ffedbada020a703fe64017d8224
 
 depends_build               port:pkgconfig
 
