--- Portfile.orig	2018-03-26 15:09:11.000000000 +0200
+++ Portfile	2018-03-26 15:09:55.000000000 +0200
@@ -46,7 +46,7 @@
                     port:fontconfig \
                     port:freetype \
                     port:jbig2dec \
-                    port:jpeg \
+                    path:lib/libjpeg.dylib:jpeg \
                     port:libiconv \
                     port:libidn \
                     port:libpaper \
@@ -61,7 +61,9 @@
 
 extract.only        ${distname}.tar.gz \
                     ghostscript-fonts-other-6.0.tar.gz \
-                    ghostscript-fonts-std-8.11.tar.gz
+                    ghostscript-fonts-std-8.11.tar.gz \
+                    ghostscript-9.10-cross.patch \
+                    ghostscript-9.05-cross-so.patch
 
 post-extract {
     system -W ${workpath} "unzip '${distpath}/${mappingresources_commit}.zip'"
@@ -105,6 +107,15 @@
 
 build.target        so
 
+if {[variant_isset universal]} {
+        build.target-delete so
+        build.target-append cross_prepare so
+        lappend merger_build_env(i386) ARCH=i386
+        lappend merger_build_env(x86_64) ARCH=x86_64
+        lappend merger_build_env(ppc) ARCH=ppc
+        lappend merger_build_env(ppc64) ARCH=ppc64
+}
+
 destroot.target     soinstall
 post-destroot {
     ln -s gsc ${destroot}${prefix}/bin/gs
